// @roddan/form-editor/es/elements/container/render-container.js
import React from 'react';
import { observer } from 'mobx-react-lite';
import c from 'classnames';
import { useEditorContext } from "../../context";
import { prefixCls } from "../../const";
import { handleSort } from "../../utils";
import store from "../../store";
import { ElementLayout } from "../../components";
import { ElementsList } from "../export";
// import { ReactSortable } from '@/components/react-sortable';
import { ReactSortable } from '@roddan/ui';
import "./style.less";
export var RenderContainerContent = function RenderContainerContent(_ref) {
  var _element$children, _element$children2;
  var element = _ref.element;
  var _store$formAttrs = store.formAttrs,
    horizontalGap = _store$formAttrs.horizontalGap,
    verticalGap = _store$formAttrs.verticalGap,
    id = _store$formAttrs.id;
  var _useEditorContext = useEditorContext(),
    mode = _useEditorContext.mode;
  var parentId = element.parentId;
  return /*#__PURE__*/React.createElement(ElementLayout, {
    element: element
  }, /*#__PURE__*/React.createElement(ReactSortable, {
    list: element.children || [],
    animation: 150,
    group: "nested",
    onSort: function onSort(e) {
      return handleSort(e, element.id);
    },
    rowProps: {
      className: c([prefixCls('row'), prefixCls('row-el'), mode === 'design' && !((_element$children = element.children) !== null && _element$children !== void 0 && _element$children.length) ? prefixCls('row-empty') : '']),
      gutter: [horizontalGap, verticalGap],
      'data-id': element.id,
      'data-parent-id': parentId || id
    },
    forbidden: mode !== 'design'
  }, (_element$children2 = element.children) === null || _element$children2 === void 0 ? void 0 : _element$children2.map(function (item) {
    var _ElementsList;
    var Component = (_ElementsList = ElementsList[item.type]) === null || _ElementsList === void 0 ? void 0 : _ElementsList.render;
    if (!Component) return null;
    store.flatElement(item);
    return /*#__PURE__*/React.createElement(Component, {
      key: item.id || String(+new Date()),
      fieldValue: store.fieldValues[item.id],
      element: item
    });
  })));
};
export var RenderContainer = observer(RenderContainerContent);
