// @roddan/form-editor/lib/elements/image/image-setting.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SettingImage = void 0;
var _react = _interopRequireDefault(require("react"));
var _antd = require("antd");
var _mobxReactLite = require("mobx-react-lite");
var _components = require("../../components");
var _store = _interopRequireDefault(require("../../store"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var SettingImageContent = function SettingImageContent() {
  var _store$selectedElemen = _store.default.selectedElement,
    defaultImgSrc = _store$selectedElemen.defaultImgSrc,
    placeholder = _store$selectedElemen.placeholder,
    preview = _store$selectedElemen.preview,
    previewSrc = _store$selectedElemen.previewSrc;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_components.SettingWrap, {
    title: "\u5143\u7D20\u8BBE\u7F6E"
  }, /*#__PURE__*/_react.default.createElement(_components.SettingItem, {
    label: "\u9ED8\u8BA4\u5730\u5740"
  }, /*#__PURE__*/_react.default.createElement(_antd.Input, {
    value: defaultImgSrc,
    onChange: function onChange(e) {
      _store.default.setSelectedProp('defaultImgSrc', e.target.value);
    }
  })), /*#__PURE__*/_react.default.createElement(_components.SettingItem, {
    label: "\u5360\u4F4D\u5730\u5740"
  }, /*#__PURE__*/_react.default.createElement(_antd.Input, {
    value: placeholder,
    onChange: function onChange(e) {
      _store.default.setSelectedProp('placeholder', e.target.value);
    }
  })), /*#__PURE__*/_react.default.createElement(_components.SettingItem, {
    label: "\u652F\u6301\u9884\u89C8"
  }, /*#__PURE__*/_react.default.createElement(_antd.Switch, {
    checked: preview,
    size: "small",
    onChange: function onChange(val) {
      return _store.default.setSelectedProp('preview', val);
    }
  })), preview && /*#__PURE__*/_react.default.createElement(_components.SettingItem, {
    label: "\u9884\u89C8\u5730\u5740"
  }, /*#__PURE__*/_react.default.createElement(_antd.Input, {
    value: previewSrc,
    onChange: function onChange(e) {
      _store.default.setSelectedProp('previewSrc', e.target.value);
    }
  }))));
};
var SettingImage = exports.SettingImage = (0, _mobxReactLite.observer)(SettingImageContent);
