// @roddan/form-editor/lib/elements/image/render-image.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RenderImage = void 0;
var _react = _interopRequireDefault(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _antd = require("antd");
var _hooks = require("../../hooks");
var _components = require("../../components");
var _types = require("../../types");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var RenderImageContent = function RenderImageContent(_ref) {
  var element = _ref.element,
    fieldValue = _ref.fieldValue;
  var placeholder = element.placeholder,
    preview = element.preview,
    defaultImgSrc = element.defaultImgSrc,
    previewSrc = element.previewSrc;
  var _useRegisterEvents = (0, _hooks.useRegisterEvents)(element),
    eventFunctions = _useRegisterEvents.eventFunctions;
  (0, _hooks.useFormUpdate)(function () {
    var _eventFunctions$EEven;
    (_eventFunctions$EEven = eventFunctions[_types.EEventAction.ON_LOADED]) === null || _eventFunctions$EEven === void 0 || _eventFunctions$EEven.call(eventFunctions);
  }, [eventFunctions[_types.EEventAction.ON_LOADED]]);
  return /*#__PURE__*/_react.default.createElement(_components.ElementLayout, {
    element: element
  }, /*#__PURE__*/_react.default.createElement(_antd.Image, {
    src: fieldValue || defaultImgSrc,
    fallback: placeholder,
    preview: preview ? previewSrc ? {
      src: previewSrc
    } : true : false
  }));
};
var RenderImage = exports.RenderImage = (0, _mobxReactLite.observer)(RenderImageContent);
