// @roddan/form-editor/es/views/material/index.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import React, { useEffect, useMemo, useRef } from 'react';
import { observer } from 'mobx-react-lite';
import { prefixCls } from "../../const";
import { idCreator } from "../../utils";
import { ElementsList } from "../../elements";
// import { ReactSortable } from '@/components/react-sortable';
import { ReactSortable } from '@roddan/ui';
import { CONTAINERS } from "../../const";
import store from "../../store";
import DragItem from "./drag-item";
import "./style.less";
var titles = ['基础组件', '容器组件', '自定义组件'];
export var Material = observer(function () {
  var wrapEl = useRef(null);
  var renderList = useMemo(function () {
    var basic = [];
    var container = [];
    var custom = [];
    Object.entries(ElementsList)
    // .filter(([name]) => !FILTER_ELEMENT.includes(name))
    .forEach(function (_ref) {
      var _ref2 = _slicedToArray(_ref, 2),
        name = _ref2[0],
        el = _ref2[1];
      if (CONTAINERS.includes(name)) {
        container.push(el);
      } else {
        basic.push(el);
      }
    });
    return [basic, container, custom];
  }, [ElementsList]);
  useEffect(function () {
    var _wrapEl$current;
    var handleClick = function handleClick(e) {
      var _item$classList;
      var item = e.target;
      if (item !== null && item !== void 0 && (_item$classList = item.classList) !== null && _item$classList !== void 0 && _item$classList.contains(prefixCls('drag-item'))) {
        var type = item.dataset.type;
        var initialData = ElementsList[type].initialData;
        store.appendEl(_objectSpread(_objectSpread({}, initialData), {}, {
          type: type,
          id: idCreator(),
          parentId: store.formAttrs.id
        }));
      }
    };
    (_wrapEl$current = wrapEl.current) === null || _wrapEl$current === void 0 || _wrapEl$current.addEventListener('click', handleClick);
    return function () {
      var _wrapEl$current2;
      (_wrapEl$current2 = wrapEl.current) === null || _wrapEl$current2 === void 0 || _wrapEl$current2.removeEventListener('click', handleClick);
    };
  }, []);
  return /*#__PURE__*/React.createElement("div", {
    className: prefixCls('material'),
    ref: wrapEl
  }, /*#__PURE__*/React.createElement("div", {
    className: prefixCls('title')
  }, "\u7EC4\u4EF6\u5E93"), /*#__PURE__*/React.createElement("div", {
    style: {
      height: 10,
      backgroundColor: '#f5f5f5'
    }
  }), renderList.map(function (list, i) {
    return /*#__PURE__*/React.createElement(React.Fragment, {
      key: i
    }, !!list.length && /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
      className: prefixCls('title')
    }, titles[i]), /*#__PURE__*/React.createElement(ReactSortable, {
      list: list,
      className: prefixCls('element-wrap'),
      animation: 150,
      sort: false,
      group: {
        name: 'nested',
        pull: 'clone',
        put: false
      }
    }, list.map(function (item) {
      return /*#__PURE__*/React.createElement(DragItem, {
        key: item.type,
        item: item
      });
    }))));
  }));
});
