// @roddan/form-editor/lib/views/settings/form-setting.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _antd = require("antd");
var _mobxReactLite = require("mobx-react-lite");
var _components = require("../../components");
var _types = require("../../types");
var _store = _interopRequireDefault(require("../../store"));
var _const = require("../../const");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var FormSetting = function FormSetting() {
  var _store$formAttrs = _store.default.formAttrs,
    id = _store$formAttrs.id,
    formName = _store$formAttrs.formName,
    horizontalGap = _store$formAttrs.horizontalGap,
    verticalGap = _store$formAttrs.verticalGap;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _const.prefixCls)('form-setting')
  }, /*#__PURE__*/_react.default.createElement(_components.SettingWrap, {
    title: "\u57FA\u672C\u5C5E\u6027"
  }, /*#__PURE__*/_react.default.createElement(_components.SettingItem, {
    label: "\u8868\u5355id"
  }, id), /*#__PURE__*/_react.default.createElement(_components.SettingItem, {
    label: "\u8868\u5355\u540D\u79F0"
  }, /*#__PURE__*/_react.default.createElement(_antd.Input, {
    value: formName,
    onChange: function onChange(e) {
      _store.default.setFormAttr('formName', e.target.value);
    }
  })), /*#__PURE__*/_react.default.createElement(_components.SettingItem, {
    label: "\u6C34\u5E73\u95F4\u9694"
  }, /*#__PURE__*/_react.default.createElement(_antd.InputNumber, {
    value: horizontalGap,
    min: 0,
    onChange: function onChange(val) {
      _store.default.setFormAttr('horizontalGap', Number(val));
    }
  })), /*#__PURE__*/_react.default.createElement(_components.SettingItem, {
    label: "\u5782\u76F4\u95F4\u9694"
  }, /*#__PURE__*/_react.default.createElement(_antd.InputNumber, {
    value: verticalGap,
    min: 0,
    onChange: function onChange(val) {
      _store.default.setFormAttr('verticalGap', Number(val));
    }
  }))), /*#__PURE__*/_react.default.createElement(_components.SettingWrap, {
    title: "\u4E8B\u4EF6"
  }, /*#__PURE__*/_react.default.createElement(_components.EventSetting, {
    type: "form",
    eventActions: [_types.EEventAction.FORM_LOADED],
    eventTypeOptions: [_types.EEventType.UPDATE_SERVICE]
  })), /*#__PURE__*/_react.default.createElement(_components.SettingWrap, {
    title: "\u6837\u5F0F",
    style: {
      flex: 1,
      height: 0
    }
  }, /*#__PURE__*/_react.default.createElement(_components.CustomCssSetting, {
    type: "form"
  })));
};
var _default = exports.default = (0, _mobxReactLite.observer)(FormSetting);
