// @roddan/form-editor/lib/views/settings/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _antd = require("antd");
var _mobxReactLite = require("mobx-react-lite");
var _react = _interopRequireDefault(require("react"));
var _elementSetting = _interopRequireDefault(require("./element-setting"));
var _formSetting = _interopRequireDefault(require("./form-setting"));
var _components = require("../../components");
var _const = require("../../const");
var _store = require("../../store");
require("./style.less");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
// console.log('prefixCls', prefixCls);
// debugger;

var items = [{
  key: 'element',
  label: "\u7EC4\u4EF6\u5C5E\u6027",
  closable: false,
  children: /*#__PURE__*/_react.default.createElement(_elementSetting.default, null)
}, {
  key: 'form',
  label: "\u8868\u5355\u5C5E\u6027",
  closable: false,
  children: /*#__PURE__*/_react.default.createElement(_formSetting.default, null)
}, {
  key: 'service',
  label: "\u8868\u5355\u670D\u52A1",
  closable: false,
  children: /*#__PURE__*/_react.default.createElement(_components.FormService, null)
}];
var Settings = function Settings() {
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _const.prefixCls)('setting')
  }, /*#__PURE__*/_react.default.createElement(_antd.Tabs, {
    activeKey: _store.tabStore.formTab,
    type: "editable-card",
    hideAdd: true,
    items: items,
    className: "".concat((0, _const.prefixCls)('form-setting-tab')),
    onChange: function onChange(tab) {
      _store.tabStore.setFormTab(tab);
    }
  }));
};
var _default = exports.default = (0, _mobxReactLite.observer)(Settings);
