// @roddan/form-editor/es/elements/number/render-number.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import React, { useMemo } from 'react';
import { InputNumber } from 'antd';
import { observer } from 'mobx-react-lite';
import store from "../../store";
import { useRegisterEvents, useFormUpdate } from "../../hooks";
import { EEventAction } from "../../types";
import { ElementLayout } from "../../components";
import { EValueType } from "./const";
var RenderNumberContent = function RenderNumberContent(_ref) {
  var _ref$element = _ref.element,
    element = _ref$element === void 0 ? {} : _ref$element,
    fieldValue = _ref.fieldValue;
  var id = element.id,
    minNum = element.minNum,
    maxNum = element.maxNum,
    valueType = element.valueType;
  var _useRegisterEvents = useRegisterEvents(element),
    eventFunctions = _useRegisterEvents.eventFunctions;
  var _useMemo = useMemo(function () {
      switch (valueType) {
        case EValueType.INT:
          return [0, 1];
        case EValueType.ONE_DECIMAL:
          return [1, 0.1];
        // case EValueType.TWO_DECIMAL:
        default:
          return [2, 0.01];
      }
    }, [valueType]),
    _useMemo2 = _slicedToArray(_useMemo, 2),
    precision = _useMemo2[0],
    step = _useMemo2[1];
  var handleEvent = function handleEvent(action) {
    return function (e) {
      var _eventFunctions$actio;
      (_eventFunctions$actio = eventFunctions[action]) === null || _eventFunctions$actio === void 0 || _eventFunctions$actio.call(eventFunctions, e.target.value);
    };
  };
  var handleChange = function handleChange(val) {
    store.setFieldValue(id, val);
  };
  useFormUpdate(function () {
    var _eventFunctions$EEven;
    (_eventFunctions$EEven = eventFunctions[EEventAction.ON_LOADED]) === null || _eventFunctions$EEven === void 0 || _eventFunctions$EEven.call(eventFunctions);
  }, [eventFunctions[EEventAction.ON_LOADED]]);
  useFormUpdate(function () {
    var _eventFunctions$EEven2;
    (_eventFunctions$EEven2 = eventFunctions[EEventAction.VALUE_CHANGE]) === null || _eventFunctions$EEven2 === void 0 || _eventFunctions$EEven2.call(eventFunctions, fieldValue);
  }, [fieldValue]);
  return /*#__PURE__*/React.createElement(ElementLayout, {
    element: element
  }, /*#__PURE__*/React.createElement(InputNumber, {
    value: fieldValue,
    id: id,
    onChange: handleChange,
    onFocus: handleEvent(EEventAction.ON_FOCUS),
    onBlur: handleEvent(EEventAction.ON_BLUR),
    min: minNum,
    max: maxNum,
    precision: precision,
    step: step
  }));
};
export var RenderNumber = observer(RenderNumberContent);
