// @roddan/form-editor/lib/components/element-layout/selected-actions.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SelectedActions = void 0;
var _react = _interopRequireDefault(require("react"));
var _styledComponents = _interopRequireWildcard(require("styled-components"));
var _icons = require("@ant-design/icons");
var _store = _interopRequireDefault(require("../../store"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var StyledDiv = _styledComponents.default.div.withConfig({
  displayName: "StyledDiv",
  componentId: "form-editor-42cd__sc-k1gxpm-0"
})(function () {
  return "\n    ".concat((0, _styledComponents.css)({
    position: 'absolute',
    right: 0,
    top: -8,
    display: 'flex',
    zIndex: 20
  }).join(';'), "\n    .fe-selected-action{\n      ").concat((0, _styledComponents.css)({
    marginRight: 6,
    borderColor: '#409EFF',
    color: '#409EFF',
    background: '#fff',
    width: 22,
    height: 22,
    lineHeight: '20px',
    textAlign: 'center',
    borderRadius: '50%',
    fontSize: 12,
    border: '1px solid',
    cursor: 'pointer'
  }).join(';'), "\n      :hover {\n        color: white;\n      }\n    }\n    .copy-action:hover {\n      background: #409EFF;\n    }\n    .delete-action {\n      border-color: #F56C6C;\n      color: #F56C6C;\n    }\n    .delete-action:hover {\n      background: #F56C6C;\n    }\n  ");
});
var SelectedActions = exports.SelectedActions = function SelectedActions() {
  return /*#__PURE__*/_react.default.createElement(StyledDiv, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "fe-selected-action copy-action",
    onClick: function onClick() {
      var newEl = _store.default.copyEl(_store.default.selectedElement);
      _store.default.setSelectedElement(newEl);
    }
  }, /*#__PURE__*/_react.default.createElement(_icons.CopyOutlined, null)), /*#__PURE__*/_react.default.createElement("div", {
    className: "fe-selected-action delete-action",
    onClick: function onClick() {
      _store.default.deleteEl(_store.default.selectedElement);
      _store.default.setSelectedElement({});
    }
  }, /*#__PURE__*/_react.default.createElement(_icons.DeleteOutlined, null)));
};
