// @roddan/form-editor/lib/components/element-layout/wrap-el.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WrapEl = void 0;
var _react = _interopRequireWildcard(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _store = _interopRequireWildcard(require("../../store"));
var _hooks = require("../../hooks");
var _eventStore = _interopRequireDefault(require("../../store/eventStore"));
var _selectedActions = require("./selected-actions");
var _styled = require("./styled");
var _const = require("../../const");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
var EventIcon = (0, _mobxReactLite.observer)(function (_ref) {
  var events = _ref.events;
  if (!(events !== null && events !== void 0 && events.length)) return null;
  var validate = events.every(function (event) {
    var eventTargets = event.eventTargets;
    if (!(eventTargets !== null && eventTargets !== void 0 && eventTargets.length)) return true;
    return eventTargets.every(function (target) {
      var targetElementId = target.targetElementId,
        targetServiceId = target.targetServiceId;
      if (!targetElementId && !targetServiceId) return true;
      if (targetElementId) return !!_store.default.getElement(targetElementId);
      if (targetServiceId) return !!_store.default.getService(targetServiceId);
      return true;
    });
  });
  return /*#__PURE__*/_react.default.createElement(_styled.Icon, {
    validate: validate
  });
});
var WrapDesignEl = (0, _mobxReactLite.observer)(function (_ref2) {
  var _store$selectedElemen, _store$selectedElemen2;
  var children = _ref2.children,
    el = _ref2.el;
  var ref = (0, _react.useRef)(null);
  (0, _hooks.useDesignEffect)(function () {
    _eventStore.default.iterateEl(el);
  }, [el.events]);
  var handleSelect = (0, _react.useCallback)(function () {
    _store.default.setSelectedElement(el);
    _store.tabStore.init();
  }, [el]);
  return /*#__PURE__*/_react.default.createElement(_styled.DesignWrapDiv, {
    selected: ((_store$selectedElemen = _store.default.selectedElement) === null || _store$selectedElemen === void 0 ? void 0 : _store$selectedElemen.id) === el.id,
    onMouseDownCapture: handleSelect,
    ref: ref
  }, /*#__PURE__*/_react.default.createElement(_styled.Mask, {
    horizontal: _store.default.formAttrs.horizontalGap + 2,
    vertical: _store.default.formAttrs.verticalGap + 2,
    hide: _const.CONTAINERS.includes(el.type)
  }), ((_store$selectedElemen2 = _store.default.selectedElement) === null || _store$selectedElemen2 === void 0 ? void 0 : _store$selectedElemen2.id) === el.id && /*#__PURE__*/_react.default.createElement(_selectedActions.SelectedActions, null), children, /*#__PURE__*/_react.default.createElement(EventIcon, {
    events: el.events
  }));
});
var WrapEl = exports.WrapEl = function WrapEl(_ref3) {
  var children = _ref3.children,
    el = _ref3.el,
    mode = _ref3.mode;
  if (mode !== 'design') return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, children);
  return /*#__PURE__*/_react.default.createElement(WrapDesignEl, {
    el: el
  }, children);
};
