// @roddan/form-editor/es/elements/switch/render-switch.js
import React, { useMemo } from 'react';
import { Switch } from 'antd';
import { observer } from 'mobx-react-lite';
import { useRegisterEvents, useFormUpdate } from "../../hooks";
import { isNil } from 'lodash-es';
import store from "../../store";
import { ElementLayout } from "../../components";
import { EEventAction } from "../../types";
import { getValueFromInput } from "../../utils";
var RenderSwitchContent = function RenderSwitchContent(_ref) {
  var element = _ref.element,
    fieldValue = _ref.fieldValue;
  var id = element.id,
    checkedValue = element.checkedValue;
  var _useRegisterEvents = useRegisterEvents(element),
    eventFunctions = _useRegisterEvents.eventFunctions;
  var realCheckedValue = useMemo(function () {
    return getValueFromInput(checkedValue);
  }, [checkedValue]);
  useFormUpdate(function () {
    var _eventFunctions$EEven;
    (_eventFunctions$EEven = eventFunctions[EEventAction.ON_LOADED]) === null || _eventFunctions$EEven === void 0 || _eventFunctions$EEven.call(eventFunctions);
  }, [eventFunctions[EEventAction.ON_LOADED]]);
  useFormUpdate(function () {
    var _eventFunctions$EEven2;
    (_eventFunctions$EEven2 = eventFunctions[EEventAction.VALUE_CHANGE]) === null || _eventFunctions$EEven2 === void 0 || _eventFunctions$EEven2.call(eventFunctions, fieldValue === realCheckedValue);
  }, [fieldValue, realCheckedValue]);
  return /*#__PURE__*/React.createElement(ElementLayout, {
    element: element
  }, /*#__PURE__*/React.createElement(Switch, {
    checked: !isNil(fieldValue) && fieldValue === realCheckedValue,
    onChange: function onChange(checked) {
      store.setFieldValue(id, checked ? realCheckedValue : '');
    }
  }));
};
export var RenderSwitch = observer(RenderSwitchContent);
