// @roddan/form-editor/es/components/setting-common/index.js
import React from 'react';
import { Popover } from 'antd';
import styled from 'styled-components';
import { QuestionCircleOutlined } from '@ant-design/icons';
import { prefixCls } from "../../const";
import "./style.less";
var Flex = styled.div.withConfig({
  displayName: "Flex",
  componentId: "form-editor-42cd__sc-1iov5q4-0"
})(function (_ref) {
  var vertical = _ref.vertical;
  if (!vertical) return '';
  return "\n      flex-direction: column;\n      align-items: start;\n      .".concat(prefixCls('setting-item-label'), " {\n        margin-bottom: 8px;\n      }\n    ");
});
export var SettingItem = function SettingItem(_ref2) {
  var label = _ref2.label,
    children = _ref2.children,
    style = _ref2.style,
    tips = _ref2.tips,
    vertical = _ref2.vertical;
  return /*#__PURE__*/React.createElement(Flex, {
    style: style || {},
    className: prefixCls('setting-item'),
    vertical: vertical
  }, /*#__PURE__*/React.createElement("div", {
    className: prefixCls('setting-item-label')
  }, label, "\xA0", tips && /*#__PURE__*/React.createElement(Popover, {
    content: tips
  }, /*#__PURE__*/React.createElement(QuestionCircleOutlined, {
    style: {
      cursor: 'pointer'
    }
  }))), /*#__PURE__*/React.createElement("div", {
    className: prefixCls('setting-item-value')
  }, children));
};
export var SettingWrap = function SettingWrap(_ref3) {
  var title = _ref3.title,
    children = _ref3.children,
    style = _ref3.style;
  return /*#__PURE__*/React.createElement("div", {
    style: style || {},
    className: prefixCls('setting-wrap')
  }, /*#__PURE__*/React.createElement("div", {
    className: prefixCls('setting-title')
  }, title), children);
};
