// @roddan/form-editor/lib/elements/input/input-setting.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SettingInput = void 0;
var _react = _interopRequireDefault(require("react"));
var _antd = require("antd");
var _mobxReactLite = require("mobx-react-lite");
var _components = require("../../components");
var _store = _interopRequireDefault(require("../../store"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var typeOptions = [{
  label: '单行文本',
  value: 'single'
}, {
  label: '多行文本',
  value: 'multiple'
}, {
  label: '密码框',
  value: 'password'
}];
var SettingInputContent = function SettingInputContent() {
  var _store$selectedElemen = _store.default.selectedElement,
    textType = _store$selectedElemen.textType,
    minRows = _store$selectedElemen.minRows,
    maxRows = _store$selectedElemen.maxRows,
    autoSize = _store$selectedElemen.autoSize;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_components.SettingWrap, {
    title: "\u5143\u7D20\u8BBE\u7F6E"
  }, /*#__PURE__*/_react.default.createElement(_components.PlaceholderSetting, null), /*#__PURE__*/_react.default.createElement(_components.SettingItem, {
    label: "\u6587\u672C\u7C7B\u578B"
  }, /*#__PURE__*/_react.default.createElement(_antd.Select, {
    value: textType,
    style: {
      width: '100%'
    },
    options: typeOptions,
    onChange: function onChange(val) {
      _store.default.setSelectedProp('textType', val);
    }
  })), textType === 'multiple' && /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_components.SettingItem, {
    label: "\u81EA\u9002\u5E94\u884C\u6570"
  }, /*#__PURE__*/_react.default.createElement(_antd.Switch, {
    checked: autoSize,
    onChange: function onChange(val) {
      _store.default.setSelectedProp('autoSize', val);
    }
  })), !autoSize && /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_components.SettingItem, {
    label: "\u6700\u5C0F\u884C\u6570"
  }, /*#__PURE__*/_react.default.createElement(_antd.InputNumber, {
    value: minRows,
    style: {
      width: '100%'
    },
    onChange: function onChange(val) {
      _store.default.setSelectedProp('minRows', Number(val));
    }
  })), /*#__PURE__*/_react.default.createElement(_components.SettingItem, {
    label: "\u6700\u5927\u884C\u6570"
  }, /*#__PURE__*/_react.default.createElement(_antd.InputNumber, {
    value: maxRows,
    min: minRows,
    style: {
      width: '100%'
    },
    onChange: function onChange(val) {
      _store.default.setSelectedProp('maxRows', Number(val));
    }
  }))))), /*#__PURE__*/_react.default.createElement(_components.RegPattern, null));
};
var SettingInput = exports.SettingInput = (0, _mobxReactLite.observer)(SettingInputContent);
