// @roddan/form-editor/lib/elements/input/render-input.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RenderInput = void 0;
var _react = _interopRequireDefault(require("react"));
var _antd = require("antd");
var _mobxReactLite = require("mobx-react-lite");
var _components = require("../../components");
var _hooks = require("../../hooks");
var _store = _interopRequireDefault(require("../../store"));
var _types = require("../../types");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var RenderInputContent = function RenderInputContent(_ref) {
  var fieldValue = _ref.fieldValue,
    _ref$element = _ref.element,
    element = _ref$element === void 0 ? {} : _ref$element;
  var textType = element.textType,
    minRows = element.minRows,
    maxRows = element.maxRows,
    id = element.id,
    autoSize = element.autoSize,
    placeholder = element.placeholder;
  var _useRegisterEvents = (0, _hooks.useRegisterEvents)(element),
    eventFunctions = _useRegisterEvents.eventFunctions;
  var handleEvent = function handleEvent(action) {
    return function (e) {
      var _eventFunctions$actio;
      (_eventFunctions$actio = eventFunctions[action]) === null || _eventFunctions$actio === void 0 || _eventFunctions$actio.call(eventFunctions, e.target.value);
    };
  };
  var handleChange = function handleChange(e) {
    _store.default.setFieldValue(id, e.target.value);
  };
  (0, _hooks.useFormUpdate)(function () {
    var _eventFunctions$EEven;
    (_eventFunctions$EEven = eventFunctions[_types.EEventAction.ON_LOADED]) === null || _eventFunctions$EEven === void 0 || _eventFunctions$EEven.call(eventFunctions);
  }, [eventFunctions[_types.EEventAction.ON_LOADED]]);
  (0, _hooks.useFormUpdate)(function () {
    var _eventFunctions$EEven2;
    (_eventFunctions$EEven2 = eventFunctions[_types.EEventAction.VALUE_CHANGE]) === null || _eventFunctions$EEven2 === void 0 || _eventFunctions$EEven2.call(eventFunctions, fieldValue);
  }, [fieldValue]);
  return /*#__PURE__*/_react.default.createElement(_components.ElementLayout, {
    element: element
  }, textType === 'multiple' ? /*#__PURE__*/_react.default.createElement(_antd.Input.TextArea, {
    autoSize: autoSize ? true : {
      minRows: minRows,
      maxRows: maxRows
    },
    value: fieldValue,
    placeholder: placeholder,
    id: id,
    onChange: handleChange,
    onFocus: handleEvent(_types.EEventAction.ON_FOCUS),
    onBlur: handleEvent(_types.EEventAction.ON_BLUR)
  }) : /*#__PURE__*/_react.default.createElement(_antd.Input, {
    placeholder: placeholder,
    id: id,
    onChange: handleChange,
    onFocus: handleEvent(_types.EEventAction.ON_FOCUS),
    onBlur: handleEvent(_types.EEventAction.ON_BLUR),
    value: fieldValue,
    type: textType === 'single' ? 'text' : 'password',
    autoComplete: "new-password"
  }));
};
var RenderInput = exports.RenderInput = (0, _mobxReactLite.observer)(RenderInputContent);
