// @roddan/form-editor/es/hooks/use-update.js
import { useEffect, useRef } from 'react';
import { useEditorContext } from "../context";
export var useUpdate = function useUpdate(effect, deps) {
  var firstRender = useRef(true);
  useEffect(firstRender.current ? function () {
    firstRender.current = false;
  } : effect, deps);
};
export var useFormEffect = function useFormEffect(effect) {
  var deps = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : [];
  var _useEditorContext = useEditorContext(),
    mode = _useEditorContext.mode;
  useEffect(mode === 'design' ? function () {} : effect, deps);
};
export var useFormUpdate = function useFormUpdate(effect, deps) {
  var _useEditorContext2 = useEditorContext(),
    mode = _useEditorContext2.mode;
  var firstRender = useRef(true);
  useEffect(firstRender.current || mode === 'design' ? function () {
    firstRender.current = false;
  } : effect, deps);
};
export var useDesignEffect = function useDesignEffect(effect) {
  var deps = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : [];
  var _useEditorContext3 = useEditorContext(),
    mode = _useEditorContext3.mode;
  useEffect(mode !== 'design' ? function () {} : effect, deps);
};
export var useDesignUpdate = function useDesignUpdate(effect, deps) {
  var _useEditorContext4 = useEditorContext(),
    mode = _useEditorContext4.mode;
  var firstRender = useRef(true);
  useEffect(firstRender.current || mode !== 'design' ? function () {
    firstRender.current = false;
  } : effect, deps);
};
