// @roddan/form-editor/lib/elements/table/render-table.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RenderTable = void 0;
var _react = _interopRequireDefault(require("react"));
var _antd = require("antd");
var _mobxReactLite = require("mobx-react-lite");
var _components = require("../../components");
var _types = require("../../types");
var _utils = require("../../utils");
var _hooks = require("../../hooks");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
var RenderTableContent = function RenderTableContent(_ref) {
  var _ref$fieldValue = _ref.fieldValue,
    fieldValue = _ref$fieldValue === void 0 ? [] : _ref$fieldValue,
    element = _ref.element;
  var _element$tableColumns = element.tableColumns,
    tableColumns = _element$tableColumns === void 0 ? '[]' : _element$tableColumns,
    _element$tableAttribu = element.tableAttributes,
    tableAttributes = _element$tableAttribu === void 0 ? '{}' : _element$tableAttribu;
  var _useRegisterEvents = (0, _hooks.useRegisterEvents)(element),
    eventFunctions = _useRegisterEvents.eventFunctions;
  (0, _hooks.useFormUpdate)(function () {
    var _eventFunctions$EEven;
    (_eventFunctions$EEven = eventFunctions[_types.EEventAction.ON_LOADED]) === null || _eventFunctions$EEven === void 0 || _eventFunctions$EEven.call(eventFunctions);
  }, [eventFunctions[_types.EEventAction.ON_LOADED]]);
  return /*#__PURE__*/_react.default.createElement(_components.ElementLayout, {
    element: element
  }, /*#__PURE__*/_react.default.createElement(_antd.Table, _extends({
    columns: (0, _utils.parseJSX)(tableColumns),
    rowKey: "id",
    dataSource: fieldValue
  }, (0, _utils.parseJSX)("[".concat(tableAttributes, "]"))[0])));
};
var RenderTable = exports.RenderTable = (0, _mobxReactLite.observer)(RenderTableContent);
