// @roddan/form-editor/lib/elements/table/table-setting.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SettingTable = void 0;
var _icons = require("@ant-design/icons");
var _antd = require("antd");
var _mobxReactLite = require("mobx-react-lite");
var _react = _interopRequireDefault(require("react"));
var _components = require("../../components");
var _store = _interopRequireDefault(require("../../store"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var SettingTableContent = function SettingTableContent() {
  return /*#__PURE__*/_react.default.createElement(_components.SettingWrap, {
    title: "\u5143\u7D20\u8BBE\u7F6E"
  }, /*#__PURE__*/_react.default.createElement(_components.SettingItem, {
    tips: /*#__PURE__*/_react.default.createElement("a", {
      target: "_blank",
      href: "https://ant.design/components/table-cn#table",
      rel: "noreferrer"
    }, "antd - table\u6587\u6863"),
    label: "\u8868\u683C\u914D\u7F6E"
  }, /*#__PURE__*/_react.default.createElement(_components.AttributesSetting, {
    editorType: "json",
    value: _store.default.selectedElement.tableAttributes,
    title: "\u8868\u683C\u914D\u7F6E",
    onChange: function onChange(val) {
      _store.default.setSelectedProp('tableAttributes', val);
    }
  }, /*#__PURE__*/_react.default.createElement(_antd.Button, {
    className: "fe-attr-setting-btn",
    size: "small"
  }, "\u7F16\u8F91"))), /*#__PURE__*/_react.default.createElement(_components.SettingItem, {
    tips: /*#__PURE__*/_react.default.createElement("a", {
      target: "_blank",
      href: "https://ant.design/components/table-cn#column",
      rel: "noreferrer"
    }, "table - column\u6587\u6863"),
    label: "\u5217\u8868\u914D\u7F6E"
  }, /*#__PURE__*/_react.default.createElement(_components.AttributesSetting, {
    editorType: "javascript",
    value: _store.default.selectedElement.tableColumns,
    title: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, "\u5217\u8868\u9879\u8BBE\u7F6E\xA0", /*#__PURE__*/_react.default.createElement(_antd.Popover, {
      content: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, "render\u51FD\u6570\u4E2D\u53EF\u76F4\u63A5\u4F7F\u7528antd\u7EC4\u4EF6\u53CAAntdIcons.XXX\u7684\u65B9\u5F0F\u4F7F\u7528icon\u56FE\u6807")
    }, /*#__PURE__*/_react.default.createElement(_icons.QuestionCircleOutlined, {
      style: {
        cursor: 'pointer'
      }
    }))),
    onChange: function onChange(val) {
      _store.default.setSelectedProp('tableColumns', val);
    }
  }, /*#__PURE__*/_react.default.createElement(_antd.Button, {
    className: "fe-attr-setting-btn",
    size: "small"
  }, "\u7F16\u8F91"))));
};
var SettingTable = exports.SettingTable = (0, _mobxReactLite.observer)(SettingTableContent);
