// @roddan/form-editor/es/elements/time/render-time.js
import React from 'react';
import { TimePicker } from 'antd';
import dayjs from 'dayjs';
import { observer } from 'mobx-react-lite';
import { useRegisterEvents, useFormUpdate } from "../../hooks";
import { formatDate } from "../../utils";
import { EEventAction } from "../../types";
import { ElementLayout } from "../../components";
import store from "../../store";
var RenderTimeContent = function RenderTimeContent(_ref) {
  var fieldValue = _ref.fieldValue,
    element = _ref.element;
  var id = element.id,
    dateFormat = element.dateFormat,
    placeholder = element.placeholder;
  var _useRegisterEvents = useRegisterEvents(element),
    eventFunctions = _useRegisterEvents.eventFunctions;
  var handleEvent = function handleEvent(action) {
    return function (e) {
      var _eventFunctions$actio;
      (_eventFunctions$actio = eventFunctions[action]) === null || _eventFunctions$actio === void 0 || _eventFunctions$actio.call(eventFunctions, e.target.value);
    };
  };
  var handleChange = function handleChange(date) {
    store.setFieldValue(id, date ? formatDate(date, dateFormat) : undefined);
  };
  useFormUpdate(function () {
    var _eventFunctions$EEven;
    (_eventFunctions$EEven = eventFunctions[EEventAction.VALUE_CHANGE]) === null || _eventFunctions$EEven === void 0 || _eventFunctions$EEven.call(eventFunctions, fieldValue);
  }, [fieldValue]);
  return /*#__PURE__*/React.createElement(ElementLayout, {
    element: element
  }, /*#__PURE__*/React.createElement(TimePicker, {
    format: dateFormat,
    value: fieldValue ? dayjs("2000-01-01 ".concat(fieldValue)) : undefined,
    getPopupContainer: function getPopupContainer(n) {
      return n.parentElement;
    },
    onChange: handleChange,
    onFocus: handleEvent(EEventAction.ON_FOCUS),
    onBlur: handleEvent(EEventAction.ON_BLUR),
    placeholder: placeholder
  }));
};
export var RenderTime = observer(RenderTimeContent);
