// @roddan/form-editor/es/elements/time/time-setting.js
import React from 'react';
import { Select } from 'antd';
import { observer } from 'mobx-react-lite';
import { SettingItem, SettingWrap, PlaceholderSetting, RegPattern } from "../../components";
import store from "../../store";
var dateOptions = ['HH:mm:ss', 'HH:mm'].map(function (per) {
  return {
    label: per,
    value: per
  };
});
var SettingTimeContent = function SettingTimeContent() {
  var dateFormat = store.selectedElement.dateFormat;
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(SettingWrap, {
    title: "\u5143\u7D20\u8BBE\u7F6E"
  }, /*#__PURE__*/React.createElement(PlaceholderSetting, null), /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u65F6\u95F4\u683C\u5F0F"
  }, /*#__PURE__*/React.createElement(Select, {
    value: dateFormat,
    style: {
      width: '100%'
    },
    options: dateOptions,
    onChange: function onChange(val) {
      // console.log(e)
      store.setSelectedProp('dateFormat', val);
    }
  }))), /*#__PURE__*/React.createElement(RegPattern, null));
};
export var SettingTime = observer(SettingTimeContent);
