// @roddan/form-editor/lib/elements/date/render-date.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RenderDate = void 0;
var _react = _interopRequireWildcard(require("react"));
var _antd = require("antd");
var _dayjs = _interopRequireDefault(require("dayjs"));
var _mobxReactLite = require("mobx-react-lite");
var _hooks = require("../../hooks");
var _utils = require("../../utils");
var _types = require("../../types");
var _components = require("../../components");
var _store = _interopRequireDefault(require("../../store"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
var RenderDateContent = function RenderDateContent(_ref) {
  var fieldValue = _ref.fieldValue,
    element = _ref.element;
  var id = element.id,
    dateFormat = element.dateFormat,
    placeholder = element.placeholder;
  var _useRegisterEvents = (0, _hooks.useRegisterEvents)(element),
    eventFunctions = _useRegisterEvents.eventFunctions;
  var handleEvent = function handleEvent(action) {
    return function (e) {
      var _eventFunctions$actio;
      (_eventFunctions$actio = eventFunctions[action]) === null || _eventFunctions$actio === void 0 || _eventFunctions$actio.call(eventFunctions, e.target.value);
    };
  };
  var handleChange = function handleChange(date) {
    _store.default.setFieldValue(id, date ? (0, _utils.formatDate)(date, dateFormat) : undefined);
  };
  (0, _hooks.useFormUpdate)(function () {
    var _eventFunctions$EEven;
    (_eventFunctions$EEven = eventFunctions[_types.EEventAction.ON_LOADED]) === null || _eventFunctions$EEven === void 0 || _eventFunctions$EEven.call(eventFunctions);
  }, [eventFunctions[_types.EEventAction.ON_LOADED]]);
  (0, _hooks.useFormUpdate)(function () {
    var _eventFunctions$EEven2;
    (_eventFunctions$EEven2 = eventFunctions[_types.EEventAction.VALUE_CHANGE]) === null || _eventFunctions$EEven2 === void 0 || _eventFunctions$EEven2.call(eventFunctions, fieldValue);
  }, [fieldValue]);
  var showTime = (0, _react.useMemo)(function () {
    var timeFormat = dateFormat === null || dateFormat === void 0 ? void 0 : dateFormat.split(' ');
    if (!timeFormat) return false;
    var res = {};
    if (timeFormat.includes('HH')) res.showHour = true;
    if (timeFormat.includes('mm')) res.showMinute = true;
    if (timeFormat.includes('ss')) res.showSecond = true;
    return res;
  }, [dateFormat]);
  return /*#__PURE__*/_react.default.createElement(_components.ElementLayout, {
    element: element
  }, /*#__PURE__*/_react.default.createElement(_antd.DatePicker, {
    format: dateFormat
    // @ts-ignore
    ,
    value: fieldValue ? (0, _dayjs.default)(fieldValue) : undefined,
    showTime: showTime,
    getPopupContainer: function getPopupContainer(n) {
      return n.parentElement;
    },
    placement: "bottomRight",
    onChange: handleChange,
    onFocus: handleEvent(_types.EEventAction.ON_FOCUS),
    onBlur: handleEvent(_types.EEventAction.ON_BLUR),
    placeholder: placeholder
  }));
};

// store.setAllElementsList(ELEMENT_INPUT, 'render', RenderInput)
var RenderDate = exports.RenderDate = (0, _mobxReactLite.observer)(RenderDateContent);
