// @roddan/form-editor/es/elements/image/image-setting.js
import React from 'react';
import { Input, Switch } from 'antd';
import { observer } from 'mobx-react-lite';
import { SettingWrap, SettingItem } from "../../components";
import store from "../../store";
var SettingImageContent = function SettingImageContent() {
  var _store$selectedElemen = store.selectedElement,
    defaultImgSrc = _store$selectedElemen.defaultImgSrc,
    placeholder = _store$selectedElemen.placeholder,
    preview = _store$selectedElemen.preview,
    previewSrc = _store$selectedElemen.previewSrc;
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(SettingWrap, {
    title: "\u5143\u7D20\u8BBE\u7F6E"
  }, /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u9ED8\u8BA4\u5730\u5740"
  }, /*#__PURE__*/React.createElement(Input, {
    value: defaultImgSrc,
    onChange: function onChange(e) {
      store.setSelectedProp('defaultImgSrc', e.target.value);
    }
  })), /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u5360\u4F4D\u5730\u5740"
  }, /*#__PURE__*/React.createElement(Input, {
    value: placeholder,
    onChange: function onChange(e) {
      store.setSelectedProp('placeholder', e.target.value);
    }
  })), /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u652F\u6301\u9884\u89C8"
  }, /*#__PURE__*/React.createElement(Switch, {
    checked: preview,
    size: "small",
    onChange: function onChange(val) {
      return store.setSelectedProp('preview', val);
    }
  })), preview && /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u9884\u89C8\u5730\u5740"
  }, /*#__PURE__*/React.createElement(Input, {
    value: previewSrc,
    onChange: function onChange(e) {
      store.setSelectedProp('previewSrc', e.target.value);
    }
  }))));
};
export var SettingImage = observer(SettingImageContent);
