// @roddan/form-editor/es/elements/image/render-image.js
import React from 'react';
import { observer } from 'mobx-react-lite';
import { Image } from 'antd';
import { useRegisterEvents, useFormUpdate } from "../../hooks";
import { ElementLayout } from "../../components";
import { EEventAction } from "../../types";
var RenderImageContent = function RenderImageContent(_ref) {
  var element = _ref.element,
    fieldValue = _ref.fieldValue;
  var placeholder = element.placeholder,
    preview = element.preview,
    defaultImgSrc = element.defaultImgSrc,
    previewSrc = element.previewSrc;
  var _useRegisterEvents = useRegisterEvents(element),
    eventFunctions = _useRegisterEvents.eventFunctions;
  useFormUpdate(function () {
    var _eventFunctions$EEven;
    (_eventFunctions$EEven = eventFunctions[EEventAction.ON_LOADED]) === null || _eventFunctions$EEven === void 0 || _eventFunctions$EEven.call(eventFunctions);
  }, [eventFunctions[EEventAction.ON_LOADED]]);
  return /*#__PURE__*/React.createElement(ElementLayout, {
    element: element
  }, /*#__PURE__*/React.createElement(Image, {
    src: fieldValue || defaultImgSrc,
    fallback: placeholder,
    preview: preview ? previewSrc ? {
      src: previewSrc
    } : true : false
  }));
};
export var RenderImage = observer(RenderImageContent);
