// @roddan/form-editor/lib/elements/number/const.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.valueTypeOptions = exports.initialData = exports.eventActions = exports.NUMBER_TEXT = exports.EValueType = exports.ELEMENT_NUMBER = void 0;
var _types = require("../../types");
var EValueType = exports.EValueType = /*#__PURE__*/function (EValueType) {
  EValueType["INT"] = "int";
  EValueType["ONE_DECIMAL"] = "one_decimal";
  EValueType["TWO_DECIMAL"] = "two_decimal";
  return EValueType;
}({});
var valueTypeOptions = exports.valueTypeOptions = [{
  label: '整数',
  value: EValueType.INT
}, {
  label: '保留一位小数',
  value: EValueType.ONE_DECIMAL
}, {
  label: '保留两位小数',
  value: EValueType.TWO_DECIMAL
}];
var ELEMENT_NUMBER = exports.ELEMENT_NUMBER = 'number';
var NUMBER_TEXT = exports.NUMBER_TEXT = '数字框';
var eventActions = exports.eventActions = [_types.EEventAction.ON_LOADED, _types.EEventAction.VALUE_CHANGE, _types.EEventAction.ON_FOCUS, _types.EEventAction.ON_BLUR];
var initialData = exports.initialData = {
  elementName: '数字框',
  gridSpan: 4,
  valueType: 'int'
};
