// @roddan/form-editor/lib/components/option-setting/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.OptionSetting = void 0;
var _react = _interopRequireDefault(require("react"));
var _antd = require("antd");
var _mobxReactLite = require("mobx-react-lite");
var _const = require("../../const");
var _optionModal = _interopRequireDefault(require("./option-modal"));
var _ = require("./..");
var _store = _interopRequireDefault(require("../../store"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var columns = [{
  title: '属性',
  dataIndex: 'label'
}, {
  title: '值',
  dataIndex: 'value'
}];
var OptionSetting = exports.OptionSetting = (0, _mobxReactLite.observer)(function () {
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_.SettingItem, {
    label: "\u9009\u9879"
  }, /*#__PURE__*/_react.default.createElement(_optionModal.default, null, /*#__PURE__*/_react.default.createElement(_antd.Button, {
    className: "".concat((0, _const.prefixCls)('attr-setting-btn')),
    size: "small"
  }, "\u7F16\u8F91"))), /*#__PURE__*/_react.default.createElement(_antd.Table, {
    columns: columns,
    dataSource: _store.default.selectedElement.valueOptions,
    pagination: false,
    scroll: {
      y: 300
    },
    rowKey: "id"
  }));
});
