// @roddan/ui/lib/monaco-editor/Editor/index.js
"use client";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __reExport = (target, mod, secondTarget) => (__copyProps(target, mod, "default"), secondTarget && __copyProps(secondTarget, mod, "default"));
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/monaco-editor/Editor/index.tsx
var Editor_exports = {};
__export(Editor_exports, {
  MonacoEditor: () => MonacoEditor
});
module.exports = __toCommonJS(Editor_exports);
var import_loader = __toESM(require("@monaco-editor/loader"));
var apiMonaco = __toESM(require("monaco-editor/esm/vs/editor/editor.api"));
var import_react = __toESM(require("react"));
var import_useMount = __toESM(require("../hooks/useMount"));
var import_usePrevious = __toESM(require("../hooks/usePrevious"));
var import_useUpdate = __toESM(require("../hooks/useUpdate"));
var import_MonacoContainer = __toESM(require("../MonacoContainer"));
var import_utils = require("../utils");
__reExport(Editor_exports, require("./types"), module.exports);
import_loader.default.config({
  paths: {
    vs: "https://cdn.jsdelivr.net/gh/lawxy/workers@2.0/vs"
  }
});
var viewStates = /* @__PURE__ */ new Map();
var MonacoEditor = (0, import_react.memo)(
  ({
    defaultValue,
    defaultLanguage,
    defaultPath,
    value,
    language,
    path,
    theme = "light",
    line,
    loading = "Loading...",
    options = {},
    overrideServices = {},
    saveViewState = true,
    keepCurrentModel = false,
    style = {},
    className,
    wrapperProps = {},
    beforeMount = import_utils.noop,
    onMount = import_utils.noop,
    onChange,
    onValidate = import_utils.noop,
    onFocus = import_utils.noop,
    onBlur = import_utils.noop
  }) => {
    const [isEditorReady, setIsEditorReady] = (0, import_react.useState)(false);
    const [isMonacoMounting, setIsMonacoMounting] = (0, import_react.useState)(true);
    const monacoRef = (0, import_react.useRef)(null);
    const editorRef = (0, import_react.useRef)(null);
    const containerRef = (0, import_react.useRef)(null);
    const onMountRef = (0, import_react.useRef)(onMount);
    const beforeMountRef = (0, import_react.useRef)(beforeMount);
    const subscriptionRef = (0, import_react.useRef)();
    const valueRef = (0, import_react.useRef)(value);
    const previousPath = (0, import_usePrevious.default)(path);
    const preventCreation = (0, import_react.useRef)(false);
    const preventTriggerChangeEvent = (0, import_react.useRef)(false);
    function disposeEditor() {
      var _a, _b;
      (_a = subscriptionRef.current) == null ? void 0 : _a.dispose();
      if (keepCurrentModel) {
        saveViewState && viewStates.set(path, editorRef.current.saveViewState());
      } else {
        (_b = editorRef.current.getModel()) == null ? void 0 : _b.dispose();
      }
      editorRef.current.dispose();
    }
    (0, import_useMount.default)(() => {
      const cancelable = import_loader.default.init();
      cancelable.then(
        (monaco) => (monacoRef.current = monaco) && setIsMonacoMounting(false)
      ).catch((error) => {
        monacoRef.current = apiMonaco;
        setIsMonacoMounting(false);
      });
      return () => editorRef.current ? disposeEditor() : cancelable.cancel();
    });
    (0, import_useUpdate.default)(
      () => {
        var _a, _b, _c, _d;
        const model = (0, import_utils.getOrCreateModel)(
          monacoRef.current,
          defaultValue || value || "",
          defaultLanguage || language || "",
          path || defaultPath || ""
        );
        if (model !== ((_a = editorRef.current) == null ? void 0 : _a.getModel())) {
          if (saveViewState)
            viewStates.set(previousPath, (_b = editorRef.current) == null ? void 0 : _b.saveViewState());
          (_c = editorRef.current) == null ? void 0 : _c.setModel(model);
          if (saveViewState)
            (_d = editorRef.current) == null ? void 0 : _d.restoreViewState(viewStates.get(path));
        }
      },
      [path],
      isEditorReady
    );
    (0, import_useUpdate.default)(
      () => {
        var _a;
        (_a = editorRef.current) == null ? void 0 : _a.updateOptions(options);
      },
      [options],
      isEditorReady
    );
    (0, import_useUpdate.default)(
      () => {
        if (!editorRef.current || value === void 0)
          return;
        if (editorRef.current.getOption(
          monacoRef.current.editor.EditorOption.readOnly
        )) {
          editorRef.current.setValue(value);
        } else if (value !== editorRef.current.getValue()) {
          preventTriggerChangeEvent.current = true;
          editorRef.current.executeEdits("", [
            {
              range: editorRef.current.getModel().getFullModelRange(),
              text: value,
              forceMoveMarkers: true
            }
          ]);
          editorRef.current.pushUndoStop();
          preventTriggerChangeEvent.current = false;
        }
      },
      [value],
      isEditorReady
    );
    (0, import_useUpdate.default)(
      () => {
        var _a, _b;
        const model = (_a = editorRef.current) == null ? void 0 : _a.getModel();
        if (model && language)
          (_b = monacoRef.current) == null ? void 0 : _b.editor.setModelLanguage(model, language);
      },
      [language],
      isEditorReady
    );
    (0, import_useUpdate.default)(
      () => {
        var _a;
        if (line !== void 0) {
          (_a = editorRef.current) == null ? void 0 : _a.revealLine(line);
        }
      },
      [line],
      isEditorReady
    );
    (0, import_useUpdate.default)(
      () => {
        var _a;
        (_a = monacoRef.current) == null ? void 0 : _a.editor.setTheme(theme);
      },
      [theme],
      isEditorReady
    );
    const createEditor = (0, import_react.useCallback)(() => {
      var _a;
      if (!containerRef.current || !monacoRef.current)
        return;
      if (!preventCreation.current) {
        beforeMountRef.current(monacoRef.current);
        const autoCreatedModelPath = path || defaultPath;
        const defaultModel = (0, import_utils.getOrCreateModel)(
          monacoRef.current,
          value || defaultValue || "",
          defaultLanguage || language || "",
          autoCreatedModelPath || ""
        );
        editorRef.current = (_a = monacoRef.current) == null ? void 0 : _a.editor.create(
          containerRef.current,
          {
            model: defaultModel,
            automaticLayout: true,
            ...options
          },
          overrideServices
        );
        saveViewState && editorRef.current.restoreViewState(
          viewStates.get(autoCreatedModelPath)
        );
        monacoRef.current.editor.setTheme(theme);
        if (line !== void 0) {
          editorRef.current.revealLine(line);
        }
        setIsEditorReady(true);
        preventCreation.current = true;
      }
    }, [
      defaultValue,
      defaultLanguage,
      defaultPath,
      value,
      language,
      path,
      options,
      overrideServices,
      saveViewState,
      theme,
      line
    ]);
    (0, import_react.useEffect)(() => {
      if (isEditorReady) {
        onMountRef.current(editorRef.current, monacoRef.current);
      }
    }, [isEditorReady]);
    (0, import_react.useEffect)(() => {
      !isMonacoMounting && !isEditorReady && createEditor();
    }, [isMonacoMounting, isEditorReady, createEditor]);
    valueRef.current = value;
    (0, import_react.useEffect)(() => {
      var _a, _b;
      if (isEditorReady && onChange) {
        (_a = subscriptionRef.current) == null ? void 0 : _a.dispose();
        subscriptionRef.current = (_b = editorRef.current) == null ? void 0 : _b.onDidChangeModelContent(
          (event) => {
            if (!preventTriggerChangeEvent.current) {
              onChange(editorRef.current.getValue(), event);
            }
          }
        );
      }
    }, [isEditorReady, onChange]);
    (0, import_react.useEffect)(() => {
      if (isEditorReady) {
        const changeMarkersListener = monacoRef.current.editor.onDidChangeMarkers((uris) => {
          var _a;
          const editorUri = (_a = editorRef.current.getModel()) == null ? void 0 : _a.uri;
          if (editorUri) {
            const currentEditorHasMarkerChanges = uris.find(
              (uri) => uri.path === editorUri.path
            );
            if (currentEditorHasMarkerChanges) {
              const markers = monacoRef.current.editor.getModelMarkers({
                resource: editorUri
              });
              onValidate == null ? void 0 : onValidate(markers);
            }
          }
        });
        return () => {
          changeMarkersListener == null ? void 0 : changeMarkersListener.dispose();
        };
      }
      return () => {
      };
    }, [isEditorReady, onValidate]);
    (0, import_react.useEffect)(() => {
      var _a, _b;
      if (!isEditorReady || !editorRef.current)
        return;
      (_a = editorRef.current) == null ? void 0 : _a.onDidFocusEditorText(() => {
        onFocus == null ? void 0 : onFocus();
      });
      (_b = editorRef.current) == null ? void 0 : _b.onDidBlurEditorText(() => {
        onBlur == null ? void 0 : onBlur();
      });
    }, [isEditorReady]);
    return /* @__PURE__ */ import_react.default.createElement(
      import_MonacoContainer.default,
      {
        style,
        isEditorReady,
        loading,
        _ref: containerRef,
        className,
        wrapperProps
      }
    );
  }
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  MonacoEditor,
  ...require("./types")
});
