// @roddan/form-editor/es/components/option-setting/option-modal.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import React, { useState, useEffect, useRef } from 'react';
import { Modal, Space, Input, message, Popover } from 'antd';
import { QuestionCircleOutlined } from '@ant-design/icons';
import { MinusIcon, PlusIcon } from "./..";
import { cloneDeep } from 'lodash-es';
import { observer } from 'mobx-react-lite';
import { TableSortable } from '@roddan/ui';
import { BatchGenerateOptions } from "../batch-generate-options";
import store from "../../store";
import { idCreator } from "../../utils";
var OptionModal = function OptionModal(_ref) {
  var children = _ref.children;
  var _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    open = _useState2[0],
    setOpen = _useState2[1];
  var _useState3 = useState([]),
    _useState4 = _slicedToArray(_useState3, 2),
    valueOptions = _useState4[0],
    setOption = _useState4[1];
  var valueOptionsRef = useRef();
  valueOptionsRef.current = valueOptions;
  useEffect(function () {
    var _store$selectedElemen;
    setOption(((_store$selectedElemen = store.selectedElement) === null || _store$selectedElemen === void 0 ? void 0 : _store$selectedElemen.valueOptions) || []);
  }, [store.selectedElement.valueOptions]);
  var handleInputChange = function handleInputChange(idx, field, value) {
    var newOptions = cloneDeep(valueOptions);
    newOptions[idx][field] = value;
    setOption(newOptions);
  };
  var judgeOptionsInvalid = function judgeOptionsInvalid() {
    return valueOptions.some(function (item) {
      return !item.label || !item.value;
    });
  };
  var addOption = function addOption() {
    if (judgeOptionsInvalid()) {
      message.error('属性或值不为空, 补充完成再新增或保存');
      return;
    }
    var newOptions = cloneDeep(valueOptions);
    newOptions.push({
      label: '',
      value: '',
      id: idCreator('option')
    });
    setOption(newOptions);
  };
  var columns = [{
    title: '选项名',
    dataIndex: 'label',
    render: function render(val, _, idx) {
      return /*#__PURE__*/React.createElement(Input, {
        value: val,
        onChange: function onChange(e) {
          handleInputChange(idx, 'label', e.target.value);
        }
      });
    }
  }, {
    title: '值',
    dataIndex: 'value',
    render: function render(val, _, idx) {
      return /*#__PURE__*/React.createElement(Input, {
        value: val,
        onChange: function onChange(e) {
          handleInputChange(idx, 'value', e.target.value);
        }
      });
    }
  }, {
    title: '操作',
    width: 60,
    render: function render(_, __, idx) {
      return /*#__PURE__*/React.createElement(Space, null, /*#__PURE__*/React.createElement("span", {
        onClick: function onClick() {
          var newOptions = cloneDeep(valueOptions);
          newOptions.splice(idx, 1);
          setOption(newOptions);
          if (!newOptions.length) {
            setOption([{
              label: '',
              value: '',
              id: idCreator('option')
            }]);
          }
        }
      }, /*#__PURE__*/React.createElement(MinusIcon, null)), idx === valueOptions.length - 1 && /*#__PURE__*/React.createElement("span", {
        onClick: addOption
      }, /*#__PURE__*/React.createElement(PlusIcon, null)));
    }
  }];
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.isValidElement(children) && /*#__PURE__*/React.cloneElement(children, {
    onClick: function onClick() {
      return setOpen(true);
    }
  }), /*#__PURE__*/React.createElement(Modal, {
    open: open,
    title: /*#__PURE__*/React.createElement(BatchGenerateOptions, {
      title: /*#__PURE__*/React.createElement(React.Fragment, null, "\u5C5E\u6027\u8BBE\u7F6E\xA0", /*#__PURE__*/React.createElement(Popover, {
        content: "\u683C\u5F0F: \u5C5E\u6027\u540D: \u5C5E\u6027\u503C, \u591A\u5B57\u6BB5\u6362\u884C\u5206\u9694"
      }, /*#__PURE__*/React.createElement(QuestionCircleOutlined, {
        style: {
          cursor: 'pointer'
        }
      }))),
      options: valueOptions,
      setOptions: setOption,
      labelField: "label",
      valueField: "value"
    }),
    maskClosable: false,
    onCancel: function onCancel() {
      setOpen(false);
    },
    onOk: function onOk() {
      if (judgeOptionsInvalid()) {
        message.error('属性或值不为空, 补充完成再新增或保存');
        return;
      }
      store.setSelectedProp('valueOptions', valueOptions);
      setOpen(false);
    }
  }, /*#__PURE__*/React.createElement(TableSortable, {
    onSort: function onSort(newDatas) {
      return setOption(newDatas);
    },
    columns: columns,
    rowKey: "id",
    dataSource: valueOptions,
    pagination: false,
    scroll: {
      y: 300
    }
  })));
};
export default observer(OptionModal);
