// @roddan/form-editor/es/components/event-modal/event-modal.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import React, { useEffect, useMemo, useState, useRef } from 'react';
import { message, Modal } from 'antd';
import { ActionConfig } from "./action-config";
import { EventModalContext } from "./context";
import { SelectComponent } from "./select-component";
import { prefixCls } from "../../const";
import { EChangeType, EEventAction, eventActionInChinese, eventTypeChinese, EEventType } from "../../types";
import { idCreator } from "../../utils";
import "./style.less";
var getNewEvent = function getNewEvent() {
  var newProps = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread({
    id: idCreator('event')
  }, newProps);
};
export var EventModal = function EventModal(_ref) {
  var children = _ref.children,
    event = _ref.event,
    eventActions = _ref.eventActions,
    _onOk = _ref.onOk,
    type = _ref.type,
    sourceId = _ref.sourceId,
    eventTypeOptions = _ref.eventTypeOptions;
  var _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    open = _useState2[0],
    setOpen = _useState2[1];
  var _useState3 = useState(event || getNewEvent()),
    _useState4 = _slicedToArray(_useState3, 2),
    tempEvent = _useState4[0],
    setTempEvent = _useState4[1];
  var edit = useRef(false);
  var setEdit = function setEdit(flag) {
    edit.current = flag;
  };
  useEffect(function () {
    if (!open) {
      edit.current = false;
      setTempEvent(getNewEvent({
        eventAction: eventActions[0]
      }));
    }
    if (event) {
      setTempEvent(event);
    }
  }, [event, open]);

  // 选择事件 菜单项
  var eventActionsMenus = useMemo(function () {
    return eventActions.map(function (action) {
      var menuItem = {
        key: action,
        label: eventActionInChinese[action]
      };
      if (type === EChangeType.EDIT && action !== (event === null || event === void 0 ? void 0 : event.eventAction)) {
        menuItem.disabled = true;
      }
      return menuItem;
    });
  }, [eventActions, type, event === null || event === void 0 ? void 0 : event.eventAction]);

  // 选择对应动作 菜单项
  var eventTypeMenus = useMemo(function () {
    return Object.entries(eventTypeChinese).map(function (_ref2) {
      var _ref3 = _slicedToArray(_ref2, 2),
        key = _ref3[0],
        label = _ref3[1];
      var menuItem = {
        key: key,
        label: label
      };
      if (type === EChangeType.EDIT && key !== (event === null || event === void 0 ? void 0 : event.eventType)) {
        menuItem.disabled = true;
      }
      return menuItem;
    }).filter(function (_ref4) {
      var key = _ref4.key;
      if (eventTypeOptions !== null && eventTypeOptions !== void 0 && eventTypeOptions.length) {
        if (eventTypeOptions.indexOf(key) === -1) return false;
      }
      // 只有组件加载后的动作才能关联服务
      if (tempEvent.eventAction === EEventAction.ON_LOADED) return true;
      return key !== EEventType.LINK_SERVICE;
    });
  }, [type, event === null || event === void 0 ? void 0 : event.eventType, tempEvent.eventAction, eventTypeOptions]);
  var handleChangeEvent = function handleChangeEvent(field, value) {
    if ((field === 'eventAction' || field === 'eventType') && edit.current) {
      if (tempEvent[field] === value) return;
      Modal.confirm({
        title: '编辑数据未保存，切换后将清除数据，确认切换？',
        onOk: function onOk() {
          setEdit(false);
          setTempEvent(function (prev) {
            return _objectSpread(_objectSpread({}, prev), {}, _defineProperty(_defineProperty({}, field, value), "eventTargets", []));
          });
        }
      });
    } else {
      setTempEvent(function (prev) {
        return _objectSpread(_objectSpread({}, prev), {}, _defineProperty({}, field, value));
      });
    }
  };
  var eventContextValue = useMemo(function () {
    return {
      currentEvent: tempEvent,
      handleChangeEvent: handleChangeEvent,
      setEdit: setEdit,
      sourceId: sourceId
    };
  }, [tempEvent, handleChangeEvent, setEdit, sourceId]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.isValidElement(children) && /*#__PURE__*/React.cloneElement(children, {
    onClick: function onClick() {
      return setOpen(true);
    }
  }), /*#__PURE__*/React.createElement(Modal, {
    width: 840,
    open: open,
    title: "".concat(type === EChangeType.ADD ? '新增' : '编辑', "\u4E8B\u4EF6"),
    maskClosable: false,
    onCancel: function onCancel() {
      setOpen(false);
    },
    destroyOnClose: true,
    onOk: function onOk() {
      if (!tempEvent.eventAction) {
        return message.error('选择事件');
      }
      if (!tempEvent.eventType) {
        return message.error('选择对应动作');
      }
      if (type === EChangeType.ADD && !edit.current) {
        tempEvent.eventTargets = [];
      }
      _onOk(tempEvent);
      setOpen(false);
    }
  }, /*#__PURE__*/React.createElement(EventModalContext.Provider, {
    value: eventContextValue
  }, /*#__PURE__*/React.createElement("div", {
    className: prefixCls('event-modal-wrap')
  }, /*#__PURE__*/React.createElement("div", {
    className: prefixCls('event-modal-content')
  }, /*#__PURE__*/React.createElement(SelectComponent, {
    className: prefixCls('action-select'),
    title: "\u9009\u62E9\u4E8B\u4EF6",
    menuItems: eventActionsMenus,
    onChange: function onChange(v) {
      return handleChangeEvent('eventAction', v);
    },
    value: tempEvent === null || tempEvent === void 0 ? void 0 : tempEvent.eventAction
  }), /*#__PURE__*/React.createElement(SelectComponent, {
    className: prefixCls('type-select'),
    title: "\u9009\u62E9\u5BF9\u5E94\u7684\u52A8\u4F5C",
    menuItems: tempEvent.eventAction ? eventTypeMenus : [],
    onChange: function onChange(v) {
      return handleChangeEvent('eventType', v);
    },
    value: tempEvent === null || tempEvent === void 0 ? void 0 : tempEvent.eventType
  }), /*#__PURE__*/React.createElement(ActionConfig, {
    className: prefixCls('event-relation-config'),
    title: "\u52A8\u4F5C\u76F8\u5173\u914D\u7F6E",
    currentEvent: tempEvent,
    operationType: type
  }))))));
};
