// @roddan/form-editor/es/utils/handle-emit-event.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { EEventType, EValidateType } from "../types";
import { getValueFromInput, dynamicGetStore } from "./";
import { validateParams } from '.';
// 设置组件值
export var emitSettingValue = function emitSettingValue(params) {
  var emitter = params.emitter,
    eventType = params.eventType,
    target = params.target;
  var targetElementId = target.targetElementId,
    setValue = target.setValue,
    targetPayload = target.targetPayload;
  var validate = validateParams([targetElementId, targetPayload]);
  if (!validate) return;
  return function (value) {
    emitter.emit(targetElementId, {
      targetElementId: targetElementId,
      eventType: eventType,
      setValue: getValueFromInput(setValue),
      targetPayload: targetPayload,
      value: value
    });
  };
};

// 刷新服务
export var emitRefreshService = function emitRefreshService(params) {
  var emitter = params.emitter,
    eventType = params.eventType,
    target = params.target;
  var targetServiceId = target.targetServiceId,
    targetPayload = target.targetPayload,
    refreshFlag = target.refreshFlag,
    updateField = target.updateField;
  var validate = validateParams([targetServiceId, targetPayload, refreshFlag]);
  if (!validate) return;
  var store = dynamicGetStore();
  if (!store.getService(targetServiceId)) return;
  return function (value) {
    emitter.emit(targetServiceId, {
      targetServiceId: targetServiceId,
      eventType: eventType,
      updateField: updateField,
      targetPayload: targetPayload,
      value: value,
      refreshFlag: refreshFlag
    });
  };
};

// 表单校验
export var emitValidateForm = function emitValidateForm(params) {
  var store = dynamicGetStore();
  var target = params.target;
  var validateField = target.validateField,
    sourceId = target.sourceId;
  var fields = validateField === EValidateType.CURRENT ? [sourceId] : undefined;
  return function () {
    var _store$formInstance;
    return (_store$formInstance = store.formInstance) === null || _store$formInstance === void 0 ? void 0 : _store$formInstance.validateFields(fields);
  };
};
export var handleEmitEvent = function handleEmitEvent(emitter, events) {
  var functions = {};
  events.forEach(function (event) {
    var eventAction = event.eventAction,
      eventType = event.eventType,
      eventTargets = event.eventTargets;
    eventTargets === null || eventTargets === void 0 || eventTargets.forEach(function (target) {
      var params = {
        emitter: emitter,
        eventType: eventType,
        target: target
      };
      var emitFn;
      switch (eventType) {
        case EEventType.SETTING_VALUE:
          emitFn = emitSettingValue(params);
          break;
        case EEventType.UPDATE_SERVICE:
          emitFn = emitRefreshService(params);
          break;
        case EEventType.VALIDATE:
          emitFn = emitValidateForm(params);
          break;
      }
      if (!emitFn) return;
      if (functions[eventAction]) {
        var _functions;
        (_functions = functions[eventAction]) === null || _functions === void 0 || _functions.push(emitFn);
      } else {
        functions[eventAction] = [emitFn];
      }
    });
  });
  var formatFunctions = Object.entries(functions).reduce(function (memo, _ref) {
    var _ref2 = _slicedToArray(_ref, 2),
      action = _ref2[0],
      emitFns = _ref2[1];
    // @ts-ignore
    memo[action] = function (v) {
      return emitFns.forEach(function (fn) {
        return fn === null || fn === void 0 ? void 0 : fn(v);
      });
    };
    return memo;
  }, {});
  return formatFunctions;
};
