// @roddan/form-editor/es/components/form-service/service-item.js
import React from 'react';
import { Popconfirm, Popover, Typography } from 'antd';
import { CopyOutlined, DeleteOutlined, EditOutlined } from '@ant-design/icons';
import ServiceModal from "./service-modal";
import { prefixCls } from "../../const";
import store from "../../store";
export var ServiceItem = function ServiceItem(_ref) {
  var service = _ref.service;
  return /*#__PURE__*/React.createElement("div", {
    className: prefixCls('service-item')
  }, /*#__PURE__*/React.createElement(Typography.Text, {
    ellipsis: {
      tooltip: service === null || service === void 0 ? void 0 : service.name
    },
    style: {
      width: 200
    }
  }, service === null || service === void 0 ? void 0 : service.name), /*#__PURE__*/React.createElement("div", {
    className: prefixCls('service-item-operate')
  }, /*#__PURE__*/React.createElement(ServiceModal, {
    service: service
  }, /*#__PURE__*/React.createElement(Popover, {
    content: "\u7F16\u8F91\u670D\u52A1"
  }, /*#__PURE__*/React.createElement(EditOutlined, null))), /*#__PURE__*/React.createElement(Popover, {
    content: "\u590D\u5236\u670D\u52A1"
  }, /*#__PURE__*/React.createElement(CopyOutlined, {
    onClick: function onClick() {
      return store.copyService(service);
    }
  })), /*#__PURE__*/React.createElement(Popconfirm, {
    title: "\u786E\u8BA4\u5220\u9664?",
    placement: "topLeft",
    onConfirm: function onConfirm() {
      store.deleteService(service.id);
    }
  }, /*#__PURE__*/React.createElement(Popover, {
    content: "\u5220\u9664\u670D\u52A1",
    placement: "bottomLeft"
  }, /*#__PURE__*/React.createElement(DeleteOutlined, null)))));
};
