// @roddan/form-editor/lib/components/event-setting/event-collapse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EventCollapse = void 0;
var _icons = require("@ant-design/icons");
var _antd = require("antd");
var _react = _interopRequireDefault(require("react"));
var _const = require("../../const");
var _types = require("../../types");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var EventCollapse = exports.EventCollapse = function EventCollapse(_ref) {
  var collopaseItems = _ref.collopaseItems,
    onDelete = _ref.onDelete,
    EditComponent = _ref.EditComponent;
  var items = collopaseItems.map(function (item, i) {
    return {
      key: i,
      label: item.label,
      children: item.events.map(function (event, innerI) {
        return /*#__PURE__*/_react.default.createElement(EditComponent, {
          evt: event,
          key: innerI
        }, /*#__PURE__*/_react.default.createElement(_antd.Dropdown, {
          menu: {
            items: [{
              label: /*#__PURE__*/_react.default.createElement(_antd.Popconfirm, {
                title: "\u786E\u8BA4\u5220\u9664",
                onConfirm: function onConfirm() {
                  return onDelete(event.id);
                }
              }, "\u5220\u9664"),
              key: 1
            }]
          },
          trigger: ['contextMenu']
          // @ts-ignore
          ,
          getPopupContainer: function getPopupContainer(n) {
            return n.parentNode;
          }
        }, /*#__PURE__*/_react.default.createElement("div", {
          className: "collapse-item"
        }, _types.eventTypeChinese[event.eventType])));
      })
    };
  });
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _const.prefixCls)('event-collapse')
  }, /*#__PURE__*/_react.default.createElement(_antd.Collapse, {
    ghost: true,
    items: items,
    defaultActiveKey: items.map(function (item) {
      return item.key;
    }),
    expandIcon: function expandIcon(_ref2) {
      var isActive = _ref2.isActive;
      return isActive ? /*#__PURE__*/_react.default.createElement(_icons.MinusSquareOutlined, null) : /*#__PURE__*/_react.default.createElement(_icons.PlusSquareOutlined, null);
    }
  }));
};
