// @roddan/form-editor/lib/components/event-setting/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  EventSetting: true
};
exports.EventSetting = void 0;
var _react = _interopRequireDefault(require("react"));
var _antd = require("antd");
var _lodashEs = require("lodash-es");
var _mobxReactLite = require("mobx-react-lite");
var _const = require("../../const");
var _hooks = require("../../hooks");
var _utils = require("../../utils");
var _types = require("../../types");
var _eventCollapse = require("./event-collapse");
Object.keys(_eventCollapse).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _eventCollapse[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _eventCollapse[key];
    }
  });
});
var _eventModal = require("../event-modal");
require("./style.less");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
var formatForCollapse = function formatForCollapse(events) {
  var group = (0, _lodashEs.groupBy)(events, 'eventAction');
  return Object.entries(group).map(function (_ref) {
    var _ref2 = _slicedToArray(_ref, 2),
      action = _ref2[0],
      events = _ref2[1];
    return {
      label: _types.eventActionInChinese[action],
      events: events
    };
  });
};
var EventSetting = exports.EventSetting = (0, _mobxReactLite.observer)(function (_ref3) {
  var eventActions = _ref3.eventActions,
    type = _ref3.type,
    eventTypeOptions = _ref3.eventTypeOptions;
  var _useCurrent = (0, _hooks.useCurrent)(type),
    current = _useCurrent.current,
    setProp = _useCurrent.setProp;
  var sourceId = current.id;
  var handleSaveEvents = function handleSaveEvents(type, event) {
    var events = (0, _lodashEs.cloneDeep)((current === null || current === void 0 ? void 0 : current.events) || []);
    if (type === _types.EChangeType.ADD) {
      var _event$eventTargets;
      if (!(event !== null && event !== void 0 && (_event$eventTargets = event.eventTargets) !== null && _event$eventTargets !== void 0 && _event$eventTargets.length)) return;
      var sameActionEvent = events.find(function (existEvent) {
        return existEvent.eventAction === event.eventAction && existEvent.eventType === event.eventType;
      });
      if (sameActionEvent) {
        var _sameActionEvent$even;
        sameActionEvent.eventTargets = (_sameActionEvent$even = sameActionEvent.eventTargets) === null || _sameActionEvent$even === void 0 ? void 0 : _sameActionEvent$even.concat(event.eventTargets || []);
        _antd.message.success('新增事件已被合并');
      } else {
        events.push(event);
      }
      (0, _utils.handleLinkService)(event);
    } else {
      var _current$events, _event$eventTargets2;
      var idx = current === null || current === void 0 || (_current$events = current.events) === null || _current$events === void 0 ? void 0 : _current$events.findIndex(function (evt) {
        return event.id === evt.id;
      });
      (0, _utils.handleUnLinkService)(events[idx]);
      events[idx] = event;
      if (event !== null && event !== void 0 && (_event$eventTargets2 = event.eventTargets) !== null && _event$eventTargets2 !== void 0 && _event$eventTargets2.length) {
        (0, _utils.handleLinkService)(event);
      } else {
        events.splice(idx, 1);
      }
    }
    // modal过度效果
    setTimeout(function () {
      setProp('events', events);
    }, 200);
  };
  var handleDelete = function handleDelete(id) {
    var _current$events2;
    var events = (0, _lodashEs.cloneDeep)(current === null || current === void 0 ? void 0 : current.events);
    var idx = current === null || current === void 0 || (_current$events2 = current.events) === null || _current$events2 === void 0 ? void 0 : _current$events2.findIndex(function (event) {
      return event.id === id;
    });
    (0, _utils.handleUnLinkService)(events[idx]);
    events.splice(idx, 1);
    setProp('events', events);
  };
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _const.prefixCls)('event-common-wrap')
  }, /*#__PURE__*/_react.default.createElement(_eventModal.EventModal, {
    eventActions: eventActions,
    onOk: function onOk(evt) {
      return handleSaveEvents(_types.EChangeType.ADD, evt);
    },
    type: _types.EChangeType.ADD,
    sourceId: sourceId,
    eventTypeOptions: eventTypeOptions
  }, /*#__PURE__*/_react.default.createElement(_antd.Button, {
    type: "dashed",
    className: (0, _const.prefixCls)('event-button-add')
  }, "+ \u65B0\u589E\u4E8B\u4EF6")), /*#__PURE__*/_react.default.createElement(_eventCollapse.EventCollapse, {
    collopaseItems: formatForCollapse((current === null || current === void 0 ? void 0 : current.events) || []),
    onDelete: handleDelete,
    EditComponent: function EditComponent(_ref4) {
      var children = _ref4.children,
        evt = _ref4.evt;
      return /*#__PURE__*/_react.default.createElement(_eventModal.EventModal, {
        eventActions: eventActions,
        onOk: function onOk(evt) {
          return handleSaveEvents(_types.EChangeType.EDIT, evt);
        },
        event: evt,
        type: _types.EChangeType.EDIT,
        sourceId: sourceId,
        eventTypeOptions: eventTypeOptions
      }, children);
    }
  }));
});
