// @roddan/form-editor/es/index.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import React, { useEffect, useMemo } from 'react';
import { ConfigProvider, Form } from 'antd';
import locale from 'antd/locale/zh_CN';
import 'dayjs/locale/zh-cn';
import { prefixCls, defaultFormAttrs } from "./const";
import store from "./store";
import { EditorContext } from "./context";
import "./index.less";
export * from "./views";
export * from "./const";
export * from "./utils";
export var FormEditor = function FormEditor(_ref) {
  var mode = _ref.mode,
    defaultValue = _ref.defaultValue,
    children = _ref.children;
  var _Form$useForm = Form.useForm(),
    _Form$useForm2 = _slicedToArray(_Form$useForm, 1),
    form = _Form$useForm2[0];
  useEffect(function () {
    store.setForm(form);
  }, [form]);
  useEffect(function () {
    var schema = {};
    if (defaultValue) {
      schema = defaultValue;
    } else if (localStorage.getItem('schema')) {
      try {
        schema = JSON.parse(localStorage.getItem('schema'));
      } catch (e) {}
    }
    var _schema = schema,
      _schema$formElements = _schema.formElements,
      formElements = _schema$formElements === void 0 ? [] : _schema$formElements,
      _schema$fieldValues = _schema.fieldValues,
      fieldValues = _schema$fieldValues === void 0 ? {} : _schema$fieldValues,
      _schema$formAttrs = _schema.formAttrs,
      formAttrs = _schema$formAttrs === void 0 ? defaultFormAttrs : _schema$formAttrs,
      _schema$formServices = _schema.formServices,
      formServices = _schema$formServices === void 0 ? [] : _schema$formServices;
    store.setFormElements(formElements);
    store.setFieldsValues(fieldValues);
    store.setFormAttrs(formAttrs);
    store.setFormServices(formServices);
  }, [defaultValue]);
  var contextValue = useMemo(function () {
    return {
      mode: mode
    };
  }, [mode]);
  return /*#__PURE__*/React.createElement(EditorContext.Provider, {
    value: contextValue
  }, /*#__PURE__*/React.createElement(ConfigProvider, {
    locale: locale
  }, /*#__PURE__*/React.createElement(Form, {
    form: form
  }, /*#__PURE__*/React.createElement("div", {
    className: prefixCls('form')
  }, children))));
};
