// @roddan/form-editor/es/components/element-layout/index.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import React, { useMemo } from 'react';
import { Col, Form } from 'antd';
import { observer } from 'mobx-react-lite';
import styled, { css } from 'styled-components';
import { cloneDeep } from 'lodash-es';
import { useElementCommon } from "../../hooks";
import { useEditorContext } from "../../context";
import { WrapEl } from "./wrap-el";
var StyledDiv = styled.div.withConfig({
  displayName: "StyledDiv",
  componentId: "form-editor-42cd__sc-10e83ga-0"
})(function (_ref) {
  var elementNameDisplay = _ref.elementNameDisplay;
  return "\n    ".concat(css({
    display: 'flex',
    flexDirection: elementNameDisplay === 'horizontal' ? 'row' : 'column',
    alignItems: elementNameDisplay === 'horizontal' ? 'center' : ''
  }).join(';'), "\n    .title-required::before {\n      display: inline-block;\n      margin-inline-end: 4px;\n      color: #ff4d4f;\n      font-size: 14px;\n      font-family: SimSun, sans-serif;\n      line-height: 1;\n      content: \"*\";\n    }\n  ");
});
export var ElementLayout = observer(function (_ref2) {
  var _rules$;
  var element = _ref2.element,
    children = _ref2.children;
  var elementName = element.elementName,
    elementNameDisplay = element.elementNameDisplay,
    id = element.id,
    gridOffset = element.gridOffset,
    gridSpan = element.gridSpan,
    showElementName = element.showElementName,
    gridLayout = element.gridLayout,
    regExps = element.regExps,
    parentId = element.parentId,
    type = element.type;
  var _useElementCommon = useElementCommon(element),
    elCss = _useElementCommon.elCss,
    contaninerCss = _useElementCommon.contaninerCss;
  var _useEditorContext = useEditorContext(),
    mode = _useEditorContext.mode;

  // 自定义css样式
  var style = useMemo(function () {
    var finnalStyle = contaninerCss ? cloneDeep(contaninerCss) : {};
    if (!gridLayout) {
      Object.assign(finnalStyle, {
        flex: 'none',
        maxWidth: 'inherit'
      });
    }
    return finnalStyle;
  }, [contaninerCss, gridLayout]);

  // 校验规则
  var rules = useMemo(function () {
    if (!(regExps !== null && regExps !== void 0 && regExps.length)) return [];
    var arr = [];
    var requiredRule = regExps[0];
    arr.push({
      required: requiredRule === null || requiredRule === void 0 ? void 0 : requiredRule.enable,
      message: requiredRule === null || requiredRule === void 0 ? void 0 : requiredRule.message
    });
    regExps.slice(1).filter(function (item) {
      return item.enable && item.regexp && item.message;
    }).forEach(function (patternItem) {
      arr.push({
        validator: function validator(_) {
          var value = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : '';
          var reg = new RegExp(patternItem.regexp);
          if (value !== null && value !== void 0 && value.match(reg)) {
            return Promise.resolve();
          }
          return Promise.reject(patternItem.message);
        }
      });
    });
    return arr;
  }, [regExps]);

  // 偏移量值 仅使用栅格布局才生效
  var offset = gridLayout ? gridOffset || 0 : 0;
  return /*#__PURE__*/React.createElement(Col, {
    span: gridSpan,
    offset: offset,
    style: style,
    "data-parent-id": parentId,
    "data-id": id,
    "data-type": type
  }, /*#__PURE__*/React.createElement(Form.Item, {
    name: id,
    rules: rules,
    style: {
      marginBottom: 0
    }
  }, /*#__PURE__*/React.createElement(WrapEl, {
    el: element,
    mode: mode
  }, /*#__PURE__*/React.createElement(StyledDiv, {
    elementNameDisplay: elementNameDisplay
  }, showElementName && /*#__PURE__*/React.createElement("div", {
    dangerouslySetInnerHTML: {
      __html: elementName
    }
    // @ts-ignore
    ,
    className: rules !== null && rules !== void 0 && (_rules$ = rules[0]) !== null && _rules$ !== void 0 && _rules$.required ? 'title-required' : ''
  }), /*#__PURE__*/React.createElement("div", {
    style: {
      flex: 1
    }
  }, /*#__PURE__*/React.isValidElement(children) && /*#__PURE__*/React.cloneElement(children, _objectSpread(_objectSpread({}, (children === null || children === void 0 ? void 0 : children.props) || {}), {}, {
    style: elCss || {}
  })))))));
});
