// @roddan/form-editor/es/components/element-layout/selected-actions.js
import React from 'react';
import styled, { css } from 'styled-components';
import { CopyOutlined, DeleteOutlined } from '@ant-design/icons';
import store from "../../store";
var StyledDiv = styled.div.withConfig({
  displayName: "StyledDiv",
  componentId: "form-editor-42cd__sc-k1gxpm-0"
})(function () {
  return "\n    ".concat(css({
    position: 'absolute',
    right: 0,
    top: -8,
    display: 'flex',
    zIndex: 20
  }).join(';'), "\n    .fe-selected-action{\n      ").concat(css({
    marginRight: 6,
    borderColor: '#409EFF',
    color: '#409EFF',
    background: '#fff',
    width: 22,
    height: 22,
    lineHeight: '20px',
    textAlign: 'center',
    borderRadius: '50%',
    fontSize: 12,
    border: '1px solid',
    cursor: 'pointer'
  }).join(';'), "\n      :hover {\n        color: white;\n      }\n    }\n    .copy-action:hover {\n      background: #409EFF;\n    }\n    .delete-action {\n      border-color: #F56C6C;\n      color: #F56C6C;\n    }\n    .delete-action:hover {\n      background: #F56C6C;\n    }\n  ");
});
export var SelectedActions = function SelectedActions() {
  return /*#__PURE__*/React.createElement(StyledDiv, null, /*#__PURE__*/React.createElement("div", {
    className: "fe-selected-action copy-action",
    onClick: function onClick() {
      var newEl = store.copyEl(store.selectedElement);
      store.setSelectedElement(newEl);
    }
  }, /*#__PURE__*/React.createElement(CopyOutlined, null)), /*#__PURE__*/React.createElement("div", {
    className: "fe-selected-action delete-action",
    onClick: function onClick() {
      store.deleteEl(store.selectedElement);
      store.setSelectedElement({});
    }
  }, /*#__PURE__*/React.createElement(DeleteOutlined, null)));
};
