// @roddan/form-editor/es/components/element-layout/wrap-el.js
import React, { useCallback, useRef } from 'react';
import { observer } from 'mobx-react-lite';
import store, { tabStore } from "../../store";
import { useDesignEffect } from "../../hooks";
import eventStore from "../../store/eventStore";
import { SelectedActions } from "./selected-actions";
import { DesignWrapDiv, Mask, Icon } from "./styled";
import { CONTAINERS } from "../../const";
var EventIcon = observer(function (_ref) {
  var events = _ref.events;
  if (!(events !== null && events !== void 0 && events.length)) return null;
  var validate = events.every(function (event) {
    var eventTargets = event.eventTargets;
    if (!(eventTargets !== null && eventTargets !== void 0 && eventTargets.length)) return true;
    return eventTargets.every(function (target) {
      var targetElementId = target.targetElementId,
        targetServiceId = target.targetServiceId;
      if (!targetElementId && !targetServiceId) return true;
      if (targetElementId) return !!store.getElement(targetElementId);
      if (targetServiceId) return !!store.getService(targetServiceId);
      return true;
    });
  });
  return /*#__PURE__*/React.createElement(Icon, {
    validate: validate
  });
});
var WrapDesignEl = observer(function (_ref2) {
  var _store$selectedElemen, _store$selectedElemen2;
  var children = _ref2.children,
    el = _ref2.el;
  var ref = useRef(null);
  useDesignEffect(function () {
    eventStore.iterateEl(el);
  }, [el.events]);
  var handleSelect = useCallback(function () {
    store.setSelectedElement(el);
    tabStore.init();
  }, [el]);
  return /*#__PURE__*/React.createElement(DesignWrapDiv, {
    selected: ((_store$selectedElemen = store.selectedElement) === null || _store$selectedElemen === void 0 ? void 0 : _store$selectedElemen.id) === el.id,
    onMouseDownCapture: handleSelect,
    ref: ref
  }, /*#__PURE__*/React.createElement(Mask, {
    horizontal: store.formAttrs.horizontalGap + 2,
    vertical: store.formAttrs.verticalGap + 2,
    hide: CONTAINERS.includes(el.type)
  }), ((_store$selectedElemen2 = store.selectedElement) === null || _store$selectedElemen2 === void 0 ? void 0 : _store$selectedElemen2.id) === el.id && /*#__PURE__*/React.createElement(SelectedActions, null), children, /*#__PURE__*/React.createElement(EventIcon, {
    events: el.events
  }));
});
export var WrapEl = function WrapEl(_ref3) {
  var children = _ref3.children,
    el = _ref3.el,
    mode = _ref3.mode;
  if (mode !== 'design') return /*#__PURE__*/React.createElement(React.Fragment, null, children);
  return /*#__PURE__*/React.createElement(WrapDesignEl, {
    el: el
  }, children);
};
