// @roddan/form-editor/lib/utils/handle-deal-service.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.unBindService = exports.unBindFromService = exports.unBindFromElement = exports.handleUnLinkService = exports.handleLinkService = exports.bindService = exports.bindFromCopiedElement = void 0;
var _lodashEs = require("lodash-es");
var _types = require("../types");
var _ = require(".");
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
var handleDealService = function handleDealService(type, event) {
  var eventTargets = event.eventTargets,
    eventType = event.eventType;
  if (eventType !== _types.EEventType.LINK_SERVICE) return;
  var store = (0, _.dynamicGetStore)();
  eventTargets === null || eventTargets === void 0 || eventTargets.forEach(function (target) {
    var targetServiceId = target.targetServiceId,
      sourceId = target.sourceId,
      linkRefreshField = target.linkRefreshField,
      getFieldFromService = target.getFieldFromService;
    if (!targetServiceId || !store.getService(targetServiceId)) return;
    var service = store.getService(targetServiceId);
    var linkingElements = service.linkingElements;
    // 关联
    if (type === 'link') {
      var linkData = {
        id: sourceId,
        field: linkRefreshField,
        getFieldFromService: getFieldFromService
      };
      if (!linkingElements) {
        store.setService(targetServiceId, {
          linkingElements: [linkData]
        });
      } else {
        linkingElements.push(linkData);
        store.setService(targetServiceId, {
          linkingElements: linkingElements
        });
      }
      bindService(sourceId, targetServiceId);
    } else {
      // 取消关联
      if (linkingElements && linkingElements.length) {
        var idx = linkingElements.findIndex(function (item) {
          return item.id === sourceId;
        });
        if (idx > -1) {
          unBindService(linkingElements[idx].id, targetServiceId);
          linkingElements.splice(idx, 1);
          store.setService(targetServiceId, {
            linkingElements: linkingElements
          });
        }
      }
    }
  });
};

// 事件 - 处理关联服务的数据
var handleLinkService = exports.handleLinkService = function handleLinkService(event) {
  handleDealService('link', event);
};

// 事件 - 解除关联数据
var handleUnLinkService = exports.handleUnLinkService = function handleUnLinkService(event) {
  handleDealService('unlink', event);
};
var getServicesFromElement = function getServicesFromElement(elId) {
  var store = (0, _.dynamicGetStore)();
  var element = store.getElement(elId);
  var linkServices = element.linkServices;
  return linkServices !== null && linkServices !== void 0 ? linkServices : [];
};
var getElementFromService = function getElementFromService(servId) {
  var _service$linkingEleme;
  var store = (0, _.dynamicGetStore)();
  var service = store.getService(servId);
  return (_service$linkingEleme = service === null || service === void 0 ? void 0 : service.linkingElements) !== null && _service$linkingEleme !== void 0 ? _service$linkingEleme : [];
};
var bindService = exports.bindService = function bindService(elId, serviceId) {
  var store = (0, _.dynamicGetStore)();
  var services = getServicesFromElement(elId);
  // console.log('services', services)
  if (services.indexOf(serviceId) > -1) return;
  services.push(serviceId);
  store.setElementProp(elId, 'linkServices', services);
};
var unBindService = exports.unBindService = function unBindService(elId, serviceId) {
  var store = (0, _.dynamicGetStore)();
  var services = getServicesFromElement(elId);
  var idx = services.findIndex(function (serv) {
    return serv === serviceId;
  });
  if (idx === -1) return;
  services.splice(idx, 1);
  store.setElementProp(elId, 'linkServices', services);
};
// 删除元素 解绑服务
var unBindFromElement = exports.unBindFromElement = function unBindFromElement(elId) {
  var store = (0, _.dynamicGetStore)();
  var services = getServicesFromElement(elId);
  services.forEach(function (servId) {
    var linkingElements = (0, _lodashEs.cloneDeep)(getElementFromService(servId));
    var idx = linkingElements.findIndex(function (el) {
      return el.id === elId;
    });
    if (idx === -1) return;
    linkingElements.splice(idx, 1);
    store.setService(servId, {
      linkingElements: linkingElements
    });
  });
};
var unBindFromService = exports.unBindFromService = function unBindFromService(servId) {
  var elements = getElementFromService(servId);
  elements.forEach(function (el) {
    unBindService(el.id, servId);
  });
};

// 复制元素 关联服务
var bindFromCopiedElement = exports.bindFromCopiedElement = function bindFromCopiedElement(oldId, newId) {
  var services = getServicesFromElement(oldId);
  var store = (0, _.dynamicGetStore)();
  services.forEach(function (servId) {
    var linkingElements = (0, _lodashEs.cloneDeep)(getElementFromService(servId));
    var linkData = linkingElements.find(function (el) {
      return el.id === oldId;
    });
    linkingElements.push(_objectSpread(_objectSpread({}, linkData), {}, {
      id: newId
    }));
    store.setService(servId, {
      linkingElements: linkingElements
    });
  });
};
