// @roddan/form-editor/lib/components/event-modal/event-modal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EventModal = void 0;
var _react = _interopRequireWildcard(require("react"));
var _antd = require("antd");
var _actionConfig = require("./action-config");
var _context = require("./context");
var _selectComponent = require("./select-component");
var _const = require("../../const");
var _types = require("../../types");
var _utils = require("../../utils");
require("./style.less");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
var getNewEvent = function getNewEvent() {
  var newProps = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread({
    id: (0, _utils.idCreator)('event')
  }, newProps);
};
var EventModal = exports.EventModal = function EventModal(_ref) {
  var children = _ref.children,
    event = _ref.event,
    eventActions = _ref.eventActions,
    _onOk = _ref.onOk,
    type = _ref.type,
    sourceId = _ref.sourceId,
    eventTypeOptions = _ref.eventTypeOptions;
  var _useState = (0, _react.useState)(false),
    _useState2 = _slicedToArray(_useState, 2),
    open = _useState2[0],
    setOpen = _useState2[1];
  var _useState3 = (0, _react.useState)(event || getNewEvent()),
    _useState4 = _slicedToArray(_useState3, 2),
    tempEvent = _useState4[0],
    setTempEvent = _useState4[1];
  var edit = (0, _react.useRef)(false);
  var setEdit = function setEdit(flag) {
    edit.current = flag;
  };
  (0, _react.useEffect)(function () {
    if (!open) {
      edit.current = false;
      setTempEvent(getNewEvent({
        eventAction: eventActions[0]
      }));
    }
    if (event) {
      setTempEvent(event);
    }
  }, [event, open]);

  // 选择事件 菜单项
  var eventActionsMenus = (0, _react.useMemo)(function () {
    return eventActions.map(function (action) {
      var menuItem = {
        key: action,
        label: _types.eventActionInChinese[action]
      };
      if (type === _types.EChangeType.EDIT && action !== (event === null || event === void 0 ? void 0 : event.eventAction)) {
        menuItem.disabled = true;
      }
      return menuItem;
    });
  }, [eventActions, type, event === null || event === void 0 ? void 0 : event.eventAction]);

  // 选择对应动作 菜单项
  var eventTypeMenus = (0, _react.useMemo)(function () {
    return Object.entries(_types.eventTypeChinese).map(function (_ref2) {
      var _ref3 = _slicedToArray(_ref2, 2),
        key = _ref3[0],
        label = _ref3[1];
      var menuItem = {
        key: key,
        label: label
      };
      if (type === _types.EChangeType.EDIT && key !== (event === null || event === void 0 ? void 0 : event.eventType)) {
        menuItem.disabled = true;
      }
      return menuItem;
    }).filter(function (_ref4) {
      var key = _ref4.key;
      if (eventTypeOptions !== null && eventTypeOptions !== void 0 && eventTypeOptions.length) {
        if (eventTypeOptions.indexOf(key) === -1) return false;
      }
      // 只有组件加载后的动作才能关联服务
      if (tempEvent.eventAction === _types.EEventAction.ON_LOADED) return true;
      return key !== _types.EEventType.LINK_SERVICE;
    });
  }, [type, event === null || event === void 0 ? void 0 : event.eventType, tempEvent.eventAction, eventTypeOptions]);
  var handleChangeEvent = function handleChangeEvent(field, value) {
    if ((field === 'eventAction' || field === 'eventType') && edit.current) {
      if (tempEvent[field] === value) return;
      _antd.Modal.confirm({
        title: '编辑数据未保存，切换后将清除数据，确认切换？',
        onOk: function onOk() {
          setEdit(false);
          setTempEvent(function (prev) {
            return _objectSpread(_objectSpread({}, prev), {}, _defineProperty(_defineProperty({}, field, value), "eventTargets", []));
          });
        }
      });
    } else {
      setTempEvent(function (prev) {
        return _objectSpread(_objectSpread({}, prev), {}, _defineProperty({}, field, value));
      });
    }
  };
  var eventContextValue = (0, _react.useMemo)(function () {
    return {
      currentEvent: tempEvent,
      handleChangeEvent: handleChangeEvent,
      setEdit: setEdit,
      sourceId: sourceId
    };
  }, [tempEvent, handleChangeEvent, setEdit, sourceId]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.isValidElement(children) && /*#__PURE__*/_react.default.cloneElement(children, {
    onClick: function onClick() {
      return setOpen(true);
    }
  }), /*#__PURE__*/_react.default.createElement(_antd.Modal, {
    width: 840,
    open: open,
    title: "".concat(type === _types.EChangeType.ADD ? '新增' : '编辑', "\u4E8B\u4EF6"),
    maskClosable: false,
    onCancel: function onCancel() {
      setOpen(false);
    },
    destroyOnClose: true,
    onOk: function onOk() {
      if (!tempEvent.eventAction) {
        return _antd.message.error('选择事件');
      }
      if (!tempEvent.eventType) {
        return _antd.message.error('选择对应动作');
      }
      if (type === _types.EChangeType.ADD && !edit.current) {
        tempEvent.eventTargets = [];
      }
      _onOk(tempEvent);
      setOpen(false);
    }
  }, /*#__PURE__*/_react.default.createElement(_context.EventModalContext.Provider, {
    value: eventContextValue
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _const.prefixCls)('event-modal-wrap')
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _const.prefixCls)('event-modal-content')
  }, /*#__PURE__*/_react.default.createElement(_selectComponent.SelectComponent, {
    className: (0, _const.prefixCls)('action-select'),
    title: "\u9009\u62E9\u4E8B\u4EF6",
    menuItems: eventActionsMenus,
    onChange: function onChange(v) {
      return handleChangeEvent('eventAction', v);
    },
    value: tempEvent === null || tempEvent === void 0 ? void 0 : tempEvent.eventAction
  }), /*#__PURE__*/_react.default.createElement(_selectComponent.SelectComponent, {
    className: (0, _const.prefixCls)('type-select'),
    title: "\u9009\u62E9\u5BF9\u5E94\u7684\u52A8\u4F5C",
    menuItems: tempEvent.eventAction ? eventTypeMenus : [],
    onChange: function onChange(v) {
      return handleChangeEvent('eventType', v);
    },
    value: tempEvent === null || tempEvent === void 0 ? void 0 : tempEvent.eventType
  }), /*#__PURE__*/_react.default.createElement(_actionConfig.ActionConfig, {
    className: (0, _const.prefixCls)('event-relation-config'),
    title: "\u52A8\u4F5C\u76F8\u5173\u914D\u7F6E",
    currentEvent: tempEvent,
    operationType: type
  }))))));
};
