// @roddan/form-editor/lib/components/event-modal/select-component.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SelectComponent = void 0;
var _antd = require("antd");
var _classnames = _interopRequireDefault(require("classnames"));
var _react = _interopRequireDefault(require("react"));
var _const = require("../../const");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var SelectComponent = exports.SelectComponent = function SelectComponent(_ref) {
  var _ref$className = _ref.className,
    className = _ref$className === void 0 ? '' : _ref$className,
    title = _ref.title,
    menuItems = _ref.menuItems,
    onChange = _ref.onChange,
    value = _ref.value;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)((0, _const.prefixCls)('event-modal-column'), className)
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _const.prefixCls)('event-modal-title')
  }, title), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _const.prefixCls)('event-list')
  }, /*#__PURE__*/_react.default.createElement(_antd.Menu, {
    selectedKeys: value ? [value] : [],
    onClick: function onClick(_ref2) {
      var key = _ref2.key;
      onChange(key);
    },
    mode: "inline",
    items: menuItems
  })));
};
