// @roddan/form-editor/es/elements/radio/render-radio.js
import React from 'react';
import { Radio, Space } from 'antd';
import { observer } from 'mobx-react-lite';
import { ElementLayout } from "../../components";
import { useRegisterEvents, useFormUpdate } from "../../hooks";
import { EEventAction } from "../../types";
import store from "../../store";
var RenderRadioContent = function RenderRadioContent(_ref) {
  var fieldValue = _ref.fieldValue,
    element = _ref.element;
  var id = element.id,
    valueOptions = element.valueOptions,
    alignDirection = element.alignDirection;
  var _useRegisterEvents = useRegisterEvents(element),
    eventFunctions = _useRegisterEvents.eventFunctions;
  var onChange = function onChange(e) {
    store.setFieldValue(id, e.target.value);
  };
  useFormUpdate(function () {
    var _eventFunctions$EEven;
    (_eventFunctions$EEven = eventFunctions[EEventAction.ON_LOADED]) === null || _eventFunctions$EEven === void 0 || _eventFunctions$EEven.call(eventFunctions);
  }, [eventFunctions[EEventAction.ON_LOADED]]);
  useFormUpdate(function () {
    var _eventFunctions$EEven2;
    (_eventFunctions$EEven2 = eventFunctions[EEventAction.VALUE_CHANGE]) === null || _eventFunctions$EEven2 === void 0 || _eventFunctions$EEven2.call(eventFunctions, fieldValue);
  }, [fieldValue]);
  return /*#__PURE__*/React.createElement(ElementLayout, {
    element: element
  }, /*#__PURE__*/React.createElement(Radio.Group, {
    onChange: onChange,
    value: fieldValue
  }, /*#__PURE__*/React.createElement(Space, {
    direction: alignDirection
  }, valueOptions === null || valueOptions === void 0 ? void 0 : valueOptions.map(function (opt) {
    return /*#__PURE__*/React.createElement(Radio, {
      key: opt.id,
      value: opt.value
    }, opt.label);
  }))));
};
export var RenderRadio = observer(RenderRadioContent);
