// @roddan/ui/lib/react-sortable/react-sortable.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/react-sortable/react-sortable.tsx
var react_sortable_exports = {};
__export(react_sortable_exports, {
  ReactSortable: () => ReactSortable
});
module.exports = __toCommonJS(react_sortable_exports);
var import_antd = require("antd");
var import_classnames = __toESM(require("classnames"));
var import_react = __toESM(require("react"));
var import_sortablejs = __toESM(require("sortablejs"));
var import_util = require("./util");
var store = { dragging: null };
var ReactSortable = class extends import_react.Component {
  constructor(props) {
    super(props);
    this.ref = (0, import_react.createRef)();
  }
  isRow() {
    return !!this.props.rowProps;
  }
  componentDidMount() {
    if (this.ref.current === null || this.props.forbidden)
      return;
    const newOptions = this.makeOptions();
    const { sortableClass } = this.props;
    this.wrapEl = this.isRow() ? this.ref.current.querySelector(sortableClass ?? ".ant-row") : this.ref.current;
    if (!this.wrapEl)
      return;
    import_sortablejs.default.create(this.wrapEl, newOptions);
  }
  componentDidUpdate(prevProps) {
    if (prevProps.disabled !== this.props.disabled && this.sortable) {
      this.sortable.option("disabled", this.props.disabled);
    }
  }
  render() {
    const { tag, style, className, id, rowProps } = this.props;
    const classicProps = { style, className, id };
    const newTag = !tag || tag === null ? "div" : tag;
    if (!this.isRow()) {
      return (0, import_react.createElement)(
        newTag,
        {
          ref: this.ref,
          ...classicProps
        },
        this.getChildren()
      );
    }
    return (0, import_react.createElement)(
      newTag,
      {
        ref: this.ref,
        ...classicProps
      },
      /* @__PURE__ */ import_react.default.createElement(import_antd.Row, { ...rowProps }, this.getChildren())
    );
  }
  getChildren() {
    const { children } = this.props;
    if (!children)
      return null;
    return import_react.Children.map(children, (child) => {
      if (!child)
        return void 0;
      const { className: prevClassName = "" } = (child == null ? void 0 : child.props) || {};
      const className = (0, import_classnames.default)(prevClassName, {});
      return (0, import_react.cloneElement)(child, {
        className
      });
    });
  }
  /** Appends the `sortable` property to this component */
  get sortable() {
    const el = this.wrapEl;
    if (!el)
      return null;
    const key = Object.keys(el).find((k) => k.includes("Sortable"));
    if (!key)
      return null;
    return el[key];
  }
  /** Converts all the props from `ReactSortable` into the `options` object that `Sortable.create(el, [options])` can use. */
  makeOptions() {
    const DOMHandlers = [
      "onAdd",
      "onChoose",
      "onDeselect",
      "onEnd",
      "onRemove",
      "onSelect",
      "onSpill",
      "onStart",
      "onUnchoose",
      "onUpdate"
    ];
    const NonDOMHandlers = [
      "onChange",
      "onClone",
      "onFilter",
      "onSort"
    ];
    const newOptions = (0, import_util.destructurePropsForOptions)(this.props);
    DOMHandlers.forEach(
      (name) => newOptions[name] = this.prepareOnHandlerPropAndDOM(name)
    );
    NonDOMHandlers.forEach(
      (name) => newOptions[name] = this.prepareOnHandlerProp(name)
    );
    const onMove = (evt, originalEvt) => {
      const { onMove: onMove2 } = this.props;
      const defaultValue = evt.willInsertAfter || -1;
      if (!onMove2)
        return defaultValue;
      const result = onMove2(evt, originalEvt, this.sortable, store);
      if (typeof result === "undefined")
        return false;
      return result;
    };
    return {
      ...newOptions,
      onMove
    };
  }
  /** Prepares a method that will be used in the sortable options to call an `on[Handler]` prop & an `on[Handler]` ReactSortable method.  */
  prepareOnHandlerPropAndDOM(evtName) {
    return (evt) => {
      this.callOnHandlerProp(evt, evtName);
      this[evtName](evt);
    };
  }
  /** Prepares a method that will be used in the sortable options to call an `on[Handler]` prop */
  prepareOnHandlerProp(evtName) {
    return (evt) => {
      this.callOnHandlerProp(evt, evtName);
    };
  }
  /** Calls the `props.on[Handler]` function */
  callOnHandlerProp(evt, evtName) {
    const propEvent = this.props[evtName];
    if (propEvent)
      propEvent(evt, this.sortable, store);
  }
  // SORTABLE DOM HANDLING
  onAdd(evt) {
    const otherList = [...store.dragging.props.list];
    const customs = (0, import_util.createCustoms)(evt, otherList);
    (0, import_util.removeNodes)(customs);
  }
  onRemove(evt) {
    const { list } = this.props;
    const customs = (0, import_util.createCustoms)(evt, list);
    (0, import_util.insertNodes)(customs);
    let newList = [...list];
    if (evt.pullMode !== "clone")
      newList = (0, import_util.handleStateRemove)(customs, newList);
    else {
      (0, import_util.removeNodes)(customs);
      customs.forEach((curr) => {
        const index = curr.oldIndex;
        const newItem = this.props.clone(curr.item, evt);
        newList.splice(index, 1, newItem);
      });
    }
    newList = newList.map(
      (item) => Object.assign({}, item, {
        selected: false
      })
    );
  }
  onUpdate(evt) {
    const { list } = this.props;
    const customs = (0, import_util.createCustoms)(evt, list);
    (0, import_util.removeNodes)(customs);
    (0, import_util.insertNodes)(customs);
    (0, import_util.handleStateChanges)(customs, list);
  }
  onStart() {
    store.dragging = this;
  }
  onEnd() {
    store.dragging = null;
  }
  onChoose() {
  }
  onUnchoose() {
  }
  onSpill(evt) {
    const { removeOnSpill, revertOnSpill } = this.props;
    if (removeOnSpill && !revertOnSpill)
      (0, import_util.removeNode)(evt.item);
  }
  onSelect(evt) {
    const { list } = this.props;
    const newList = list.map(
      (item) => Object.assign({}, item, {
        selected: false
      })
    );
    evt.newIndicies.forEach((curr) => {
      const index = curr.index;
      if (index === -1) {
        console.log(
          `"${evt.type}" had indice of "${curr.index}", which is probably -1 and doesn't usually happen here.`
        );
        console.log(evt);
        return;
      }
      newList[index].selected = true;
    });
  }
  onDeselect(evt) {
    const { list } = this.props;
    const newList = list.map(
      (item) => Object.assign({}, item, {
        selected: false
      })
    );
    evt.newIndicies.forEach((curr) => {
      const index = curr.index;
      if (index === -1)
        return;
      newList[index].selected = true;
    });
  }
};
ReactSortable.defaultProps = {
  clone: (item) => item
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ReactSortable
});
