// @roddan/ui/lib/react-sortable/table-sortable.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/react-sortable/table-sortable.tsx
var table_sortable_exports = {};
__export(table_sortable_exports, {
  TableSortable: () => TableSortable
});
module.exports = __toCommonJS(table_sortable_exports);
var import_react = __toESM(require("react"));
var import_sortablejs = __toESM(require("sortablejs"));
var import_antd = require("antd");
var import_lodash_es = require("lodash-es");
var import_icons = require("@ant-design/icons");
var import_array_move = require("array-move");
var TableSortable = ({ onSort, ...props }) => {
  const tableRef = (0, import_react.useRef)();
  const dataSourceRef = (0, import_react.useRef)();
  dataSourceRef.current = (0, import_lodash_es.cloneDeep)(props.dataSource);
  (0, import_react.useEffect)(() => {
    if (!tableRef.current)
      return;
    const rowEl = tableRef.current.querySelector(".ant-table-tbody");
    const sortIns = new import_sortablejs.default(rowEl, {
      animation: 150,
      // group: 'list',
      handle: ".draggble-btn",
      onSort: function(e) {
        const { newIndex, oldIndex } = e;
        console.log("newIndex", newIndex);
        console.log("oldIndex", oldIndex);
        const newValueOptions = (0, import_array_move.arrayMoveImmutable)(
          dataSourceRef.current || [],
          oldIndex - 1,
          newIndex - 1
        );
        onSort(newValueOptions);
      }
    });
    return () => {
    };
  }, [tableRef.current]);
  const columns = [
    {
      key: "sort",
      align: "center",
      width: 40,
      render: () => {
        return /* @__PURE__ */ import_react.default.createElement(
          import_antd.Button,
          {
            className: "draggble-btn",
            type: "text",
            size: "small",
            icon: /* @__PURE__ */ import_react.default.createElement(import_icons.MenuOutlined, null),
            style: { cursor: "move" }
          }
        );
      }
    },
    ...props.columns
  ];
  return /* @__PURE__ */ import_react.default.createElement("div", { ref: tableRef }, /* @__PURE__ */ import_react.default.createElement(import_antd.Table, { ...props, columns }));
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  TableSortable
});
