// @roddan/ui/lib/react-sortable/util.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/react-sortable/util.ts
var util_exports = {};
__export(util_exports, {
  createCustoms: () => createCustoms,
  createNormalized: () => createNormalized,
  destructurePropsForOptions: () => destructurePropsForOptions,
  handleDOMChanges: () => handleDOMChanges,
  handleStateAdd: () => handleStateAdd,
  handleStateChanges: () => handleStateChanges,
  handleStateRemove: () => handleStateRemove,
  insertNodeAt: () => insertNodeAt,
  insertNodes: () => insertNodes,
  removeNode: () => removeNode,
  removeNodes: () => removeNodes
});
module.exports = __toCommonJS(util_exports);
function removeNode(node) {
  if (node.parentElement !== null)
    node.parentElement.removeChild(node);
}
function insertNodeAt(parent, newChild, index) {
  const refChild = parent.children[index] || null;
  parent.insertBefore(newChild, refChild);
}
function removeNodes(customs) {
  customs.forEach((curr) => removeNode(curr.element));
}
function insertNodes(customs) {
  customs.forEach((curr) => {
    insertNodeAt(curr.parentElement, curr.element, curr.oldIndex);
  });
}
function handleDOMChanges(customs) {
  removeNodes(customs);
  insertNodes(customs);
}
function createCustoms(evt, list) {
  const custom = [
    {
      element: evt.item,
      newIndex: evt.newIndex,
      oldIndex: evt.oldIndex,
      parentElement: evt.from
    }
  ];
  const customs = createNormalized(custom, list);
  return customs;
}
function handleStateChanges(normalized, list) {
  const a = handleStateRemove(normalized, list);
  const b = handleStateAdd(normalized, a);
  return b;
}
function handleStateRemove(normalized, list) {
  const newList = [...list];
  normalized.concat().reverse().forEach((curr) => newList.splice(curr.oldIndex, 1));
  return newList;
}
function handleStateAdd(normalized, list, evt, clone) {
  const newList = [...list];
  normalized.forEach((curr) => {
    const newItem = clone && evt && clone(curr.item, evt);
    newList.splice(curr.newIndex, 0, newItem || curr.item);
  });
  return newList;
}
function createNormalized(inputs, list) {
  const normalized = inputs.map((curr) => ({ ...curr, item: list[curr.oldIndex] })).sort((a, b) => a.oldIndex - b.oldIndex);
  return normalized;
}
function destructurePropsForOptions(props) {
  const {
    // react sortable props
    list,
    children,
    tag,
    style,
    className,
    clone,
    // sortable options that have methods we want to overwrite
    onAdd,
    onChange,
    onChoose,
    onClone,
    onEnd,
    onFilter,
    onRemove,
    onSort,
    onStart,
    onUnchoose,
    onUpdate,
    onMove,
    onSpill,
    onSelect,
    onDeselect,
    rowProps,
    ...options
  } = props;
  return options;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createCustoms,
  createNormalized,
  destructurePropsForOptions,
  handleDOMChanges,
  handleStateAdd,
  handleStateChanges,
  handleStateRemove,
  insertNodeAt,
  insertNodes,
  removeNode,
  removeNodes
});
