// @roddan/ui/es/react-sortable/react-sortable.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }
function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }
function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); Object.defineProperty(subClass, "prototype", { writable: false }); if (superClass) _setPrototypeOf(subClass, superClass); }
function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }
function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } else if (call !== void 0) { throw new TypeError("Derived constructors may only return object or undefined"); } return _assertThisInitialized(self); }
function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import { Row } from 'antd';
import classNames from 'classnames';
import React, { Children, cloneElement, Component, createElement, createRef } from 'react';
import Sortable from 'sortablejs';
import { createCustoms, destructurePropsForOptions, handleStateChanges, handleStateRemove, insertNodes, removeNode, removeNodes } from "./util";

/** Holds a global reference for which react element is being dragged */
var store = {
  dragging: null
};
export var ReactSortable = /*#__PURE__*/function (_Component) {
  _inherits(ReactSortable, _Component);
  var _super = _createSuper(ReactSortable);
  function ReactSortable(props) {
    var _this;
    _classCallCheck(this, ReactSortable);
    _this = _super.call(this, props);
    _defineProperty(_assertThisInitialized(_this), "wrapEl", void 0);
    _defineProperty(_assertThisInitialized(_this), "ref", void 0);
    _this.ref = /*#__PURE__*/createRef();
    return _this;
  }
  _createClass(ReactSortable, [{
    key: "isRow",
    value: function isRow() {
      return !!this.props.rowProps;
    }
  }, {
    key: "componentDidMount",
    value: function componentDidMount() {
      if (this.ref.current === null || this.props.forbidden) return;
      var newOptions = this.makeOptions();
      var sortableClass = this.props.sortableClass;
      this.wrapEl = this.isRow() ? this.ref.current.querySelector(sortableClass !== null && sortableClass !== void 0 ? sortableClass : '.ant-row') : this.ref.current;
      if (!this.wrapEl) return;
      Sortable.create(this.wrapEl, newOptions);
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      if (prevProps.disabled !== this.props.disabled && this.sortable) {
        this.sortable.option('disabled', this.props.disabled);
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
        tag = _this$props.tag,
        style = _this$props.style,
        className = _this$props.className,
        id = _this$props.id,
        rowProps = _this$props.rowProps;
      var classicProps = {
        style: style,
        className: className,
        id: id
      };

      // if no tag, default to a `div` element.
      var newTag = !tag || tag === null ? 'div' : tag;
      if (!this.isRow()) {
        return /*#__PURE__*/createElement(newTag, _objectSpread({
          ref: this.ref
        }, classicProps), this.getChildren());
      }
      return /*#__PURE__*/createElement(newTag, _objectSpread({
        ref: this.ref
      }, classicProps), /*#__PURE__*/React.createElement(Row, rowProps, this.getChildren()));
    }
  }, {
    key: "getChildren",
    value: function getChildren() {
      var children = this.props.children;

      // if no children, don't do anything.
      if (!children) return null;
      return Children.map(children, function (child) {
        if (!child) return undefined;
        var _ref = (child === null || child === void 0 ? void 0 : child.props) || {},
          _ref$className = _ref.className,
          prevClassName = _ref$className === void 0 ? '' : _ref$className;
        var className = classNames(prevClassName, {});
        return /*#__PURE__*/cloneElement(child, {
          className: className
        });
      });
    }

    /** Appends the `sortable` property to this component */
  }, {
    key: "sortable",
    get: function get() {
      var el = this.wrapEl;
      if (!el) return null;
      var key = Object.keys(el).find(function (k) {
        return k.includes('Sortable');
      });
      if (!key) return null;
      return el[key];
    }

    /** Converts all the props from `ReactSortable` into the `options` object that `Sortable.create(el, [options])` can use. */
  }, {
    key: "makeOptions",
    value: function makeOptions() {
      var _this2 = this;
      var DOMHandlers = ['onAdd', 'onChoose', 'onDeselect', 'onEnd', 'onRemove', 'onSelect', 'onSpill', 'onStart', 'onUnchoose', 'onUpdate'];
      var NonDOMHandlers = ['onChange', 'onClone', 'onFilter', 'onSort'];
      var newOptions = destructurePropsForOptions(this.props);
      DOMHandlers.forEach(function (name) {
        return newOptions[name] = _this2.prepareOnHandlerPropAndDOM(name);
      });
      NonDOMHandlers.forEach(function (name) {
        return newOptions[name] = _this2.prepareOnHandlerProp(name);
      });

      /** onMove has 2 arguments and needs to be handled seperately. */
      var onMove = function onMove(evt, originalEvt) {
        var onMove = _this2.props.onMove;
        var defaultValue = evt.willInsertAfter || -1;
        if (!onMove) return defaultValue;
        var result = onMove(evt, originalEvt, _this2.sortable, store);
        if (typeof result === 'undefined') return false;
        return result;
      };
      return _objectSpread(_objectSpread({}, newOptions), {}, {
        onMove: onMove
      });
    }

    /** Prepares a method that will be used in the sortable options to call an `on[Handler]` prop & an `on[Handler]` ReactSortable method.  */
  }, {
    key: "prepareOnHandlerPropAndDOM",
    value: function prepareOnHandlerPropAndDOM(evtName) {
      var _this3 = this;
      return function (evt) {
        // call the component prop
        _this3.callOnHandlerProp(evt, evtName);
        // calls state change
        _this3[evtName](evt);
      };
    }

    /** Prepares a method that will be used in the sortable options to call an `on[Handler]` prop */
  }, {
    key: "prepareOnHandlerProp",
    value: function prepareOnHandlerProp(evtName) {
      var _this4 = this;
      return function (evt) {
        // call the component prop
        _this4.callOnHandlerProp(evt, evtName);
      };
    }

    /** Calls the `props.on[Handler]` function */
  }, {
    key: "callOnHandlerProp",
    value: function callOnHandlerProp(evt, evtName) {
      var propEvent = this.props[evtName];
      if (propEvent) propEvent(evt, this.sortable, store);
    }

    // SORTABLE DOM HANDLING
  }, {
    key: "onAdd",
    value: function onAdd(evt) {
      var otherList = _toConsumableArray(store.dragging.props.list);
      var customs = createCustoms(evt, otherList);
      removeNodes(customs);
    }
  }, {
    key: "onRemove",
    value: function onRemove(evt) {
      var _this5 = this;
      var list = this.props.list;
      // const mode = getMode(evt);
      var customs = createCustoms(evt, list);
      insertNodes(customs);
      var newList = _toConsumableArray(list);
      // remove state if not in clone mode. otherwise, keep.
      if (evt.pullMode !== 'clone') newList = handleStateRemove(customs, newList);
      // if clone, it doesn't really remove. instead it clones in place.
      else {
        removeNodes(customs);
        // replace selected items with cloned items
        customs.forEach(function (curr) {
          var index = curr.oldIndex;
          var newItem = _this5.props.clone(curr.item, evt);
          newList.splice(index, 1, newItem);
        });
      }

      // remove item.selected from list
      newList = newList.map(function (item) {
        return Object.assign({}, item, {
          selected: false
        });
      });
    }
  }, {
    key: "onUpdate",
    value: function onUpdate(evt) {
      var list = this.props.list;
      var customs = createCustoms(evt, list);
      removeNodes(customs);
      insertNodes(customs);
      handleStateChanges(customs, list);
    }
  }, {
    key: "onStart",
    value: function onStart() {
      store.dragging = this;
    }
  }, {
    key: "onEnd",
    value: function onEnd() {
      store.dragging = null;
    }
  }, {
    key: "onChoose",
    value: function onChoose() {}
  }, {
    key: "onUnchoose",
    value: function onUnchoose() {}
  }, {
    key: "onSpill",
    value: function onSpill(evt) {
      var _this$props2 = this.props,
        removeOnSpill = _this$props2.removeOnSpill,
        revertOnSpill = _this$props2.revertOnSpill;
      if (removeOnSpill && !revertOnSpill) removeNode(evt.item);
    }
  }, {
    key: "onSelect",
    value: function onSelect(evt) {
      var list = this.props.list;
      var newList = list.map(function (item) {
        return Object.assign({}, item, {
          selected: false
        });
      });
      evt.newIndicies.forEach(function (curr) {
        var index = curr.index;
        if (index === -1) {
          console.log("\"".concat(evt.type, "\" had indice of \"").concat(curr.index, "\", which is probably -1 and doesn't usually happen here."));
          console.log(evt);
          return;
        }
        newList[index].selected = true;
      });
    }
  }, {
    key: "onDeselect",
    value: function onDeselect(evt) {
      var list = this.props.list;
      var newList = list.map(function (item) {
        return Object.assign({}, item, {
          selected: false
        });
      });
      evt.newIndicies.forEach(function (curr) {
        var index = curr.index;
        if (index === -1) return;
        newList[index].selected = true;
      });
    }
  }]);
  return ReactSortable;
}(Component);

// interface MultiIndices {
//   multiDragElement: HTMLElement;
//   index: number;
// }

// export interface SortableEvent extends SortableEvent {
//   clones: HTMLElement[];
//   oldIndicies: MultiIndices[];
//   newIndicies: MultiIndices[];
//   swapItem: HTMLElement | null;
// }
_defineProperty(ReactSortable, "defaultProps", {
  clone: function clone(item) {
    return item;
  }
});
