// @roddan/ui/es/react-sortable/util.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
var _excluded = ["list", "children", "tag", "style", "className", "clone", "onAdd", "onChange", "onChoose", "onClone", "onEnd", "onFilter", "onRemove", "onSort", "onStart", "onUnchoose", "onUpdate", "onMove", "onSpill", "onSelect", "onDeselect", "rowProps"];
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }
function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }
function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
/**
 * Removes the `node` from the DOM
 * @param node
 */
export function removeNode(node) {
  if (node.parentElement !== null) node.parentElement.removeChild(node);
}

/**
 * Inserts the `newChild` node at the given index in a parent
 * @param parent The parent HTML Element.
 * @param newChild A HTML eement to add as a child of the parent.
 * @param index index of the parent to place the new child in.
 */
export function insertNodeAt(parent, newChild, index) {
  var refChild = parent.children[index] || null;
  parent.insertBefore(newChild, refChild);
}
export function removeNodes(customs) {
  customs.forEach(function (curr) {
    return removeNode(curr.element);
  });
}
export function insertNodes(customs) {
  customs.forEach(function (curr) {
    insertNodeAt(curr.parentElement, curr.element, curr.oldIndex);
  });
}

/** removes stuff from the dom in a nice order */
export function handleDOMChanges(customs) {
  removeNodes(customs);
  insertNodes(customs);
}
export function createCustoms(evt, list) {
  var custom = [{
    element: evt.item,
    newIndex: evt.newIndex,
    oldIndex: evt.oldIndex,
    parentElement: evt.from
  }];

  /* eslint-enable */

  var customs = createNormalized(custom, list);
  return customs;
}

/** moves items form old index to new index without breaking anything ideally. */
export function handleStateChanges(normalized, list) {
  var a = handleStateRemove(normalized, list);
  var b = handleStateAdd(normalized, a);
  return b;
}
export function handleStateRemove(normalized, list) {
  var newList = _toConsumableArray(list);
  normalized.concat().reverse().forEach(function (curr) {
    return newList.splice(curr.oldIndex, 1);
  });
  return newList;
}
export function handleStateAdd(normalized, list, evt, clone) {
  var newList = _toConsumableArray(list);
  normalized.forEach(function (curr) {
    var newItem = clone && evt && clone(curr.item, evt);
    newList.splice(curr.newIndex, 0, newItem || curr.item);
  });
  return newList;
}
export function createNormalized(inputs, list) {
  var normalized = inputs.map(function (curr) {
    return _objectSpread(_objectSpread({}, curr), {}, {
      item: list[curr.oldIndex]
    });
  }).sort(function (a, b) {
    return a.oldIndex - b.oldIndex;
  });
  return normalized;
}
/**
 * Removes the following group of properties from `props`,
 * leaving only `Sortable.Options` without any `on` methods.
 * @param props `ReactSortable.Props`
 */
export function destructurePropsForOptions(props) {
  /* eslint-disable */
  var list = props.list,
    children = props.children,
    tag = props.tag,
    style = props.style,
    className = props.className,
    clone = props.clone,
    onAdd = props.onAdd,
    onChange = props.onChange,
    onChoose = props.onChoose,
    onClone = props.onClone,
    onEnd = props.onEnd,
    onFilter = props.onFilter,
    onRemove = props.onRemove,
    onSort = props.onSort,
    onStart = props.onStart,
    onUnchoose = props.onUnchoose,
    onUpdate = props.onUpdate,
    onMove = props.onMove,
    onSpill = props.onSpill,
    onSelect = props.onSelect,
    onDeselect = props.onDeselect,
    rowProps = props.rowProps,
    options = _objectWithoutProperties(props, _excluded);
  /* eslint-enable */
  return options;
}

/**
 * Construct a type with the properties of T except for those in type K.
 * Including this allows for backwards compatibility with earlier versions of TS.
 */
