// @roddan/form-editor/lib/components/setting-common/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SettingWrap = exports.SettingItem = void 0;
var _react = _interopRequireDefault(require("react"));
var _antd = require("antd");
var _styledComponents = _interopRequireDefault(require("styled-components"));
var _icons = require("@ant-design/icons");
var _const = require("../../const");
require("./style.less");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var Flex = _styledComponents.default.div.withConfig({
  displayName: "Flex",
  componentId: "form-editor-42cd__sc-1iov5q4-0"
})(function (_ref) {
  var vertical = _ref.vertical;
  if (!vertical) return '';
  return "\n      flex-direction: column;\n      align-items: start;\n      .".concat((0, _const.prefixCls)('setting-item-label'), " {\n        margin-bottom: 8px;\n      }\n    ");
});
var SettingItem = exports.SettingItem = function SettingItem(_ref2) {
  var label = _ref2.label,
    children = _ref2.children,
    style = _ref2.style,
    tips = _ref2.tips,
    vertical = _ref2.vertical;
  return /*#__PURE__*/_react.default.createElement(Flex, {
    style: style || {},
    className: (0, _const.prefixCls)('setting-item'),
    vertical: vertical
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _const.prefixCls)('setting-item-label')
  }, label, "\xA0", tips && /*#__PURE__*/_react.default.createElement(_antd.Popover, {
    content: tips
  }, /*#__PURE__*/_react.default.createElement(_icons.QuestionCircleOutlined, {
    style: {
      cursor: 'pointer'
    }
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _const.prefixCls)('setting-item-value')
  }, children));
};
var SettingWrap = exports.SettingWrap = function SettingWrap(_ref3) {
  var title = _ref3.title,
    children = _ref3.children,
    style = _ref3.style;
  return /*#__PURE__*/_react.default.createElement("div", {
    style: style || {},
    className: (0, _const.prefixCls)('setting-wrap')
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _const.prefixCls)('setting-title')
  }, title), children);
};
