// @roddan/form-editor/lib/elements/checkbox/render-checkbox.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RenderCheckbox = void 0;
var _react = _interopRequireDefault(require("react"));
var _antd = require("antd");
var _mobxReactLite = require("mobx-react-lite");
var _components = require("../../components");
var _hooks = require("../../hooks");
var _types = require("../../types");
var _store = _interopRequireDefault(require("../../store"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var RenderCheckboxContent = function RenderCheckboxContent(_ref) {
  var fieldValue = _ref.fieldValue,
    element = _ref.element;
  var id = element.id,
    valueOptions = element.valueOptions,
    alignDirection = element.alignDirection;
  var _useRegisterEvents = (0, _hooks.useRegisterEvents)(element),
    eventFunctions = _useRegisterEvents.eventFunctions;
  var handleChange = function handleChange(val) {
    _store.default.setFieldValue(id, val);
  };
  (0, _hooks.useFormUpdate)(function () {
    var _eventFunctions$EEven;
    (_eventFunctions$EEven = eventFunctions[_types.EEventAction.ON_LOADED]) === null || _eventFunctions$EEven === void 0 || _eventFunctions$EEven.call(eventFunctions);
  }, [eventFunctions[_types.EEventAction.ON_LOADED]]);
  (0, _hooks.useFormUpdate)(function () {
    var _eventFunctions$EEven2;
    (_eventFunctions$EEven2 = eventFunctions[_types.EEventAction.VALUE_CHANGE]) === null || _eventFunctions$EEven2 === void 0 || _eventFunctions$EEven2.call(eventFunctions, fieldValue);
  }, [fieldValue]);
  return /*#__PURE__*/_react.default.createElement(_components.ElementLayout, {
    element: element
  }, /*#__PURE__*/_react.default.createElement(_antd.Checkbox.Group, {
    onChange: handleChange,
    value: fieldValue
  }, /*#__PURE__*/_react.default.createElement(_antd.Space, {
    direction: alignDirection
  }, valueOptions === null || valueOptions === void 0 ? void 0 : valueOptions.map(function (opt) {
    return /*#__PURE__*/_react.default.createElement(_antd.Checkbox, {
      key: opt.id,
      value: opt.value
    }, opt.label);
  }))));
};

// store.setAllElementsList(ELEMENT_INPUT, 'render', RenderInput)
var RenderCheckbox = exports.RenderCheckbox = (0, _mobxReactLite.observer)(RenderCheckboxContent);
