// @roddan/form-editor/es/elements/table/render-table.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import React from 'react';
import { Table } from 'antd';
import { observer } from 'mobx-react-lite';
import { ElementLayout } from "../../components";
import { EEventAction } from "../../types";
import { parseJSX } from "../../utils";
import { useFormUpdate, useRegisterEvents } from "../../hooks";
var RenderTableContent = function RenderTableContent(_ref) {
  var _ref$fieldValue = _ref.fieldValue,
    fieldValue = _ref$fieldValue === void 0 ? [] : _ref$fieldValue,
    element = _ref.element;
  var _element$tableColumns = element.tableColumns,
    tableColumns = _element$tableColumns === void 0 ? '[]' : _element$tableColumns,
    _element$tableAttribu = element.tableAttributes,
    tableAttributes = _element$tableAttribu === void 0 ? '{}' : _element$tableAttribu;
  var _useRegisterEvents = useRegisterEvents(element),
    eventFunctions = _useRegisterEvents.eventFunctions;
  useFormUpdate(function () {
    var _eventFunctions$EEven;
    (_eventFunctions$EEven = eventFunctions[EEventAction.ON_LOADED]) === null || _eventFunctions$EEven === void 0 || _eventFunctions$EEven.call(eventFunctions);
  }, [eventFunctions[EEventAction.ON_LOADED]]);
  return /*#__PURE__*/React.createElement(ElementLayout, {
    element: element
  }, /*#__PURE__*/React.createElement(Table, _extends({
    columns: parseJSX(tableColumns),
    rowKey: "id",
    dataSource: fieldValue
  }, parseJSX("[".concat(tableAttributes, "]"))[0])));
};
export var RenderTable = observer(RenderTableContent);
