// @roddan/form-editor/es/elements/table/table-setting.js
import { QuestionCircleOutlined } from '@ant-design/icons';
import { Button, Popover } from 'antd';
import { observer } from 'mobx-react-lite';
import React from 'react';
import { AttributesSetting } from "../../components";
import { SettingItem, SettingWrap } from "../../components";
import store from "../../store";
var SettingTableContent = function SettingTableContent() {
  return /*#__PURE__*/React.createElement(SettingWrap, {
    title: "\u5143\u7D20\u8BBE\u7F6E"
  }, /*#__PURE__*/React.createElement(SettingItem, {
    tips: /*#__PURE__*/React.createElement("a", {
      target: "_blank",
      href: "https://ant.design/components/table-cn#table",
      rel: "noreferrer"
    }, "antd - table\u6587\u6863"),
    label: "\u8868\u683C\u914D\u7F6E"
  }, /*#__PURE__*/React.createElement(AttributesSetting, {
    editorType: "json",
    value: store.selectedElement.tableAttributes,
    title: "\u8868\u683C\u914D\u7F6E",
    onChange: function onChange(val) {
      store.setSelectedProp('tableAttributes', val);
    }
  }, /*#__PURE__*/React.createElement(Button, {
    className: "fe-attr-setting-btn",
    size: "small"
  }, "\u7F16\u8F91"))), /*#__PURE__*/React.createElement(SettingItem, {
    tips: /*#__PURE__*/React.createElement("a", {
      target: "_blank",
      href: "https://ant.design/components/table-cn#column",
      rel: "noreferrer"
    }, "table - column\u6587\u6863"),
    label: "\u5217\u8868\u914D\u7F6E"
  }, /*#__PURE__*/React.createElement(AttributesSetting, {
    editorType: "javascript",
    value: store.selectedElement.tableColumns,
    title: /*#__PURE__*/React.createElement(React.Fragment, null, "\u5217\u8868\u9879\u8BBE\u7F6E\xA0", /*#__PURE__*/React.createElement(Popover, {
      content: /*#__PURE__*/React.createElement(React.Fragment, null, "render\u51FD\u6570\u4E2D\u53EF\u76F4\u63A5\u4F7F\u7528antd\u7EC4\u4EF6\u53CAAntdIcons.XXX\u7684\u65B9\u5F0F\u4F7F\u7528icon\u56FE\u6807")
    }, /*#__PURE__*/React.createElement(QuestionCircleOutlined, {
      style: {
        cursor: 'pointer'
      }
    }))),
    onChange: function onChange(val) {
      store.setSelectedProp('tableColumns', val);
    }
  }, /*#__PURE__*/React.createElement(Button, {
    className: "fe-attr-setting-btn",
    size: "small"
  }, "\u7F16\u8F91"))));
};
export var SettingTable = observer(SettingTableContent);
