// @roddan/form-editor/es/elements/date/render-date.js
import React, { useMemo } from 'react';
import { DatePicker } from 'antd';
import dayjs from 'dayjs';
import { observer } from 'mobx-react-lite';
import { useRegisterEvents, useFormUpdate } from "../../hooks";
import { formatDate } from "../../utils";
import { EEventAction } from "../../types";
import { ElementLayout } from "../../components";
import store from "../../store";
var RenderDateContent = function RenderDateContent(_ref) {
  var fieldValue = _ref.fieldValue,
    element = _ref.element;
  var id = element.id,
    dateFormat = element.dateFormat,
    placeholder = element.placeholder;
  var _useRegisterEvents = useRegisterEvents(element),
    eventFunctions = _useRegisterEvents.eventFunctions;
  var handleEvent = function handleEvent(action) {
    return function (e) {
      var _eventFunctions$actio;
      (_eventFunctions$actio = eventFunctions[action]) === null || _eventFunctions$actio === void 0 || _eventFunctions$actio.call(eventFunctions, e.target.value);
    };
  };
  var handleChange = function handleChange(date) {
    store.setFieldValue(id, date ? formatDate(date, dateFormat) : undefined);
  };
  useFormUpdate(function () {
    var _eventFunctions$EEven;
    (_eventFunctions$EEven = eventFunctions[EEventAction.ON_LOADED]) === null || _eventFunctions$EEven === void 0 || _eventFunctions$EEven.call(eventFunctions);
  }, [eventFunctions[EEventAction.ON_LOADED]]);
  useFormUpdate(function () {
    var _eventFunctions$EEven2;
    (_eventFunctions$EEven2 = eventFunctions[EEventAction.VALUE_CHANGE]) === null || _eventFunctions$EEven2 === void 0 || _eventFunctions$EEven2.call(eventFunctions, fieldValue);
  }, [fieldValue]);
  var showTime = useMemo(function () {
    var timeFormat = dateFormat === null || dateFormat === void 0 ? void 0 : dateFormat.split(' ');
    if (!timeFormat) return false;
    var res = {};
    if (timeFormat.includes('HH')) res.showHour = true;
    if (timeFormat.includes('mm')) res.showMinute = true;
    if (timeFormat.includes('ss')) res.showSecond = true;
    return res;
  }, [dateFormat]);
  return /*#__PURE__*/React.createElement(ElementLayout, {
    element: element
  }, /*#__PURE__*/React.createElement(DatePicker, {
    format: dateFormat
    // @ts-ignore
    ,
    value: fieldValue ? dayjs(fieldValue) : undefined,
    showTime: showTime,
    getPopupContainer: function getPopupContainer(n) {
      return n.parentElement;
    },
    placement: "bottomRight",
    onChange: handleChange,
    onFocus: handleEvent(EEventAction.ON_FOCUS),
    onBlur: handleEvent(EEventAction.ON_BLUR),
    placeholder: placeholder
  }));
};

// store.setAllElementsList(ELEMENT_INPUT, 'render', RenderInput)
export var RenderDate = observer(RenderDateContent);
