// @roddan/form-editor/lib/views/form-canvas/render-canvas.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _hooks = require("../../hooks");
var _types = require("../../types");
var _const = require("../../const");
var _export = require("../../elements/export");
var _eventStore = _interopRequireDefault(require("../../store/eventStore"));
var _store = _interopRequireDefault(require("../../store"));
var _utils = require("../../utils");
var _ui = require("@roddan/ui");
require("./style.less");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
// import { ReactSortable } from '@/components/react-sortable';

var EditorCanvas = function EditorCanvas(_ref) {
  var mode = _ref.mode,
    actions = _ref.actions;
  var _store$formAttrs = _store.default.formAttrs,
    horizontalGap = _store$formAttrs.horizontalGap,
    verticalGap = _store$formAttrs.verticalGap,
    id = _store$formAttrs.id,
    events = _store$formAttrs.events,
    customCss = _store$formAttrs.customCss;
  var _useRegisterEvents = (0, _hooks.useRegisterEvents)({
      id: id,
      events: events
    }),
    eventFunctions = _useRegisterEvents.eventFunctions;
  var formStyle = (0, _react.useMemo)(function () {
    if (!customCss) return {};
    var cssObj = (0, _utils.parseCSS)(customCss);
    return Object.values(cssObj)[0];
  }, [customCss]);
  (0, _hooks.useDesignEffect)(function () {
    var keydonwFn = function keydonwFn(e) {
      if (e.key === 'Backspace' && document.activeElement === document.body) {
        _store.default.deleteEl(_store.default.selectedElement);
      }
    };
    document.addEventListener('keydown', keydonwFn);
    return function () {
      document.removeEventListener('keydown', keydonwFn);
    };
  });

  // 表单加载事件
  (0, _hooks.useFormUpdate)(function () {
    var _eventFunctions$EEven;
    (_eventFunctions$EEven = eventFunctions[_types.EEventAction.FORM_LOADED]) === null || _eventFunctions$EEven === void 0 || _eventFunctions$EEven.call(eventFunctions);
  }, [eventFunctions[_types.EEventAction.FORM_LOADED]]);

  // 服务监听事件 - ps:不用关心设计模式下的运行
  (0, _hooks.useFormUpdate)(function () {
    if (!_store.default.formServices.length) return;
    _store.default.formServices.forEach(function (serv) {
      _eventStore.default.emitter.on(serv.id, _utils.handleOnEvent);
    });
    return function () {
      _store.default.formServices.forEach(function (serv) {
        _eventStore.default.emitter.off(serv.id, _utils.handleOnEvent);
      });
    };
  }, [_store.default.formServices]);
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _const.prefixCls)('canvas-wrap')
  }, actions && /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, actions), /*#__PURE__*/_react.default.createElement(_ui.ReactSortable, {
    className: (0, _const.prefixCls)('canvas'),
    style: formStyle,
    list: _store.default.formElements,
    onSort: function onSort(e) {
      return (0, _utils.handleSort)(e, id);
    },
    animation: 150,
    group: "nested",
    rowProps: {
      className: (0, _const.prefixCls)('row'),
      gutter: [horizontalGap, verticalGap],
      'data-id': id
    },
    forbidden: mode !== 'design'
  }, _store.default.formElements.map(function (item) {
    var _ElementsList;
    var Component = (_ElementsList = _export.ElementsList[item.type]) === null || _ElementsList === void 0 ? void 0 : _ElementsList.render;
    if (!Component) return null;
    _store.default.flatElement(item);
    return /*#__PURE__*/_react.default.createElement(Component, {
      key: item.id,
      fieldValue: _store.default.fieldValues[item.id],
      element: item
    });
  })));
};
var _default = exports.default = (0, _mobxReactLite.observer)(EditorCanvas);
