// @roddan/ui/es/monaco-editor/Editor/index.js
'use client';

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import loader from '@monaco-editor/loader';
import * as apiMonaco from 'monaco-editor/esm/vs/editor/editor.api';
import React, { memo, useCallback, useEffect, useRef, useState } from 'react';
import useMount from "../hooks/useMount";
import usePrevious from "../hooks/usePrevious";
import useUpdate from "../hooks/useUpdate";
import MonacoContainer from "../MonacoContainer";
import { getOrCreateModel, noop } from "../utils";
loader.config({
  paths: {
    vs: 'https://cdn.jsdelivr.net/gh/lawxy/workers@2.0/vs'
  }
});
export * from "./types";
var viewStates = new Map();
export var MonacoEditor = /*#__PURE__*/memo(function (_ref) {
  var defaultValue = _ref.defaultValue,
    defaultLanguage = _ref.defaultLanguage,
    defaultPath = _ref.defaultPath,
    value = _ref.value,
    language = _ref.language,
    path = _ref.path,
    _ref$theme = _ref.theme,
    theme = _ref$theme === void 0 ? 'light' : _ref$theme,
    line = _ref.line,
    _ref$loading = _ref.loading,
    loading = _ref$loading === void 0 ? 'Loading...' : _ref$loading,
    _ref$options = _ref.options,
    options = _ref$options === void 0 ? {} : _ref$options,
    _ref$overrideServices = _ref.overrideServices,
    overrideServices = _ref$overrideServices === void 0 ? {} : _ref$overrideServices,
    _ref$saveViewState = _ref.saveViewState,
    saveViewState = _ref$saveViewState === void 0 ? true : _ref$saveViewState,
    _ref$keepCurrentModel = _ref.keepCurrentModel,
    keepCurrentModel = _ref$keepCurrentModel === void 0 ? false : _ref$keepCurrentModel,
    _ref$style = _ref.style,
    style = _ref$style === void 0 ? {} : _ref$style,
    className = _ref.className,
    _ref$wrapperProps = _ref.wrapperProps,
    wrapperProps = _ref$wrapperProps === void 0 ? {} : _ref$wrapperProps,
    _ref$beforeMount = _ref.beforeMount,
    beforeMount = _ref$beforeMount === void 0 ? noop : _ref$beforeMount,
    _ref$onMount = _ref.onMount,
    onMount = _ref$onMount === void 0 ? noop : _ref$onMount,
    onChange = _ref.onChange,
    _ref$onValidate = _ref.onValidate,
    onValidate = _ref$onValidate === void 0 ? noop : _ref$onValidate,
    _ref$onFocus = _ref.onFocus,
    onFocus = _ref$onFocus === void 0 ? noop : _ref$onFocus,
    _ref$onBlur = _ref.onBlur,
    onBlur = _ref$onBlur === void 0 ? noop : _ref$onBlur;
  var _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    isEditorReady = _useState2[0],
    setIsEditorReady = _useState2[1];
  var _useState3 = useState(true),
    _useState4 = _slicedToArray(_useState3, 2),
    isMonacoMounting = _useState4[0],
    setIsMonacoMounting = _useState4[1];
  var monacoRef = useRef(null);
  var editorRef = useRef(null);
  var containerRef = useRef(null);
  var onMountRef = useRef(onMount);
  var beforeMountRef = useRef(beforeMount);
  var subscriptionRef = useRef();
  var valueRef = useRef(value);
  var previousPath = usePrevious(path);
  var preventCreation = useRef(false);
  var preventTriggerChangeEvent = useRef(false);
  function disposeEditor() {
    var _subscriptionRef$curr;
    (_subscriptionRef$curr = subscriptionRef.current) === null || _subscriptionRef$curr === void 0 || _subscriptionRef$curr.dispose();
    if (keepCurrentModel) {
      /* eslint-disable */
      saveViewState && viewStates.set(path, editorRef.current.saveViewState());
    } else {
      var _getModel;
      (_getModel = editorRef.current.getModel()) === null || _getModel === void 0 || _getModel.dispose();
    }
    editorRef.current.dispose();
  }
  useMount(function () {
    var cancelable = loader.init();
    cancelable.then(function (monaco) {
      return (monacoRef.current = monaco) && setIsMonacoMounting(false);
    }).catch(function (error) {
      monacoRef.current = apiMonaco;
      setIsMonacoMounting(false);
    });
    return function () {
      return editorRef.current ? disposeEditor() : cancelable.cancel();
    };
  });
  useUpdate(function () {
    var _editorRef$current;
    var model = getOrCreateModel(monacoRef.current, defaultValue || value || '', defaultLanguage || language || '', path || defaultPath || '');
    if (model !== ((_editorRef$current = editorRef.current) === null || _editorRef$current === void 0 ? void 0 : _editorRef$current.getModel())) {
      var _editorRef$current2, _editorRef$current3, _editorRef$current4;
      if (saveViewState) viewStates.set(previousPath, (_editorRef$current2 = editorRef.current) === null || _editorRef$current2 === void 0 ? void 0 : _editorRef$current2.saveViewState());
      (_editorRef$current3 = editorRef.current) === null || _editorRef$current3 === void 0 || _editorRef$current3.setModel(model);
      if (saveViewState) (_editorRef$current4 = editorRef.current) === null || _editorRef$current4 === void 0 || _editorRef$current4.restoreViewState(viewStates.get(path));
    }
  }, [path], isEditorReady);
  useUpdate(function () {
    var _editorRef$current5;
    (_editorRef$current5 = editorRef.current) === null || _editorRef$current5 === void 0 || _editorRef$current5.updateOptions(options);
  }, [options], isEditorReady);
  useUpdate(function () {
    if (!editorRef.current || value === undefined) return;
    if (editorRef.current.getOption(monacoRef.current.editor.EditorOption.readOnly)) {
      editorRef.current.setValue(value);
    } else if (value !== editorRef.current.getValue()) {
      preventTriggerChangeEvent.current = true;
      editorRef.current.executeEdits('', [{
        range: editorRef.current.getModel().getFullModelRange(),
        text: value,
        forceMoveMarkers: true
      }]);
      editorRef.current.pushUndoStop();
      preventTriggerChangeEvent.current = false;
    }
  }, [value], isEditorReady);
  useUpdate(function () {
    var _editorRef$current6, _monacoRef$current;
    var model = (_editorRef$current6 = editorRef.current) === null || _editorRef$current6 === void 0 ? void 0 : _editorRef$current6.getModel();
    if (model && language) (_monacoRef$current = monacoRef.current) === null || _monacoRef$current === void 0 || _monacoRef$current.editor.setModelLanguage(model, language);
  }, [language], isEditorReady);
  useUpdate(function () {
    // reason for undefined check: https://github.com/suren-atoyan/monaco-react/pull/188
    if (line !== undefined) {
      var _editorRef$current7;
      (_editorRef$current7 = editorRef.current) === null || _editorRef$current7 === void 0 || _editorRef$current7.revealLine(line);
    }
  }, [line], isEditorReady);
  useUpdate(function () {
    var _monacoRef$current2;
    (_monacoRef$current2 = monacoRef.current) === null || _monacoRef$current2 === void 0 || _monacoRef$current2.editor.setTheme(theme);
  }, [theme], isEditorReady);
  var createEditor = useCallback(function () {
    if (!containerRef.current || !monacoRef.current) return;
    if (!preventCreation.current) {
      var _monacoRef$current3;
      beforeMountRef.current(monacoRef.current);
      var autoCreatedModelPath = path || defaultPath;
      var defaultModel = getOrCreateModel(monacoRef.current, value || defaultValue || '', defaultLanguage || language || '', autoCreatedModelPath || '');
      editorRef.current = (_monacoRef$current3 = monacoRef.current) === null || _monacoRef$current3 === void 0 ? void 0 : _monacoRef$current3.editor.create(containerRef.current, _objectSpread({
        model: defaultModel,
        automaticLayout: true
      }, options), overrideServices);
      saveViewState && editorRef.current.restoreViewState(viewStates.get(autoCreatedModelPath));
      monacoRef.current.editor.setTheme(theme);
      if (line !== undefined) {
        editorRef.current.revealLine(line);
      }
      setIsEditorReady(true);
      preventCreation.current = true;
    }
  }, [defaultValue, defaultLanguage, defaultPath, value, language, path, options, overrideServices, saveViewState, theme, line]);
  useEffect(function () {
    if (isEditorReady) {
      onMountRef.current(editorRef.current, monacoRef.current);
    }
  }, [isEditorReady]);
  useEffect(function () {
    !isMonacoMounting && !isEditorReady && createEditor();
  }, [isMonacoMounting, isEditorReady, createEditor]);

  // subscription
  // to avoid unnecessary updates (attach - dispose listener) in subscription
  valueRef.current = value;

  // onChange
  useEffect(function () {
    if (isEditorReady && onChange) {
      var _subscriptionRef$curr2, _editorRef$current8;
      (_subscriptionRef$curr2 = subscriptionRef.current) === null || _subscriptionRef$curr2 === void 0 || _subscriptionRef$curr2.dispose();
      subscriptionRef.current = (_editorRef$current8 = editorRef.current) === null || _editorRef$current8 === void 0 ? void 0 : _editorRef$current8.onDidChangeModelContent(function (event) {
        if (!preventTriggerChangeEvent.current) {
          onChange(editorRef.current.getValue(), event);
        }
      });
    }
  }, [isEditorReady, onChange]);

  // onValidate
  useEffect(function () {
    if (isEditorReady) {
      var changeMarkersListener = monacoRef.current.editor.onDidChangeMarkers(function (uris) {
        var _getModel2;
        var editorUri = (_getModel2 = editorRef.current.getModel()) === null || _getModel2 === void 0 ? void 0 : _getModel2.uri;
        if (editorUri) {
          var currentEditorHasMarkerChanges = uris.find(function (uri) {
            return uri.path === editorUri.path;
          });
          if (currentEditorHasMarkerChanges) {
            var markers = monacoRef.current.editor.getModelMarkers({
              resource: editorUri
            });
            onValidate === null || onValidate === void 0 || onValidate(markers);
          }
        }
      });
      return function () {
        changeMarkersListener === null || changeMarkersListener === void 0 || changeMarkersListener.dispose();
      };
    }
    return function () {
      // eslint happy
    };
  }, [isEditorReady, onValidate]);
  useEffect(function () {
    var _editorRef$current9, _editorRef$current10;
    if (!isEditorReady || !editorRef.current) return;
    // 监听编辑器聚焦事件
    (_editorRef$current9 = editorRef.current) === null || _editorRef$current9 === void 0 || _editorRef$current9.onDidFocusEditorText(function () {
      onFocus === null || onFocus === void 0 || onFocus();
    });

    // 监听编辑器失去焦点事件
    (_editorRef$current10 = editorRef.current) === null || _editorRef$current10 === void 0 || _editorRef$current10.onDidBlurEditorText(function () {
      onBlur === null || onBlur === void 0 || onBlur();
    });
  }, [isEditorReady]);
  return /*#__PURE__*/React.createElement(MonacoContainer, {
    style: style,
    isEditorReady: isEditorReady,
    loading: loading,
    _ref: containerRef,
    className: className,
    wrapperProps: wrapperProps
  });
});
