// @roddan/form-editor/es/views/settings/form-setting.js
import React from 'react';
import { Input, InputNumber } from 'antd';
import { observer } from 'mobx-react-lite';
import { SettingItem, SettingWrap, EventSetting, CustomCssSetting } from "../../components";
import { EEventAction, EEventType } from "../../types";
import store from "../../store";
import { prefixCls } from "../../const";
var FormSetting = function FormSetting() {
  var _store$formAttrs = store.formAttrs,
    id = _store$formAttrs.id,
    formName = _store$formAttrs.formName,
    horizontalGap = _store$formAttrs.horizontalGap,
    verticalGap = _store$formAttrs.verticalGap;
  return /*#__PURE__*/React.createElement("div", {
    className: prefixCls('form-setting')
  }, /*#__PURE__*/React.createElement(SettingWrap, {
    title: "\u57FA\u672C\u5C5E\u6027"
  }, /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u8868\u5355id"
  }, id), /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u8868\u5355\u540D\u79F0"
  }, /*#__PURE__*/React.createElement(Input, {
    value: formName,
    onChange: function onChange(e) {
      store.setFormAttr('formName', e.target.value);
    }
  })), /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u6C34\u5E73\u95F4\u9694"
  }, /*#__PURE__*/React.createElement(InputNumber, {
    value: horizontalGap,
    min: 0,
    onChange: function onChange(val) {
      store.setFormAttr('horizontalGap', Number(val));
    }
  })), /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u5782\u76F4\u95F4\u9694"
  }, /*#__PURE__*/React.createElement(InputNumber, {
    value: verticalGap,
    min: 0,
    onChange: function onChange(val) {
      store.setFormAttr('verticalGap', Number(val));
    }
  }))), /*#__PURE__*/React.createElement(SettingWrap, {
    title: "\u4E8B\u4EF6"
  }, /*#__PURE__*/React.createElement(EventSetting, {
    type: "form",
    eventActions: [EEventAction.FORM_LOADED],
    eventTypeOptions: [EEventType.UPDATE_SERVICE]
  })), /*#__PURE__*/React.createElement(SettingWrap, {
    title: "\u6837\u5F0F",
    style: {
      flex: 1,
      height: 0
    }
  }, /*#__PURE__*/React.createElement(CustomCssSetting, {
    type: "form"
  })));
};
export default observer(FormSetting);
