// @roddan/form-editor/es/views/settings/index.js
import { Tabs } from 'antd';
import { observer } from 'mobx-react-lite';
import React from 'react';
import ElementSetting from "./element-setting";
import FormSetting from "./form-setting";
import { FormService } from "../../components";
import { prefixCls } from "../../const";
import { tabStore } from "../../store";
import "./style.less";

// console.log('prefixCls', prefixCls);
// debugger;

var items = [{
  key: 'element',
  label: "\u7EC4\u4EF6\u5C5E\u6027",
  closable: false,
  children: /*#__PURE__*/React.createElement(ElementSetting, null)
}, {
  key: 'form',
  label: "\u8868\u5355\u5C5E\u6027",
  closable: false,
  children: /*#__PURE__*/React.createElement(FormSetting, null)
}, {
  key: 'service',
  label: "\u8868\u5355\u670D\u52A1",
  closable: false,
  children: /*#__PURE__*/React.createElement(FormService, null)
}];
var Settings = function Settings() {
  return /*#__PURE__*/React.createElement("div", {
    className: prefixCls('setting')
  }, /*#__PURE__*/React.createElement(Tabs, {
    activeKey: tabStore.formTab,
    type: "editable-card",
    hideAdd: true,
    items: items,
    className: "".concat(prefixCls('form-setting-tab')),
    onChange: function onChange(tab) {
      tabStore.setFormTab(tab);
    }
  }));
};
export default observer(Settings);
