// @roddan/form-editor/es/components/attributes-setting.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import React, { useEffect, useState, useRef } from 'react';
import { Modal, message } from 'antd';
import { observer } from 'mobx-react-lite';
import { MonacoEditor } from '@roddan/ui';
export var AttributesSetting = observer(function (_ref) {
  var children = _ref.children,
    title = _ref.title,
    value = _ref.value,
    onChange = _ref.onChange,
    editorType = _ref.editorType,
    style = _ref.style;
  var _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    open = _useState2[0],
    setOpen = _useState2[1];
  var _useState3 = useState(''),
    _useState4 = _slicedToArray(_useState3, 2),
    val = _useState4[0],
    setVal = _useState4[1];
  var isJsonValidate = useRef(true);
  useEffect(function () {
    setVal(value);
  }, [value]);
  return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.isValidElement(children) && /*#__PURE__*/React.cloneElement(children, {
    onClick: function onClick() {
      return setOpen(true);
    }
  }), /*#__PURE__*/React.createElement(Modal, {
    width: 600,
    open: open,
    title: title,
    onCancel: function onCancel() {
      setOpen(false);
    },
    onOk: function onOk() {
      if (isJsonValidate.current) {
        onChange === null || onChange === void 0 || onChange(val);
        setOpen(false);
        return;
      }
      message.error('格式不对');
    }
  }, /*#__PURE__*/React.createElement(MonacoEditor, {
    style: style !== null && style !== void 0 ? style : {
      height: 400
    },
    language: editorType,
    value: val,
    onChange: function onChange(v) {
      return setVal(v);
    },
    onValidate: function onValidate(errors) {
      // 参数变量未使用时不校验
      isJsonValidate.current = errors.filter(function (item) {
        return (item === null || item === void 0 ? void 0 : item.code) !== '6133';
      }).length === 0;
    },
    options: {
      tabSize: 2
    }
  })));
});
