// @roddan/form-editor/es/components/basic-info.js
import React from 'react';
import { Input, Select, Slider, Switch } from 'antd';
import { observer } from 'mobx-react-lite';
import { DirectionOpions } from "../const";
import store from "../store";
import { SettingItem, SettingWrap } from "./setting-common";
var BasicInfo = function BasicInfo() {
  var _store$selectedElemen2, _store$selectedElemen3, _store$selectedElemen4, _store$selectedElemen5;
  var _store$selectedElemen = store.selectedElement,
    gridSpan = _store$selectedElemen.gridSpan,
    id = _store$selectedElemen.id,
    gridOffset = _store$selectedElemen.gridOffset;
  return /*#__PURE__*/React.createElement(SettingWrap, {
    title: "\u57FA\u7840\u8BBE\u7F6E"
  }, /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u5143\u7D20id"
  }, /*#__PURE__*/React.createElement("div", null, id)), /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u5143\u7D20\u540D\u79F0"
  }, /*#__PURE__*/React.createElement(Input, {
    value: (_store$selectedElemen2 = store.selectedElement.elementName) === null || _store$selectedElemen2 === void 0 ? void 0 : _store$selectedElemen2.replace(/&nbsp;/g, ' '),
    onChange: function onChange(e) {
      store.setSelectedProp('elementName', e.target.value.replace(/\s/g, '&nbsp;'));
    }
  })), /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u663E\u793A\u540D\u79F0"
  }, /*#__PURE__*/React.createElement(Switch, {
    checked: (_store$selectedElemen3 = store.selectedElement) === null || _store$selectedElemen3 === void 0 ? void 0 : _store$selectedElemen3.showElementName,
    onChange: function onChange(checked) {
      store.setSelectedProp('showElementName', checked);
    },
    size: "small"
  })), /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u540D\u79F0\u5BF9\u9F50"
  }, /*#__PURE__*/React.createElement(Select, {
    options: DirectionOpions,
    value: store.selectedElement.elementNameDisplay || 'vertical',
    onChange: function onChange(val) {
      store.setSelectedProp('elementNameDisplay', val);
    }
  })), /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u6805\u683C\u5E03\u5C40"
  }, /*#__PURE__*/React.createElement(Switch, {
    checked: (_store$selectedElemen4 = store.selectedElement) === null || _store$selectedElemen4 === void 0 ? void 0 : _store$selectedElemen4.gridLayout,
    onChange: function onChange(checked) {
      store.setSelectedProp('gridLayout', checked);
    },
    size: "small"
  })), ((_store$selectedElemen5 = store.selectedElement) === null || _store$selectedElemen5 === void 0 ? void 0 : _store$selectedElemen5.gridLayout) && /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u5143\u7D20\u6805\u683C"
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      width: '90%'
    }
  }, /*#__PURE__*/React.createElement(Slider, {
    value: gridSpan,
    max: 24,
    min: 1,
    onChange: function onChange(v) {
      store.setSelectedProp('gridSpan', v);
    }
  }))), /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u5143\u7D20\u504F\u79FB"
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      width: '90%'
    }
  }, /*#__PURE__*/React.createElement(Slider, {
    value: gridOffset,
    max: 24,
    min: 0,
    onChange: function onChange(v) {
      store.setSelectedProp('gridOffset', v);
    }
  })))));
};
export default observer(BasicInfo);
