// @roddan/form-editor/es/components/batch-generate-options.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import React, { useState, useMemo, useEffect } from 'react';
import { Modal, Button, Input } from 'antd';
import { idCreator } from "../utils";
export var BatchGenerateOptions = function BatchGenerateOptions(_ref) {
  var title = _ref.title,
    options = _ref.options,
    setOptions = _ref.setOptions,
    labelField = _ref.labelField,
    valueField = _ref.valueField;
  var _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    open = _useState2[0],
    setOpen = _useState2[1];
  var _useState3 = useState(options),
    _useState4 = _slicedToArray(_useState3, 2),
    tempOptions = _useState4[0],
    setTempOptions = _useState4[1];
  useEffect(function () {
    setTempOptions(options);
  }, [options]);
  var text = useMemo(function () {
    return tempOptions.reduce(function (memo, cur, idx) {
      return memo + (idx === 0 ? '' : '\n') + cur[labelField] + (cur[valueField] ? ":".concat(cur[valueField]) : '');
    }, '');
  }, [tempOptions]);
  var handleChange = function handleChange(str) {
    var couples = str.split('\n');
    var newOptions = couples.map(function (couple) {
      var _couple$split = couple.split(':'),
        _couple$split2 = _slicedToArray(_couple$split, 2),
        _couple$split2$ = _couple$split2[0],
        label = _couple$split2$ === void 0 ? '' : _couple$split2$,
        _couple$split2$2 = _couple$split2[1],
        value = _couple$split2$2 === void 0 ? '' : _couple$split2$2;
      var oldOpt = options.find(function (item) {
        return item[labelField] === label;
      });
      if (oldOpt) return oldOpt;
      return _defineProperty(_defineProperty(_defineProperty({}, labelField, label.trim()), valueField, value.trim()), "id", idCreator('option'));
    });
    setTempOptions(newOptions);
  };
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    style: {
      display: 'flex',
      justifyContent: 'space-between',
      paddingRight: 30
    }
  }, /*#__PURE__*/React.createElement("div", null, title), /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement(Button, {
    onClick: function onClick() {
      setOpen(true);
    }
  }, "\u6279\u91CF\u7F16\u8F91"))), /*#__PURE__*/React.createElement(Modal, {
    open: open,
    title: "\u6279\u91CF\u7F16\u8F91",
    onCancel: function onCancel() {
      setOpen(false);
    },
    onOk: function onOk() {
      setOptions(tempOptions.filter(function (item) {
        return Boolean(item[labelField]);
      }));
      setOpen(false);
    },
    destroyOnClose: true
  }, /*#__PURE__*/React.createElement(Input.TextArea, {
    defaultValue: text,
    autoSize: {
      maxRows: 10
    },
    onChange: function onChange(e) {
      handleChange(e.target.value);
    }
  })));
};
