// @roddan/form-editor/es/components/tabs-setting.js
import React from 'react';
import { prefixCls } from "../const";
import { observer } from 'mobx-react-lite';
import { tabStore } from "../store";
import BasicInfo from "./basic-info";
import { CustomCssSetting } from "./custom-css-setting";
import { Tabs } from 'antd';
export var TabsSetting = observer(function (_ref) {
  var attributes = _ref.attributes,
    events = _ref.events,
    hideCss = _ref.hideCss;
  var items = [{
    key: 'attribute',
    label: "\u5C5E\u6027",
    children: /*#__PURE__*/React.createElement("div", {
      className: prefixCls('basic-info')
    }, /*#__PURE__*/React.createElement(BasicInfo, null), attributes)
  }, !hideCss && {
    key: 'style',
    label: "\u6837\u5F0F",
    children: /*#__PURE__*/React.createElement(CustomCssSetting, {
      type: "element"
    })
  }, {
    key: 'event',
    label: '事件',
    children: /*#__PURE__*/React.createElement(React.Fragment, null, events)
  }].filter(Boolean);
  return /*#__PURE__*/React.createElement(Tabs, {
    activeKey: tabStore.elementTab,
    onChange: function onChange(key) {
      tabStore.setElementTab(key);
    },
    items: items
  });
});
