// @roddan/form-editor/es/elements/tabs/render-tabs.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import React, { useEffect } from 'react';
import { observer } from 'mobx-react-lite';
import { Tabs } from 'antd';
import store from "../../store";
import { ElementLayout } from "../../components";
import { idCreator } from "../../utils";
// import { createPanel } from './const';
import { initialData, ELEMENT_CONTAINER } from "../container";
import { RenderContainer } from "../container";
export var createPanel = function createPanel() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var panel = _objectSpread(_objectSpread({}, initialData), {}, {
    elementName: 'tab选项卡',
    type: ELEMENT_CONTAINER,
    id: idCreator('el')
  }, props);
  store.appendEl(panel, false);
};
var RenderTabsContent = function RenderTabsContent(_ref) {
  var element = _ref.element;
  var children = element.children;
  var items = children === null || children === void 0 ? void 0 : children.map(function (child) {
    store.flatElement(child);
    return {
      label: child.elementName,
      key: child.id,
      children: /*#__PURE__*/React.createElement(RenderContainer, {
        element: child
      })
    };
  });
  useEffect(function () {
    if (!(children !== null && children !== void 0 && children.length)) {
      createPanel({
        parentId: element.id
      });
      store.setSelectedElement(element);
    }
  }, [children === null || children === void 0 ? void 0 : children.length]);
  return /*#__PURE__*/React.createElement(ElementLayout, {
    element: element
  }, /*#__PURE__*/React.createElement(Tabs, {
    items: items,
    type: element.tabType
  }));
};
export var RenderTabs = observer(RenderTabsContent);
