// @roddan/form-editor/es/elements/tabs/tabs-setting.js
import React from 'react';
import { observer } from 'mobx-react-lite';
import { Input, Space, Popconfirm, Select } from 'antd';
import { MinusCircleOutlined } from '@ant-design/icons';
import { SettingWrap, PlusIcon, SettingItem } from "../../components";
import { TableSortable } from '@roddan/ui';
import store from "../../store";
import { createPanel } from "./render-tabs";
var ObserverInput = observer(function (_ref) {
  var idx = _ref.idx;
  var current = store.selectedElement.children[idx];
  return /*#__PURE__*/React.createElement(Input, {
    value: current.elementName,
    onChange: function onChange(e) {
      store.setElementProp(current.id, 'elementName', e.target.value);
    }
  });
});
var SettingTabsContent = function SettingTabsContent() {
  var _store$selectedElemen = store.selectedElement,
    children = _store$selectedElemen.children,
    id = _store$selectedElemen.id,
    tabType = _store$selectedElemen.tabType;
  var _ref2 = children,
    length = _ref2.length;
  var handleAddPanel = function handleAddPanel() {
    createPanel({
      elementName: "tab\u9009\u9879\u5361".concat(length + 1),
      parentId: id
    });
  };
  var columns = [{
    title: '选项卡',
    dataIndex: 'elementName',
    render: function render(val, _, idx) {
      return /*#__PURE__*/React.createElement(ObserverInput, {
        idx: idx
      });
    }
  }, {
    title: '操作',
    width: 60,
    render: function render(_, __, idx) {
      return /*#__PURE__*/React.createElement(Space, null, /*#__PURE__*/React.createElement("span", null, /*#__PURE__*/React.createElement(Popconfirm, {
        title: "\u786E\u8BA4\u5220\u9664?",
        onConfirm: function onConfirm() {
          store.deleteEl(children[idx]);
        }
      }, /*#__PURE__*/React.createElement(MinusCircleOutlined, {
        style: {
          color: '#D40000',
          cursor: 'pointer'
        }
      }))), idx === length - 1 && /*#__PURE__*/React.createElement("span", null, /*#__PURE__*/React.createElement(PlusIcon, {
        onClick: handleAddPanel
      })));
    }
  }];
  return /*#__PURE__*/React.createElement(SettingWrap, {
    title: "\u5143\u7D20\u8BBE\u7F6E"
  }, /*#__PURE__*/React.createElement(SettingItem, {
    label: "tab\u7C7B\u578B"
  }, /*#__PURE__*/React.createElement(Select, {
    value: tabType,
    options: ['line', 'card'].map(function (item) {
      return {
        label: item,
        value: item
      };
    }),
    onChange: function onChange(val) {
      store.setSelectedProp('tabType', val);
    }
  })), /*#__PURE__*/React.createElement(TableSortable, {
    columns: columns,
    rowKey: "id",
    onSort: function onSort(newChildren) {
      store.setSelectedProp('children', newChildren);
    },
    dataSource: children,
    pagination: false,
    scroll: {
      y: 300
    }
    // children字段在tabs组件中有用
    ,
    childrenColumnName: "other"
  }));
};
export var SettingTabs = observer(SettingTabsContent);
