// @roddan/form-editor/es/components/event-setting/event-collapse.js
import { MinusSquareOutlined, PlusSquareOutlined } from '@ant-design/icons';
import { Collapse, Dropdown, Popconfirm } from 'antd';
import React from 'react';
import { prefixCls } from "../../const";
import { eventTypeChinese } from "../../types";
export var EventCollapse = function EventCollapse(_ref) {
  var collopaseItems = _ref.collopaseItems,
    onDelete = _ref.onDelete,
    EditComponent = _ref.EditComponent;
  var items = collopaseItems.map(function (item, i) {
    return {
      key: i,
      label: item.label,
      children: item.events.map(function (event, innerI) {
        return /*#__PURE__*/React.createElement(EditComponent, {
          evt: event,
          key: innerI
        }, /*#__PURE__*/React.createElement(Dropdown, {
          menu: {
            items: [{
              label: /*#__PURE__*/React.createElement(Popconfirm, {
                title: "\u786E\u8BA4\u5220\u9664",
                onConfirm: function onConfirm() {
                  return onDelete(event.id);
                }
              }, "\u5220\u9664"),
              key: 1
            }]
          },
          trigger: ['contextMenu']
          // @ts-ignore
          ,
          getPopupContainer: function getPopupContainer(n) {
            return n.parentNode;
          }
        }, /*#__PURE__*/React.createElement("div", {
          className: "collapse-item"
        }, eventTypeChinese[event.eventType])));
      })
    };
  });
  return /*#__PURE__*/React.createElement("div", {
    className: prefixCls('event-collapse')
  }, /*#__PURE__*/React.createElement(Collapse, {
    ghost: true,
    items: items,
    defaultActiveKey: items.map(function (item) {
      return item.key;
    }),
    expandIcon: function expandIcon(_ref2) {
      var isActive = _ref2.isActive;
      return isActive ? /*#__PURE__*/React.createElement(MinusSquareOutlined, null) : /*#__PURE__*/React.createElement(PlusSquareOutlined, null);
    }
  }));
};
