// @roddan/form-editor/es/components/event-setting/index.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import React from 'react';
import { Button, message } from 'antd';
import { cloneDeep, groupBy } from 'lodash-es';
import { observer } from 'mobx-react-lite';
import { prefixCls } from "../../const";
import { useCurrent } from "../../hooks";
import { handleLinkService, handleUnLinkService } from "../../utils";
import { EChangeType, eventActionInChinese } from "../../types";
import { EventCollapse } from "./event-collapse";
import { EventModal } from "../event-modal";
export * from "./event-collapse";
import "./style.less";
var formatForCollapse = function formatForCollapse(events) {
  var group = groupBy(events, 'eventAction');
  return Object.entries(group).map(function (_ref) {
    var _ref2 = _slicedToArray(_ref, 2),
      action = _ref2[0],
      events = _ref2[1];
    return {
      label: eventActionInChinese[action],
      events: events
    };
  });
};
export var EventSetting = observer(function (_ref3) {
  var eventActions = _ref3.eventActions,
    type = _ref3.type,
    eventTypeOptions = _ref3.eventTypeOptions;
  var _useCurrent = useCurrent(type),
    current = _useCurrent.current,
    setProp = _useCurrent.setProp;
  var sourceId = current.id;
  var handleSaveEvents = function handleSaveEvents(type, event) {
    var events = cloneDeep((current === null || current === void 0 ? void 0 : current.events) || []);
    if (type === EChangeType.ADD) {
      var _event$eventTargets;
      if (!(event !== null && event !== void 0 && (_event$eventTargets = event.eventTargets) !== null && _event$eventTargets !== void 0 && _event$eventTargets.length)) return;
      var sameActionEvent = events.find(function (existEvent) {
        return existEvent.eventAction === event.eventAction && existEvent.eventType === event.eventType;
      });
      if (sameActionEvent) {
        var _sameActionEvent$even;
        sameActionEvent.eventTargets = (_sameActionEvent$even = sameActionEvent.eventTargets) === null || _sameActionEvent$even === void 0 ? void 0 : _sameActionEvent$even.concat(event.eventTargets || []);
        message.success('新增事件已被合并');
      } else {
        events.push(event);
      }
      handleLinkService(event);
    } else {
      var _current$events, _event$eventTargets2;
      var idx = current === null || current === void 0 || (_current$events = current.events) === null || _current$events === void 0 ? void 0 : _current$events.findIndex(function (evt) {
        return event.id === evt.id;
      });
      handleUnLinkService(events[idx]);
      events[idx] = event;
      if (event !== null && event !== void 0 && (_event$eventTargets2 = event.eventTargets) !== null && _event$eventTargets2 !== void 0 && _event$eventTargets2.length) {
        handleLinkService(event);
      } else {
        events.splice(idx, 1);
      }
    }
    // modal过度效果
    setTimeout(function () {
      setProp('events', events);
    }, 200);
  };
  var handleDelete = function handleDelete(id) {
    var _current$events2;
    var events = cloneDeep(current === null || current === void 0 ? void 0 : current.events);
    var idx = current === null || current === void 0 || (_current$events2 = current.events) === null || _current$events2 === void 0 ? void 0 : _current$events2.findIndex(function (event) {
      return event.id === id;
    });
    handleUnLinkService(events[idx]);
    events.splice(idx, 1);
    setProp('events', events);
  };
  return /*#__PURE__*/React.createElement("div", {
    className: prefixCls('event-common-wrap')
  }, /*#__PURE__*/React.createElement(EventModal, {
    eventActions: eventActions,
    onOk: function onOk(evt) {
      return handleSaveEvents(EChangeType.ADD, evt);
    },
    type: EChangeType.ADD,
    sourceId: sourceId,
    eventTypeOptions: eventTypeOptions
  }, /*#__PURE__*/React.createElement(Button, {
    type: "dashed",
    className: prefixCls('event-button-add')
  }, "+ \u65B0\u589E\u4E8B\u4EF6")), /*#__PURE__*/React.createElement(EventCollapse, {
    collopaseItems: formatForCollapse((current === null || current === void 0 ? void 0 : current.events) || []),
    onDelete: handleDelete,
    EditComponent: function EditComponent(_ref4) {
      var children = _ref4.children,
        evt = _ref4.evt;
      return /*#__PURE__*/React.createElement(EventModal, {
        eventActions: eventActions,
        onOk: function onOk(evt) {
          return handleSaveEvents(EChangeType.EDIT, evt);
        },
        event: evt,
        type: EChangeType.EDIT,
        sourceId: sourceId,
        eventTypeOptions: eventTypeOptions
      }, children);
    }
  }));
});
