// @roddan/form-editor/lib/elements/time/render-time.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RenderTime = void 0;
var _react = _interopRequireDefault(require("react"));
var _antd = require("antd");
var _dayjs = _interopRequireDefault(require("dayjs"));
var _mobxReactLite = require("mobx-react-lite");
var _hooks = require("../../hooks");
var _utils = require("../../utils");
var _types = require("../../types");
var _components = require("../../components");
var _store = _interopRequireDefault(require("../../store"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var RenderTimeContent = function RenderTimeContent(_ref) {
  var fieldValue = _ref.fieldValue,
    element = _ref.element;
  var id = element.id,
    dateFormat = element.dateFormat,
    placeholder = element.placeholder;
  var _useRegisterEvents = (0, _hooks.useRegisterEvents)(element),
    eventFunctions = _useRegisterEvents.eventFunctions;
  var handleEvent = function handleEvent(action) {
    return function (e) {
      var _eventFunctions$actio;
      (_eventFunctions$actio = eventFunctions[action]) === null || _eventFunctions$actio === void 0 || _eventFunctions$actio.call(eventFunctions, e.target.value);
    };
  };
  var handleChange = function handleChange(date) {
    _store.default.setFieldValue(id, date ? (0, _utils.formatDate)(date, dateFormat) : undefined);
  };
  (0, _hooks.useFormUpdate)(function () {
    var _eventFunctions$EEven;
    (_eventFunctions$EEven = eventFunctions[_types.EEventAction.VALUE_CHANGE]) === null || _eventFunctions$EEven === void 0 || _eventFunctions$EEven.call(eventFunctions, fieldValue);
  }, [fieldValue]);
  return /*#__PURE__*/_react.default.createElement(_components.ElementLayout, {
    element: element
  }, /*#__PURE__*/_react.default.createElement(_antd.TimePicker, {
    format: dateFormat,
    value: fieldValue ? (0, _dayjs.default)("2000-01-01 ".concat(fieldValue)) : undefined,
    getPopupContainer: function getPopupContainer(n) {
      return n.parentElement;
    },
    onChange: handleChange,
    onFocus: handleEvent(_types.EEventAction.ON_FOCUS),
    onBlur: handleEvent(_types.EEventAction.ON_BLUR),
    placeholder: placeholder
  }));
};
var RenderTime = exports.RenderTime = (0, _mobxReactLite.observer)(RenderTimeContent);
