// @roddan/form-editor/lib/elements/time/time-setting.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SettingTime = void 0;
var _react = _interopRequireDefault(require("react"));
var _antd = require("antd");
var _mobxReactLite = require("mobx-react-lite");
var _components = require("../../components");
var _store = _interopRequireDefault(require("../../store"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var dateOptions = ['HH:mm:ss', 'HH:mm'].map(function (per) {
  return {
    label: per,
    value: per
  };
});
var SettingTimeContent = function SettingTimeContent() {
  var dateFormat = _store.default.selectedElement.dateFormat;
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_components.SettingWrap, {
    title: "\u5143\u7D20\u8BBE\u7F6E"
  }, /*#__PURE__*/_react.default.createElement(_components.PlaceholderSetting, null), /*#__PURE__*/_react.default.createElement(_components.SettingItem, {
    label: "\u65F6\u95F4\u683C\u5F0F"
  }, /*#__PURE__*/_react.default.createElement(_antd.Select, {
    value: dateFormat,
    style: {
      width: '100%'
    },
    options: dateOptions,
    onChange: function onChange(val) {
      // console.log(e)
      _store.default.setSelectedProp('dateFormat', val);
    }
  }))), /*#__PURE__*/_react.default.createElement(_components.RegPattern, null));
};
var SettingTime = exports.SettingTime = (0, _mobxReactLite.observer)(SettingTimeContent);
