// @roddan/form-editor/lib/components/event-modal/action-config/index.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ActionConfig = void 0;
var _react = _interopRequireWildcard(require("react"));
var _antd = require("antd");
var _classnames = _interopRequireDefault(require("classnames"));
var _lodashEs = require("lodash-es");
var _commonIcon = require("../../common-icon");
var _const = require("../../../const");
var _types = require("../../../types");
var _utils = require("../../../utils");
var _linkService = _interopRequireDefault(require("./link-service"));
var _refreshService = _interopRequireDefault(require("./refresh-service"));
var _setElementValue = _interopRequireDefault(require("./set-element-value"));
var _validate = _interopRequireDefault(require("./validate"));
var _context = require("../context");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
var ActionItem = function ActionItem(_ref) {
  var type = _ref.type,
    onChange = _ref.onChange,
    eventTarget = _ref.eventTarget,
    last = _ref.last,
    onAdd = _ref.onAdd,
    onDelete = _ref.onDelete;
  var renderConfig = function renderConfig() {
    var props = {
      onChange: onChange,
      eventTarget: eventTarget
    };
    switch (type) {
      case _types.EEventType.SETTING_VALUE:
        return /*#__PURE__*/_react.default.createElement(_setElementValue.default, props);
      case _types.EEventType.UPDATE_SERVICE:
        return /*#__PURE__*/_react.default.createElement(_refreshService.default, props);
      case _types.EEventType.LINK_SERVICE:
        return /*#__PURE__*/_react.default.createElement(_linkService.default, props);
      case _types.EEventType.VALIDATE:
        return /*#__PURE__*/_react.default.createElement(_validate.default, props);
      default:
        return null;
    }
  };
  var Config = renderConfig();
  return Config && /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _const.prefixCls)('event-action-config')
  }, Config, /*#__PURE__*/_react.default.createElement(_antd.Space, null, /*#__PURE__*/_react.default.createElement(_antd.Popconfirm, {
    title: "\u786E\u8BA4\u5220\u9664",
    onConfirm: onDelete
  }, /*#__PURE__*/_react.default.createElement("span", null, /*#__PURE__*/_react.default.createElement(_commonIcon.MinusIcon, null))), last && /*#__PURE__*/_react.default.createElement("span", null, /*#__PURE__*/_react.default.createElement(_commonIcon.PlusIcon, {
    onClick: onAdd
  }))));
};
var getNewTarget = function getNewTarget(sourceId) {
  return {
    id: (0, _utils.idCreator)('event-target'),
    sourceId: sourceId
  };
};
var ActionConfig = exports.ActionConfig = function ActionConfig(_ref2) {
  var _currentEvent$eventTa2, _currentEvent$eventTa3;
  var title = _ref2.title,
    className = _ref2.className,
    currentEvent = _ref2.currentEvent,
    operationType = _ref2.operationType;
  var _useContext = (0, _react.useContext)(_context.EventModalContext),
    handleChangeEvent = _useContext.handleChangeEvent,
    setEdit = _useContext.setEdit,
    sourceId = _useContext.sourceId;
  var handleChange = function handleChange(type, targetAttr, idx) {
    if (type === _types.EChangeType.EDIT) {
      var newEventTargets = (0, _lodashEs.cloneDeep)(currentEvent.eventTargets);
      newEventTargets[idx] = Object.assign(newEventTargets[idx], targetAttr);
      handleChangeEvent('eventTargets', newEventTargets);
      setEdit(true);
    } else {
      handleChangeEvent('eventTargets', (currentEvent.eventTargets || []).concat(getNewTarget(sourceId)));
    }
  };
  var handleDelete = function handleDelete(idx) {
    var newEventTargets = (0, _lodashEs.cloneDeep)(currentEvent.eventTargets);
    newEventTargets.splice(idx, 1);
    handleChangeEvent('eventTargets', newEventTargets);
    setEdit(true);
  };
  (0, _react.useEffect)(function () {
    var _currentEvent$eventTa;
    if (currentEvent !== null && currentEvent !== void 0 && (_currentEvent$eventTa = currentEvent.eventTargets) !== null && _currentEvent$eventTa !== void 0 && _currentEvent$eventTa.length || operationType === _types.EChangeType.EDIT) return;
    handleChange(_types.EChangeType.ADD);
  }, [currentEvent === null || currentEvent === void 0 ? void 0 : currentEvent.eventTargets, operationType]);
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)((0, _const.prefixCls)('event-modal-column'), className)
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _const.prefixCls)('event-modal-title'),
    style: {
      borderRight: 'none'
    }
  }, title), (currentEvent === null || currentEvent === void 0 ? void 0 : currentEvent.eventAction) && (currentEvent === null || currentEvent === void 0 ? void 0 : currentEvent.eventType) && /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, currentEvent === null || currentEvent === void 0 || (_currentEvent$eventTa2 = currentEvent.eventTargets) === null || _currentEvent$eventTa2 === void 0 ? void 0 : _currentEvent$eventTa2.map(function (eventTarget, i) {
    return /*#__PURE__*/_react.default.createElement(ActionItem, {
      key: eventTarget.id,
      type: currentEvent.eventType,
      onChange: function onChange(targetAttr) {
        return handleChange(_types.EChangeType.EDIT, targetAttr, i);
      },
      eventTarget: eventTarget,
      last: i === (currentEvent === null || currentEvent === void 0 ? void 0 : currentEvent.eventTargets.length) - 1,
      onAdd: function onAdd() {
        return handleChange(_types.EChangeType.ADD);
      },
      onDelete: function onDelete() {
        return handleDelete(i);
      }
    });
  }), !(currentEvent !== null && currentEvent !== void 0 && (_currentEvent$eventTa3 = currentEvent.eventTargets) !== null && _currentEvent$eventTa3 !== void 0 && _currentEvent$eventTa3.length) && /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _const.prefixCls)('event-action-config'),
    style: {
      justifyContent: 'flex-end'
    }
  }, /*#__PURE__*/_react.default.createElement(_commonIcon.PlusIcon, {
    onClick: function onClick() {
      return handleChange(_types.EChangeType.ADD);
    }
  }))));
};
