// @roddan/form-editor/lib/components/event-modal/action-config/link-service.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _antd = require("antd");
var _mobxReactLite = require("mobx-react-lite");
var _const = require("../../../const");
var _store = _interopRequireDefault(require("../../../store"));
var _types = require("../../../types");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var LinkService = function LinkService(_ref) {
  var _onChange = _ref.onChange,
    eventTarget = _ref.eventTarget;
  var _ref2 = eventTarget || {},
    targetServiceId = _ref2.targetServiceId,
    linkRefreshField = _ref2.linkRefreshField,
    getFieldFromService = _ref2.getFieldFromService;
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      lineHeight: '40px'
    }
  }, /*#__PURE__*/_react.default.createElement("div", null, "\u76EE\u6807\u670D\u52A1 =", ' ', /*#__PURE__*/_react.default.createElement(_antd.Select, {
    className: (0, _const.prefixCls)('event-select'),
    options: _store.default.getFormServices(),
    fieldNames: {
      label: 'name',
      value: 'id'
    },
    defaultValue: targetServiceId,
    onChange: function onChange(v) {
      _onChange({
        targetServiceId: v
      });
    }
  })), /*#__PURE__*/_react.default.createElement("div", null, "\u83B7\u53D6\u7ED3\u679C\u5B57\u6BB5 =", ' ', /*#__PURE__*/_react.default.createElement(_antd.Input, {
    className: (0, _const.prefixCls)('event-select'),
    value: getFieldFromService !== null && getFieldFromService !== void 0 ? getFieldFromService : 'data',
    onChange: function onChange(e) {
      _onChange({
        getFieldFromService: e.target.value
      });
    }
  })), /*#__PURE__*/_react.default.createElement("div", null, "\u83B7\u53D6\u7ED3\u679C\u66F4\u65B0\u7EC4\u4EF6", ' ', /*#__PURE__*/_react.default.createElement(_antd.Select, {
    style: {
      width: 120
    },
    options: _types.linkRefreshFieldOptions,
    defaultValue: linkRefreshField,
    onChange: function onChange(v) {
      _onChange({
        linkRefreshField: v
      });
    }
  })));
};
var _default = exports.default = (0, _mobxReactLite.observer)(LinkService);
