// @roddan/form-editor/lib/components/event-modal/action-config/refresh-service.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _antd = require("antd");
var _mobxReactLite = require("mobx-react-lite");
var _react = _interopRequireDefault(require("react"));
var _const = require("../../../const");
var _store = _interopRequireDefault(require("../../../store"));
var _types = require("../../../types");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var actions = (0, _types.changeStateActions)([_types.EChangeStatePayload.UPDATE, _types.EChangeStatePayload.CLEAR, _types.EChangeStatePayload.APPEND]);
var RefreshService = function RefreshService(_ref) {
  var _onChange = _ref.onChange,
    eventTarget = _ref.eventTarget;
  var _ref2 = eventTarget || {},
    targetServiceId = _ref2.targetServiceId,
    targetPayload = _ref2.targetPayload,
    refreshFlag = _ref2.refreshFlag,
    updateField = _ref2.updateField;
  var renderAction = function renderAction() {
    if (targetPayload === _types.EChangeStatePayload.UPDATE || targetPayload === _types.EChangeStatePayload.APPEND) {
      return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, "\u4F20\u5165\u7EC4\u4EF6\u5143\u7D20\u503C\xA0", /*#__PURE__*/_react.default.createElement(_antd.Select, {
        className: (0, _const.prefixCls)('event-input'),
        options: actions,
        key: "action",
        defaultValue: targetPayload,
        onChange: function onChange(v) {
          _onChange({
            targetPayload: v
          });
        }
      }), "\xA0", /*#__PURE__*/_react.default.createElement(_antd.Input, {
        className: (0, _const.prefixCls)('event-input'),
        defaultValue: updateField,
        onChange: function onChange(e) {
          _onChange({
            updateField: e.target.value
          });
        }
      }), "\xA0\u5B57\u6BB5", targetPayload === _types.EChangeStatePayload.APPEND && /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, "\u5230url\u4E0A"));
    }
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_antd.Select, {
      className: (0, _const.prefixCls)('event-input'),
      options: actions,
      key: "action",
      defaultValue: targetPayload,
      onChange: function onChange(v) {
        _onChange({
          targetPayload: v
        });
      }
    }), "\xA0\u6240\u6709\u5B57\u6BB5\uFF0C");
  };
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      lineHeight: '40px'
    }
  }, /*#__PURE__*/_react.default.createElement("div", null, "\u76EE\u6807\u670D\u52A1 =", ' ', /*#__PURE__*/_react.default.createElement(_antd.Select, {
    allowClear: true,
    className: (0, _const.prefixCls)('event-input'),
    options: _store.default.getFormServices(),
    fieldNames: {
      label: 'name',
      value: 'id'
    },
    style: {
      width: 200
    },
    defaultValue: targetServiceId,
    onChange: function onChange(v) {
      _onChange({
        targetServiceId: v
      });
    }
  })), /*#__PURE__*/_react.default.createElement("div", null, renderAction()), "\u5E76", ' ', /*#__PURE__*/_react.default.createElement(_antd.Select, {
    className: (0, _const.prefixCls)('event-input'),
    options: _types.refreshOptions,
    defaultValue: refreshFlag,
    onChange: function onChange(v) {
      _onChange({
        refreshFlag: v
      });
    }
  }), ' ', "\u670D\u52A1");
};
var _default = exports.default = (0, _mobxReactLite.observer)(RefreshService);
