// @roddan/form-editor/lib/components/event-modal/action-config/validate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _antd = require("antd");
var _const = require("../../../const");
var _types = require("../../../types");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var Validate = function Validate(_ref) {
  var _onChange = _ref.onChange,
    eventTarget = _ref.eventTarget;
  var _ref2 = eventTarget || {},
    validateField = _ref2.validateField;
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      lineHeight: '40px'
    }
  }, "\u4E8B\u4EF6\u53D1\u751F\u65F6, \u5C06\u6821\u9A8C", ' ', /*#__PURE__*/_react.default.createElement(_antd.Select, {
    allowClear: true,
    className: (0, _const.prefixCls)('event-select'),
    options: _types.validateTypeOptions,
    defaultValue: validateField,
    onChange: function onChange(v) {
      _onChange({
        validateField: v
      });
    }
  }));
};
var _default = exports.default = Validate;
