// @roddan/form-editor/es/elements/input/input-setting.js
import React from 'react';
import { InputNumber, Select, Switch } from 'antd';
import { observer } from 'mobx-react-lite';
import { PlaceholderSetting, RegPattern, SettingItem, SettingWrap } from "../../components";
import store from "../../store";
var typeOptions = [{
  label: '单行文本',
  value: 'single'
}, {
  label: '多行文本',
  value: 'multiple'
}, {
  label: '密码框',
  value: 'password'
}];
var SettingInputContent = function SettingInputContent() {
  var _store$selectedElemen = store.selectedElement,
    textType = _store$selectedElemen.textType,
    minRows = _store$selectedElemen.minRows,
    maxRows = _store$selectedElemen.maxRows,
    autoSize = _store$selectedElemen.autoSize;
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(SettingWrap, {
    title: "\u5143\u7D20\u8BBE\u7F6E"
  }, /*#__PURE__*/React.createElement(PlaceholderSetting, null), /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u6587\u672C\u7C7B\u578B"
  }, /*#__PURE__*/React.createElement(Select, {
    value: textType,
    style: {
      width: '100%'
    },
    options: typeOptions,
    onChange: function onChange(val) {
      store.setSelectedProp('textType', val);
    }
  })), textType === 'multiple' && /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u81EA\u9002\u5E94\u884C\u6570"
  }, /*#__PURE__*/React.createElement(Switch, {
    checked: autoSize,
    onChange: function onChange(val) {
      store.setSelectedProp('autoSize', val);
    }
  })), !autoSize && /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u6700\u5C0F\u884C\u6570"
  }, /*#__PURE__*/React.createElement(InputNumber, {
    value: minRows,
    style: {
      width: '100%'
    },
    onChange: function onChange(val) {
      store.setSelectedProp('minRows', Number(val));
    }
  })), /*#__PURE__*/React.createElement(SettingItem, {
    label: "\u6700\u5927\u884C\u6570"
  }, /*#__PURE__*/React.createElement(InputNumber, {
    value: maxRows,
    min: minRows,
    style: {
      width: '100%'
    },
    onChange: function onChange(val) {
      store.setSelectedProp('maxRows', Number(val));
    }
  }))))), /*#__PURE__*/React.createElement(RegPattern, null));
};
export var SettingInput = observer(SettingInputContent);
