// @roddan/form-editor/es/elements/input/render-input.js
import React from 'react';
import { Input } from 'antd';
import { observer } from 'mobx-react-lite';
import { ElementLayout } from "../../components";
import { useRegisterEvents, useFormUpdate } from "../../hooks";
import store from "../../store";
import { EEventAction } from "../../types";
var RenderInputContent = function RenderInputContent(_ref) {
  var fieldValue = _ref.fieldValue,
    _ref$element = _ref.element,
    element = _ref$element === void 0 ? {} : _ref$element;
  var textType = element.textType,
    minRows = element.minRows,
    maxRows = element.maxRows,
    id = element.id,
    autoSize = element.autoSize,
    placeholder = element.placeholder;
  var _useRegisterEvents = useRegisterEvents(element),
    eventFunctions = _useRegisterEvents.eventFunctions;
  var handleEvent = function handleEvent(action) {
    return function (e) {
      var _eventFunctions$actio;
      (_eventFunctions$actio = eventFunctions[action]) === null || _eventFunctions$actio === void 0 || _eventFunctions$actio.call(eventFunctions, e.target.value);
    };
  };
  var handleChange = function handleChange(e) {
    store.setFieldValue(id, e.target.value);
  };
  useFormUpdate(function () {
    var _eventFunctions$EEven;
    (_eventFunctions$EEven = eventFunctions[EEventAction.ON_LOADED]) === null || _eventFunctions$EEven === void 0 || _eventFunctions$EEven.call(eventFunctions);
  }, [eventFunctions[EEventAction.ON_LOADED]]);
  useFormUpdate(function () {
    var _eventFunctions$EEven2;
    (_eventFunctions$EEven2 = eventFunctions[EEventAction.VALUE_CHANGE]) === null || _eventFunctions$EEven2 === void 0 || _eventFunctions$EEven2.call(eventFunctions, fieldValue);
  }, [fieldValue]);
  return /*#__PURE__*/React.createElement(ElementLayout, {
    element: element
  }, textType === 'multiple' ? /*#__PURE__*/React.createElement(Input.TextArea, {
    autoSize: autoSize ? true : {
      minRows: minRows,
      maxRows: maxRows
    },
    value: fieldValue,
    placeholder: placeholder,
    id: id,
    onChange: handleChange,
    onFocus: handleEvent(EEventAction.ON_FOCUS),
    onBlur: handleEvent(EEventAction.ON_BLUR)
  }) : /*#__PURE__*/React.createElement(Input, {
    placeholder: placeholder,
    id: id,
    onChange: handleChange,
    onFocus: handleEvent(EEventAction.ON_FOCUS),
    onBlur: handleEvent(EEventAction.ON_BLUR),
    value: fieldValue,
    type: textType === 'single' ? 'text' : 'password',
    autoComplete: "new-password"
  }));
};
export var RenderInput = observer(RenderInputContent);
