// @roddan/form-editor/lib/elements/container/render-container.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RenderContainerContent = exports.RenderContainer = void 0;
var _react = _interopRequireDefault(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _classnames = _interopRequireDefault(require("classnames"));
var _context = require("../../context");
var _const = require("../../const");
var _utils = require("../../utils");
var _store = _interopRequireDefault(require("../../store"));
var _components = require("../../components");
var _export = require("../export");
var _ui = require("@roddan/ui");
require("./style.less");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
// import { ReactSortable } from '@/components/react-sortable';

var RenderContainerContent = exports.RenderContainerContent = function RenderContainerContent(_ref) {
  var _element$children, _element$children2;
  var element = _ref.element;
  var _store$formAttrs = _store.default.formAttrs,
    horizontalGap = _store$formAttrs.horizontalGap,
    verticalGap = _store$formAttrs.verticalGap,
    id = _store$formAttrs.id;
  var _useEditorContext = (0, _context.useEditorContext)(),
    mode = _useEditorContext.mode;
  var parentId = element.parentId;
  return /*#__PURE__*/_react.default.createElement(_components.ElementLayout, {
    element: element
  }, /*#__PURE__*/_react.default.createElement(_ui.ReactSortable, {
    list: element.children || [],
    animation: 150,
    group: "nested",
    onSort: function onSort(e) {
      return (0, _utils.handleSort)(e, element.id);
    },
    rowProps: {
      className: (0, _classnames.default)([(0, _const.prefixCls)('row'), (0, _const.prefixCls)('row-el'), mode === 'design' && !((_element$children = element.children) !== null && _element$children !== void 0 && _element$children.length) ? (0, _const.prefixCls)('row-empty') : '']),
      gutter: [horizontalGap, verticalGap],
      'data-id': element.id,
      'data-parent-id': parentId || id
    },
    forbidden: mode !== 'design'
  }, (_element$children2 = element.children) === null || _element$children2 === void 0 ? void 0 : _element$children2.map(function (item) {
    var _ElementsList;
    var Component = (_ElementsList = _export.ElementsList[item.type]) === null || _ElementsList === void 0 ? void 0 : _ElementsList.render;
    if (!Component) return null;
    _store.default.flatElement(item);
    return /*#__PURE__*/_react.default.createElement(Component, {
      key: item.id || String(+new Date()),
      fieldValue: _store.default.fieldValues[item.id],
      element: item
    });
  })));
};
var RenderContainer = exports.RenderContainer = (0, _mobxReactLite.observer)(RenderContainerContent);
