// @roddan/form-editor/lib/elements/tabs/tabs-setting.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SettingTabs = void 0;
var _react = _interopRequireDefault(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _antd = require("antd");
var _icons = require("@ant-design/icons");
var _components = require("../../components");
var _ui = require("@roddan/ui");
var _store = _interopRequireDefault(require("../../store"));
var _renderTabs = require("./render-tabs");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var ObserverInput = (0, _mobxReactLite.observer)(function (_ref) {
  var idx = _ref.idx;
  var current = _store.default.selectedElement.children[idx];
  return /*#__PURE__*/_react.default.createElement(_antd.Input, {
    value: current.elementName,
    onChange: function onChange(e) {
      _store.default.setElementProp(current.id, 'elementName', e.target.value);
    }
  });
});
var SettingTabsContent = function SettingTabsContent() {
  var _store$selectedElemen = _store.default.selectedElement,
    children = _store$selectedElemen.children,
    id = _store$selectedElemen.id,
    tabType = _store$selectedElemen.tabType;
  var _ref2 = children,
    length = _ref2.length;
  var handleAddPanel = function handleAddPanel() {
    (0, _renderTabs.createPanel)({
      elementName: "tab\u9009\u9879\u5361".concat(length + 1),
      parentId: id
    });
  };
  var columns = [{
    title: '选项卡',
    dataIndex: 'elementName',
    render: function render(val, _, idx) {
      return /*#__PURE__*/_react.default.createElement(ObserverInput, {
        idx: idx
      });
    }
  }, {
    title: '操作',
    width: 60,
    render: function render(_, __, idx) {
      return /*#__PURE__*/_react.default.createElement(_antd.Space, null, /*#__PURE__*/_react.default.createElement("span", null, /*#__PURE__*/_react.default.createElement(_antd.Popconfirm, {
        title: "\u786E\u8BA4\u5220\u9664?",
        onConfirm: function onConfirm() {
          _store.default.deleteEl(children[idx]);
        }
      }, /*#__PURE__*/_react.default.createElement(_icons.MinusCircleOutlined, {
        style: {
          color: '#D40000',
          cursor: 'pointer'
        }
      }))), idx === length - 1 && /*#__PURE__*/_react.default.createElement("span", null, /*#__PURE__*/_react.default.createElement(_components.PlusIcon, {
        onClick: handleAddPanel
      })));
    }
  }];
  return /*#__PURE__*/_react.default.createElement(_components.SettingWrap, {
    title: "\u5143\u7D20\u8BBE\u7F6E"
  }, /*#__PURE__*/_react.default.createElement(_components.SettingItem, {
    label: "tab\u7C7B\u578B"
  }, /*#__PURE__*/_react.default.createElement(_antd.Select, {
    value: tabType,
    options: ['line', 'card'].map(function (item) {
      return {
        label: item,
        value: item
      };
    }),
    onChange: function onChange(val) {
      _store.default.setSelectedProp('tabType', val);
    }
  })), /*#__PURE__*/_react.default.createElement(_ui.TableSortable, {
    columns: columns,
    rowKey: "id",
    onSort: function onSort(newChildren) {
      _store.default.setSelectedProp('children', newChildren);
    },
    dataSource: children,
    pagination: false,
    scroll: {
      y: 300
    }
    // children字段在tabs组件中有用
    ,
    childrenColumnName: "other"
  }));
};
var SettingTabs = exports.SettingTabs = (0, _mobxReactLite.observer)(SettingTabsContent);
