// @roddan/form-editor/lib/elements/select/render-select.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RenderSelect = void 0;
var _react = _interopRequireDefault(require("react"));
var _antd = require("antd");
var _mobxReactLite = require("mobx-react-lite");
var _store = _interopRequireDefault(require("../../store"));
var _components = require("../../components");
var _hooks = require("../../hooks");
var _types = require("../../types");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var RenderSelectContent = function RenderSelectContent(_ref) {
  var element = _ref.element,
    fieldValue = _ref.fieldValue;
  var id = element.id,
    valueOptions = element.valueOptions,
    placeholder = element.placeholder;
  var _useRegisterEvents = (0, _hooks.useRegisterEvents)(element),
    eventFunctions = _useRegisterEvents.eventFunctions;
  var onChange = function onChange(val) {
    _store.default.setFieldValue(id, val);
  };
  (0, _hooks.useFormUpdate)(function () {
    var _eventFunctions$EEven;
    (_eventFunctions$EEven = eventFunctions[_types.EEventAction.ON_LOADED]) === null || _eventFunctions$EEven === void 0 || _eventFunctions$EEven.call(eventFunctions);
  }, [eventFunctions[_types.EEventAction.ON_LOADED]]);
  (0, _hooks.useFormUpdate)(function () {
    var _eventFunctions$EEven2;
    (_eventFunctions$EEven2 = eventFunctions[_types.EEventAction.VALUE_CHANGE]) === null || _eventFunctions$EEven2 === void 0 || _eventFunctions$EEven2.call(eventFunctions, fieldValue);
  }, [fieldValue]);
  return /*#__PURE__*/_react.default.createElement(_components.ElementLayout, {
    element: element
  }, /*#__PURE__*/_react.default.createElement(_antd.Select, {
    placeholder: placeholder,
    options: valueOptions,
    onChange: onChange
  }));
};
var RenderSelect = exports.RenderSelect = (0, _mobxReactLite.observer)(RenderSelectContent);
