// @roddan/form-editor/es/views/form-canvas/render-canvas.js
import React, { useMemo } from 'react';
import { observer } from 'mobx-react-lite';
import { useRegisterEvents, useFormUpdate, useDesignEffect } from "../../hooks";
import { EEventAction } from "../../types";
import { prefixCls } from "../../const";
import { ElementsList } from "../../elements/export";
import eventStore from "../../store/eventStore";
import store from "../../store";
import { handleOnEvent, parseCSS, handleSort } from "../../utils";
// import { ReactSortable } from '@/components/react-sortable';
import { ReactSortable } from '@roddan/ui';
import "./style.less";
var EditorCanvas = function EditorCanvas(_ref) {
  var mode = _ref.mode,
    actions = _ref.actions;
  var _store$formAttrs = store.formAttrs,
    horizontalGap = _store$formAttrs.horizontalGap,
    verticalGap = _store$formAttrs.verticalGap,
    id = _store$formAttrs.id,
    events = _store$formAttrs.events,
    customCss = _store$formAttrs.customCss;
  var _useRegisterEvents = useRegisterEvents({
      id: id,
      events: events
    }),
    eventFunctions = _useRegisterEvents.eventFunctions;
  var formStyle = useMemo(function () {
    if (!customCss) return {};
    var cssObj = parseCSS(customCss);
    return Object.values(cssObj)[0];
  }, [customCss]);
  useDesignEffect(function () {
    var keydonwFn = function keydonwFn(e) {
      if (e.key === 'Backspace' && document.activeElement === document.body) {
        store.deleteEl(store.selectedElement);
      }
    };
    document.addEventListener('keydown', keydonwFn);
    return function () {
      document.removeEventListener('keydown', keydonwFn);
    };
  });

  // 表单加载事件
  useFormUpdate(function () {
    var _eventFunctions$EEven;
    (_eventFunctions$EEven = eventFunctions[EEventAction.FORM_LOADED]) === null || _eventFunctions$EEven === void 0 || _eventFunctions$EEven.call(eventFunctions);
  }, [eventFunctions[EEventAction.FORM_LOADED]]);

  // 服务监听事件 - ps:不用关心设计模式下的运行
  useFormUpdate(function () {
    if (!store.formServices.length) return;
    store.formServices.forEach(function (serv) {
      eventStore.emitter.on(serv.id, handleOnEvent);
    });
    return function () {
      store.formServices.forEach(function (serv) {
        eventStore.emitter.off(serv.id, handleOnEvent);
      });
    };
  }, [store.formServices]);
  return /*#__PURE__*/React.createElement("div", {
    className: prefixCls('canvas-wrap')
  }, actions && /*#__PURE__*/React.createElement(React.Fragment, null, actions), /*#__PURE__*/React.createElement(ReactSortable, {
    className: prefixCls('canvas'),
    style: formStyle,
    list: store.formElements,
    onSort: function onSort(e) {
      return handleSort(e, id);
    },
    animation: 150,
    group: "nested",
    rowProps: {
      className: prefixCls('row'),
      gutter: [horizontalGap, verticalGap],
      'data-id': id
    },
    forbidden: mode !== 'design'
  }, store.formElements.map(function (item) {
    var _ElementsList;
    var Component = (_ElementsList = ElementsList[item.type]) === null || _ElementsList === void 0 ? void 0 : _ElementsList.render;
    if (!Component) return null;
    store.flatElement(item);
    return /*#__PURE__*/React.createElement(Component, {
      key: item.id,
      fieldValue: store.fieldValues[item.id],
      element: item
    });
  })));
};
export default observer(EditorCanvas);
