// @roddan/form-editor/lib/components/reg-pattern/index.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RegPattern = void 0;
var _react = _interopRequireDefault(require("react"));
var _antd = require("antd");
var _icons = require("@ant-design/icons");
var _mobxReactLite = require("mobx-react-lite");
var _lodashEs = require("lodash-es");
var _utils = require("../../utils");
var _store = _interopRequireDefault(require("../../store"));
var _const = require("../../const");
var _editInput = require("./edit-input");
var _ = require("..");
require("./style.less");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }
function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }
function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
var requiredPattern = {
  required: true,
  name: '必填',
  message: '该项必填',
  enable: false,
  id: (0, _utils.idCreator)('regexp')
};
var defaultPattern = {
  name: '自定义正则',
  message: '',
  enable: false
};
var RegPattern = exports.RegPattern = (0, _mobxReactLite.observer)(function () {
  var _store$selectedElemen, _store$selectedElemen2, _regExps$, _regExps$2, _regExps$3, _regExps$slice;
  var regExps = (_store$selectedElemen = (_store$selectedElemen2 = _store.default.selectedElement) === null || _store$selectedElemen2 === void 0 ? void 0 : _store$selectedElemen2.regExps) !== null && _store$selectedElemen !== void 0 ? _store$selectedElemen : [requiredPattern];
  var handleChange = function handleChange(idx, field, value) {
    var newRegExps = (0, _lodashEs.cloneDeep)(regExps);
    newRegExps[idx][field] = value;
    _store.default.setSelectedProp('regExps', newRegExps);
  };
  var handleAdd = function handleAdd() {
    _store.default.setSelectedProp('regExps', [].concat(_toConsumableArray(regExps), [_objectSpread(_objectSpread({}, defaultPattern), {}, {
      id: (0, _utils.idCreator)('regexp')
    })]));
  };
  var handleDelete = function handleDelete(idx) {
    var newRegExps = (0, _lodashEs.cloneDeep)(regExps);
    newRegExps.splice(idx, 1);
    _store.default.setSelectedProp('regExps', newRegExps);
  };
  return /*#__PURE__*/_react.default.createElement(_.SettingWrap, {
    title: "\u6821\u9A8C"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _const.prefixCls)('patter-item')
  }, /*#__PURE__*/_react.default.createElement(_.SettingItem, {
    label: "\u5FC5\u586B"
  }, /*#__PURE__*/_react.default.createElement(_antd.Switch, {
    checked: (_regExps$ = regExps[0]) === null || _regExps$ === void 0 ? void 0 : _regExps$.enable,
    onChange: function onChange(checked) {
      return handleChange(0, 'enable', checked);
    },
    size: "small"
  })), ((_regExps$2 = regExps[0]) === null || _regExps$2 === void 0 ? void 0 : _regExps$2.enable) && /*#__PURE__*/_react.default.createElement(_.SettingItem, {
    label: "\u9519\u8BEF\u63D0\u793A"
  }, /*#__PURE__*/_react.default.createElement(_antd.Input, {
    value: (_regExps$3 = regExps[0]) === null || _regExps$3 === void 0 ? void 0 : _regExps$3.message,
    onChange: function onChange(e) {
      return handleChange(0, 'message', e.target.value);
    }
  }))), regExps === null || regExps === void 0 || (_regExps$slice = regExps.slice(1)) === null || _regExps$slice === void 0 ? void 0 : _regExps$slice.map(function (patternItem, i) {
    return /*#__PURE__*/_react.default.createElement("div", {
      key: patternItem.id,
      className: (0, _const.prefixCls)('patter-item')
    }, /*#__PURE__*/_react.default.createElement(_.SettingItem, {
      label: /*#__PURE__*/_react.default.createElement(_editInput.EditInput, {
        value: patternItem.name,
        onChange: function onChange(val) {
          handleChange(i + 1, 'name', val);
        }
      })
    }, /*#__PURE__*/_react.default.createElement(_antd.Switch, {
      checked: patternItem.enable,
      onChange: function onChange(checked) {
        return handleChange(i + 1, 'enable', checked);
      },
      size: "small"
    }), /*#__PURE__*/_react.default.createElement(_antd.Popconfirm, {
      title: "\u786E\u8BA4\u5220\u9664",
      onConfirm: function onConfirm() {
        return handleDelete(i + 1);
      }
    }, /*#__PURE__*/_react.default.createElement(_icons.DeleteOutlined, {
      className: (0, _const.prefixCls)('pattern-delete')
    }))), patternItem.enable && /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, !patternItem.required && /*#__PURE__*/_react.default.createElement(_.SettingItem, {
      label: "\u8868\u8FBE\u5F0F"
    }, /*#__PURE__*/_react.default.createElement(_antd.Input, {
      value: patternItem.regexp,
      prefix: "/",
      suffix: "/",
      onChange: function onChange(e) {
        return handleChange(i + 1, 'regexp', e.target.value);
      }
    })), /*#__PURE__*/_react.default.createElement(_.SettingItem, {
      label: "\u9519\u8BEF\u63D0\u793A"
    }, /*#__PURE__*/_react.default.createElement(_antd.Input, {
      value: patternItem.message,
      onChange: function onChange(e) {
        return handleChange(i + 1, 'message', e.target.value);
      }
    }))));
  }), /*#__PURE__*/_react.default.createElement(_antd.Button, {
    type: "dashed",
    onClick: handleAdd,
    className: (0, _const.prefixCls)('pattern-add-button')
  }, "+ \u65B0\u589E\u81EA\u5B9A\u4E49\u6B63\u5219"));
});
