// @roddan/form-editor/lib/components/basic-info.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _antd = require("antd");
var _mobxReactLite = require("mobx-react-lite");
var _const = require("../const");
var _store = _interopRequireDefault(require("../store"));
var _settingCommon = require("./setting-common");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var BasicInfo = function BasicInfo() {
  var _store$selectedElemen2, _store$selectedElemen3, _store$selectedElemen4, _store$selectedElemen5;
  var _store$selectedElemen = _store.default.selectedElement,
    gridSpan = _store$selectedElemen.gridSpan,
    id = _store$selectedElemen.id,
    gridOffset = _store$selectedElemen.gridOffset;
  return /*#__PURE__*/_react.default.createElement(_settingCommon.SettingWrap, {
    title: "\u57FA\u7840\u8BBE\u7F6E"
  }, /*#__PURE__*/_react.default.createElement(_settingCommon.SettingItem, {
    label: "\u5143\u7D20id"
  }, /*#__PURE__*/_react.default.createElement("div", null, id)), /*#__PURE__*/_react.default.createElement(_settingCommon.SettingItem, {
    label: "\u5143\u7D20\u540D\u79F0"
  }, /*#__PURE__*/_react.default.createElement(_antd.Input, {
    value: (_store$selectedElemen2 = _store.default.selectedElement.elementName) === null || _store$selectedElemen2 === void 0 ? void 0 : _store$selectedElemen2.replace(/&nbsp;/g, ' '),
    onChange: function onChange(e) {
      _store.default.setSelectedProp('elementName', e.target.value.replace(/\s/g, '&nbsp;'));
    }
  })), /*#__PURE__*/_react.default.createElement(_settingCommon.SettingItem, {
    label: "\u663E\u793A\u540D\u79F0"
  }, /*#__PURE__*/_react.default.createElement(_antd.Switch, {
    checked: (_store$selectedElemen3 = _store.default.selectedElement) === null || _store$selectedElemen3 === void 0 ? void 0 : _store$selectedElemen3.showElementName,
    onChange: function onChange(checked) {
      _store.default.setSelectedProp('showElementName', checked);
    },
    size: "small"
  })), /*#__PURE__*/_react.default.createElement(_settingCommon.SettingItem, {
    label: "\u540D\u79F0\u5BF9\u9F50"
  }, /*#__PURE__*/_react.default.createElement(_antd.Select, {
    options: _const.DirectionOpions,
    value: _store.default.selectedElement.elementNameDisplay || 'vertical',
    onChange: function onChange(val) {
      _store.default.setSelectedProp('elementNameDisplay', val);
    }
  })), /*#__PURE__*/_react.default.createElement(_settingCommon.SettingItem, {
    label: "\u6805\u683C\u5E03\u5C40"
  }, /*#__PURE__*/_react.default.createElement(_antd.Switch, {
    checked: (_store$selectedElemen4 = _store.default.selectedElement) === null || _store$selectedElemen4 === void 0 ? void 0 : _store$selectedElemen4.gridLayout,
    onChange: function onChange(checked) {
      _store.default.setSelectedProp('gridLayout', checked);
    },
    size: "small"
  })), ((_store$selectedElemen5 = _store.default.selectedElement) === null || _store$selectedElemen5 === void 0 ? void 0 : _store$selectedElemen5.gridLayout) && /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_settingCommon.SettingItem, {
    label: "\u5143\u7D20\u6805\u683C"
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: '90%'
    }
  }, /*#__PURE__*/_react.default.createElement(_antd.Slider, {
    value: gridSpan,
    max: 24,
    min: 1,
    onChange: function onChange(v) {
      _store.default.setSelectedProp('gridSpan', v);
    }
  }))), /*#__PURE__*/_react.default.createElement(_settingCommon.SettingItem, {
    label: "\u5143\u7D20\u504F\u79FB"
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: '90%'
    }
  }, /*#__PURE__*/_react.default.createElement(_antd.Slider, {
    value: gridOffset,
    max: 24,
    min: 0,
    onChange: function onChange(v) {
      _store.default.setSelectedProp('gridOffset', v);
    }
  })))));
};
var _default = exports.default = (0, _mobxReactLite.observer)(BasicInfo);
