// @roddan/form-editor/lib/components/custom-css-setting.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CustomCssSetting = void 0;
var _react = _interopRequireWildcard(require("react"));
var _mobxReactLite = require("mobx-react-lite");
var _antd = require("antd");
var _ui = require("@roddan/ui");
var _hooks = require("../hooks");
var _settingCommon = require("./setting-common");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
var removeId = function removeId(id) {
  return id === null || id === void 0 ? void 0 : id.replace(/\-(.*)?/, '');
};
var defaultCSS = function defaultCSS(id) {
  var _removeId;
  return {
    element: "/* \u7EC4\u4EF6\u6837\u5F0F */\n#".concat(removeId(id), "{\n\t\n} \n /* \u5BB9\u5668\u6837\u5F0F */\n#").concat((_removeId = removeId(id)) === null || _removeId === void 0 ? void 0 : _removeId.replace(/^el/, 'container'), "{\n\t\n}"),
    form: "#".concat(removeId(id), "{\n\t\n}")
  };
};
var CustomCssSetting = exports.CustomCssSetting = (0, _mobxReactLite.observer)(function (_ref) {
  var type = _ref.type;
  var _useState = (0, _react.useState)(false),
    _useState2 = _slicedToArray(_useState, 2),
    canSave = _useState2[0],
    setCanSave = _useState2[1];
  var isJsonValidate = (0, _react.useRef)(true);
  var tempVal = (0, _react.useRef)('');
  var monaco = (0, _ui.useMonaco)();
  var _useCurrent = (0, _hooks.useCurrent)(type),
    current = _useCurrent.current,
    setProp = _useCurrent.setProp;
  var focus = (0, _react.useRef)(false);
  (0, _hooks.useDesignEffect)(function () {
    var keydonwFn = function keydonwFn(e) {
      if (!focus.current) return;
      if ((e.ctrlKey || e.metaKey) && e.key === 's') {
        e.preventDefault();
        handleSave();
      }
    };
    document.addEventListener('keydown', keydonwFn);
    return function () {
      document.removeEventListener('keydown', keydonwFn);
    };
  });
  var value = (0, _react.useMemo)(function () {
    setCanSave(false);
    if (!current.customCss) {
      return defaultCSS(current === null || current === void 0 ? void 0 : current.id)[type];
    }
    return current.customCss;
  }, [current.customCss, current === null || current === void 0 ? void 0 : current.id]);
  (0, _hooks.useDesignEffect)(function () {
    if (!monaco) return;
    // console.log(monaco.languages);
    monaco.languages.css.cssDefaults.setDiagnosticsOptions({
      validate: true,
      lint: {
        emptyRules: 'ignore' // 忽略空规则校验
      }
    });
  }, [monaco]);
  var handleSave = function handleSave() {
    if (isJsonValidate.current) {
      setProp('customCss', tempVal.current);
      setCanSave(false);
      return;
    }
    _antd.message.error('格式不对');
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_settingCommon.SettingItem, {
    tips: "\u9009\u62E9\u5668\u662F\u4E3A\u4E86\u65B9\u4FBF\u683C\u5F0F\u4E66\u5199\uFF0C\u5E76\u4E0D\u5B58\u5728\u4E8Ehtml\u4E2D",
    label: "\u81EA\u5B9A\u4E49CSS"
  }, /*#__PURE__*/_react.default.createElement(_antd.Button, {
    disabled: !canSave,
    onClick: handleSave,
    className: "fe-attr-setting-btn",
    size: "small",
    type: "primary"
  }, "\u4FDD\u5B58")), /*#__PURE__*/_react.default.createElement(_ui.MonacoEditor, {
    style: {
      height: type === 'element' ? 'calc(100vh - 150px)' : 'calc(100% - 100px)'
    },
    language: "css",
    value: value,
    onChange: function onChange(v) {
      setCanSave(true);
      tempVal.current = v;
    },
    onValidate: function onValidate(errors) {
      isJsonValidate.current = errors.length === 0;
    },
    options: {
      tabSize: 2
    },
    onFocus: function onFocus() {
      focus.current = true;
    },
    onBlur: function onBlur() {
      focus.current = false;
    }
  }));
});
