// @roddan/form-editor/lib/components/tabs-setting.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TabsSetting = void 0;
var _react = _interopRequireDefault(require("react"));
var _const = require("../const");
var _mobxReactLite = require("mobx-react-lite");
var _store = require("../store");
var _basicInfo = _interopRequireDefault(require("./basic-info"));
var _customCssSetting = require("./custom-css-setting");
var _antd = require("antd");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var TabsSetting = exports.TabsSetting = (0, _mobxReactLite.observer)(function (_ref) {
  var attributes = _ref.attributes,
    events = _ref.events,
    hideCss = _ref.hideCss;
  var items = [{
    key: 'attribute',
    label: "\u5C5E\u6027",
    children: /*#__PURE__*/_react.default.createElement("div", {
      className: (0, _const.prefixCls)('basic-info')
    }, /*#__PURE__*/_react.default.createElement(_basicInfo.default, null), attributes)
  }, !hideCss && {
    key: 'style',
    label: "\u6837\u5F0F",
    children: /*#__PURE__*/_react.default.createElement(_customCssSetting.CustomCssSetting, {
      type: "element"
    })
  }, {
    key: 'event',
    label: '事件',
    children: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, events)
  }].filter(Boolean);
  return /*#__PURE__*/_react.default.createElement(_antd.Tabs, {
    activeKey: _store.tabStore.elementTab,
    onChange: function onChange(key) {
      _store.tabStore.setElementTab(key);
    },
    items: items
  });
});
