// @roddan/form-editor/lib/elements/export.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "BUTTON_TEXT", {
  enumerable: true,
  get: function get() {
    return _button.BUTTON_TEXT;
  }
});
Object.defineProperty(exports, "CHECKBOX_TEXT", {
  enumerable: true,
  get: function get() {
    return _checkbox.CHECKBOX_TEXT;
  }
});
Object.defineProperty(exports, "CONTAINER_TEXT", {
  enumerable: true,
  get: function get() {
    return _container.CONTAINER_TEXT;
  }
});
Object.defineProperty(exports, "DATE_TEXT", {
  enumerable: true,
  get: function get() {
    return _date.DATE_TEXT;
  }
});
Object.defineProperty(exports, "ELEMENT_BUTTON", {
  enumerable: true,
  get: function get() {
    return _button.ELEMENT_BUTTON;
  }
});
Object.defineProperty(exports, "ELEMENT_CHECKBOX", {
  enumerable: true,
  get: function get() {
    return _checkbox.ELEMENT_CHECKBOX;
  }
});
Object.defineProperty(exports, "ELEMENT_CONTAINER", {
  enumerable: true,
  get: function get() {
    return _container.ELEMENT_CONTAINER;
  }
});
Object.defineProperty(exports, "ELEMENT_DATE", {
  enumerable: true,
  get: function get() {
    return _date.ELEMENT_DATE;
  }
});
Object.defineProperty(exports, "ELEMENT_IMAGE", {
  enumerable: true,
  get: function get() {
    return _image.ELEMENT_IMAGE;
  }
});
Object.defineProperty(exports, "ELEMENT_INPUT", {
  enumerable: true,
  get: function get() {
    return _input.ELEMENT_INPUT;
  }
});
Object.defineProperty(exports, "ELEMENT_NUMBER", {
  enumerable: true,
  get: function get() {
    return _number.ELEMENT_NUMBER;
  }
});
Object.defineProperty(exports, "ELEMENT_RADIO", {
  enumerable: true,
  get: function get() {
    return _radio.ELEMENT_RADIO;
  }
});
Object.defineProperty(exports, "ELEMENT_SELECT", {
  enumerable: true,
  get: function get() {
    return _select.ELEMENT_SELECT;
  }
});
Object.defineProperty(exports, "ELEMENT_SWITCH", {
  enumerable: true,
  get: function get() {
    return _switch.ELEMENT_SWITCH;
  }
});
Object.defineProperty(exports, "ELEMENT_TABLE", {
  enumerable: true,
  get: function get() {
    return _table.ELEMENT_TABLE;
  }
});
Object.defineProperty(exports, "ELEMENT_TABS", {
  enumerable: true,
  get: function get() {
    return _tabs.ELEMENT_TABS;
  }
});
Object.defineProperty(exports, "ELEMENT_TEXT", {
  enumerable: true,
  get: function get() {
    return _text.ELEMENT_TEXT;
  }
});
Object.defineProperty(exports, "ELEMENT_TIME", {
  enumerable: true,
  get: function get() {
    return _time.ELEMENT_TIME;
  }
});
Object.defineProperty(exports, "ELEMENT_UPLOAD", {
  enumerable: true,
  get: function get() {
    return _upload.ELEMENT_UPLOAD;
  }
});
exports.ElementsList = void 0;
Object.defineProperty(exports, "IMAGE_TEXT", {
  enumerable: true,
  get: function get() {
    return _image.IMAGE_TEXT;
  }
});
Object.defineProperty(exports, "INPUT_TEXT", {
  enumerable: true,
  get: function get() {
    return _input.INPUT_TEXT;
  }
});
Object.defineProperty(exports, "NUMBER_TEXT", {
  enumerable: true,
  get: function get() {
    return _number.NUMBER_TEXT;
  }
});
Object.defineProperty(exports, "RADIO_TEXT", {
  enumerable: true,
  get: function get() {
    return _radio.RADIO_TEXT;
  }
});
Object.defineProperty(exports, "RenderButton", {
  enumerable: true,
  get: function get() {
    return _button.RenderButton;
  }
});
Object.defineProperty(exports, "RenderCheckbox", {
  enumerable: true,
  get: function get() {
    return _checkbox.RenderCheckbox;
  }
});
Object.defineProperty(exports, "RenderContainer", {
  enumerable: true,
  get: function get() {
    return _container.RenderContainer;
  }
});
Object.defineProperty(exports, "RenderDate", {
  enumerable: true,
  get: function get() {
    return _date.RenderDate;
  }
});
Object.defineProperty(exports, "RenderImage", {
  enumerable: true,
  get: function get() {
    return _image.RenderImage;
  }
});
Object.defineProperty(exports, "RenderInput", {
  enumerable: true,
  get: function get() {
    return _input.RenderInput;
  }
});
Object.defineProperty(exports, "RenderNumber", {
  enumerable: true,
  get: function get() {
    return _number.RenderNumber;
  }
});
Object.defineProperty(exports, "RenderRadio", {
  enumerable: true,
  get: function get() {
    return _radio.RenderRadio;
  }
});
Object.defineProperty(exports, "RenderSelect", {
  enumerable: true,
  get: function get() {
    return _select.RenderSelect;
  }
});
Object.defineProperty(exports, "RenderSwitch", {
  enumerable: true,
  get: function get() {
    return _switch.RenderSwitch;
  }
});
Object.defineProperty(exports, "RenderTable", {
  enumerable: true,
  get: function get() {
    return _table.RenderTable;
  }
});
Object.defineProperty(exports, "RenderTabs", {
  enumerable: true,
  get: function get() {
    return _tabs.RenderTabs;
  }
});
Object.defineProperty(exports, "RenderText", {
  enumerable: true,
  get: function get() {
    return _text.RenderText;
  }
});
Object.defineProperty(exports, "RenderTime", {
  enumerable: true,
  get: function get() {
    return _time.RenderTime;
  }
});
Object.defineProperty(exports, "RenderUpload", {
  enumerable: true,
  get: function get() {
    return _upload.RenderUpload;
  }
});
Object.defineProperty(exports, "SELECT_TEXT", {
  enumerable: true,
  get: function get() {
    return _select.SELECT_TEXT;
  }
});
Object.defineProperty(exports, "SWITCH_TEXT", {
  enumerable: true,
  get: function get() {
    return _switch.SWITCH_TEXT;
  }
});
Object.defineProperty(exports, "SettingButton", {
  enumerable: true,
  get: function get() {
    return _button.SettingButton;
  }
});
Object.defineProperty(exports, "SettingCheckbox", {
  enumerable: true,
  get: function get() {
    return _checkbox.SettingCheckbox;
  }
});
Object.defineProperty(exports, "SettingContainer", {
  enumerable: true,
  get: function get() {
    return _container.SettingContainer;
  }
});
Object.defineProperty(exports, "SettingDate", {
  enumerable: true,
  get: function get() {
    return _date.SettingDate;
  }
});
Object.defineProperty(exports, "SettingImage", {
  enumerable: true,
  get: function get() {
    return _image.SettingImage;
  }
});
Object.defineProperty(exports, "SettingInput", {
  enumerable: true,
  get: function get() {
    return _input.SettingInput;
  }
});
Object.defineProperty(exports, "SettingNumber", {
  enumerable: true,
  get: function get() {
    return _number.SettingNumber;
  }
});
Object.defineProperty(exports, "SettingRadio", {
  enumerable: true,
  get: function get() {
    return _radio.SettingRadio;
  }
});
Object.defineProperty(exports, "SettingSelect", {
  enumerable: true,
  get: function get() {
    return _select.SettingSelect;
  }
});
Object.defineProperty(exports, "SettingSwitch", {
  enumerable: true,
  get: function get() {
    return _switch.SettingSwitch;
  }
});
Object.defineProperty(exports, "SettingTable", {
  enumerable: true,
  get: function get() {
    return _table.SettingTable;
  }
});
Object.defineProperty(exports, "SettingTabs", {
  enumerable: true,
  get: function get() {
    return _tabs.SettingTabs;
  }
});
Object.defineProperty(exports, "SettingText", {
  enumerable: true,
  get: function get() {
    return _text.SettingText;
  }
});
Object.defineProperty(exports, "SettingTime", {
  enumerable: true,
  get: function get() {
    return _time.SettingTime;
  }
});
Object.defineProperty(exports, "SettingUpload", {
  enumerable: true,
  get: function get() {
    return _upload.SettingUpload;
  }
});
Object.defineProperty(exports, "TABLE_TEXT", {
  enumerable: true,
  get: function get() {
    return _table.TABLE_TEXT;
  }
});
Object.defineProperty(exports, "TABS_TEXT", {
  enumerable: true,
  get: function get() {
    return _tabs.TABS_TEXT;
  }
});
Object.defineProperty(exports, "TEXT_TEXT", {
  enumerable: true,
  get: function get() {
    return _text.TEXT_TEXT;
  }
});
Object.defineProperty(exports, "TIME_TEXT", {
  enumerable: true,
  get: function get() {
    return _time.TIME_TEXT;
  }
});
Object.defineProperty(exports, "UPLOAD_TEXT", {
  enumerable: true,
  get: function get() {
    return _upload.UPLOAD_TEXT;
  }
});
var _button = require("./button");
var _checkbox = require("./checkbox");
var _container = require("./container");
var _date = require("./date");
var _image = require("./image");
var _input = require("./input");
var _number = require("./number");
var _radio = require("./radio");
var _select = require("./select");
var _switch = require("./switch");
var _table = require("./table");
var _tabs = require("./tabs");
var _text = require("./text");
var _time = require("./time");
var _upload = require("./upload");
var _ElementsList;
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
  * 此文件可自动生成, 生成脚本在根目录/scripts/auto-export.js
  * 生成规则是文件中的变量名都跟文件夹本身的name强相关
  * eg:
  *   input目录下的文件 ELEMENT_INPUT、RenderInput、INPUT_TEXT、SettingInput
  * 前缀和后缀都是固定的, 大小写也是固定的 -> 组件名采用驼峰命名法, 常量使用全部大写
  * 
*/
var ElementsList = exports.ElementsList = (_ElementsList = {}, _defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_ElementsList, _text.ELEMENT_TEXT, {
  type: _text.ELEMENT_TEXT,
  render: _text.RenderText,
  setting: _text.SettingText,
  text: _text.TEXT_TEXT,
  eventActions: _text.eventActions,
  initialData: _text.initialData,
  Icon: _text.Icon
}), _input.ELEMENT_INPUT, {
  type: _input.ELEMENT_INPUT,
  render: _input.RenderInput,
  setting: _input.SettingInput,
  text: _input.INPUT_TEXT,
  eventActions: _input.eventActions,
  initialData: _input.initialData,
  Icon: _input.Icon
}), _number.ELEMENT_NUMBER, {
  type: _number.ELEMENT_NUMBER,
  render: _number.RenderNumber,
  setting: _number.SettingNumber,
  text: _number.NUMBER_TEXT,
  eventActions: _number.eventActions,
  initialData: _number.initialData,
  Icon: _number.Icon
}), _date.ELEMENT_DATE, {
  type: _date.ELEMENT_DATE,
  render: _date.RenderDate,
  setting: _date.SettingDate,
  text: _date.DATE_TEXT,
  eventActions: _date.eventActions,
  initialData: _date.initialData,
  Icon: _date.Icon
}), _time.ELEMENT_TIME, {
  type: _time.ELEMENT_TIME,
  render: _time.RenderTime,
  setting: _time.SettingTime,
  text: _time.TIME_TEXT,
  eventActions: _time.eventActions,
  initialData: _time.initialData,
  Icon: _time.Icon
}), _select.ELEMENT_SELECT, {
  type: _select.ELEMENT_SELECT,
  render: _select.RenderSelect,
  setting: _select.SettingSelect,
  text: _select.SELECT_TEXT,
  eventActions: _select.eventActions,
  initialData: _select.initialData,
  Icon: _select.Icon
}), _radio.ELEMENT_RADIO, {
  type: _radio.ELEMENT_RADIO,
  render: _radio.RenderRadio,
  setting: _radio.SettingRadio,
  text: _radio.RADIO_TEXT,
  eventActions: _radio.eventActions,
  initialData: _radio.initialData,
  Icon: _radio.Icon
}), _checkbox.ELEMENT_CHECKBOX, {
  type: _checkbox.ELEMENT_CHECKBOX,
  render: _checkbox.RenderCheckbox,
  setting: _checkbox.SettingCheckbox,
  text: _checkbox.CHECKBOX_TEXT,
  eventActions: _checkbox.eventActions,
  initialData: _checkbox.initialData,
  Icon: _checkbox.Icon
}), _switch.ELEMENT_SWITCH, {
  type: _switch.ELEMENT_SWITCH,
  render: _switch.RenderSwitch,
  setting: _switch.SettingSwitch,
  text: _switch.SWITCH_TEXT,
  eventActions: _switch.eventActions,
  initialData: _switch.initialData,
  Icon: _switch.Icon
}), _button.ELEMENT_BUTTON, {
  type: _button.ELEMENT_BUTTON,
  render: _button.RenderButton,
  setting: _button.SettingButton,
  text: _button.BUTTON_TEXT,
  eventActions: _button.eventActions,
  initialData: _button.initialData,
  Icon: _button.Icon
}), _defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_ElementsList, _image.ELEMENT_IMAGE, {
  type: _image.ELEMENT_IMAGE,
  render: _image.RenderImage,
  setting: _image.SettingImage,
  text: _image.IMAGE_TEXT,
  eventActions: _image.eventActions,
  initialData: _image.initialData,
  Icon: _image.Icon
}), _upload.ELEMENT_UPLOAD, {
  type: _upload.ELEMENT_UPLOAD,
  render: _upload.RenderUpload,
  setting: _upload.SettingUpload,
  text: _upload.UPLOAD_TEXT,
  eventActions: _upload.eventActions,
  initialData: _upload.initialData,
  Icon: _upload.Icon
}), _table.ELEMENT_TABLE, {
  type: _table.ELEMENT_TABLE,
  render: _table.RenderTable,
  setting: _table.SettingTable,
  text: _table.TABLE_TEXT,
  eventActions: _table.eventActions,
  initialData: _table.initialData,
  Icon: _table.Icon
}), _container.ELEMENT_CONTAINER, {
  type: _container.ELEMENT_CONTAINER,
  render: _container.RenderContainer,
  setting: _container.SettingContainer,
  text: _container.CONTAINER_TEXT,
  eventActions: _container.eventActions,
  initialData: _container.initialData,
  Icon: _container.Icon
}), _tabs.ELEMENT_TABS, {
  type: _tabs.ELEMENT_TABS,
  render: _tabs.RenderTabs,
  setting: _tabs.SettingTabs,
  text: _tabs.TABS_TEXT,
  eventActions: _tabs.eventActions,
  initialData: _tabs.initialData,
  Icon: _tabs.Icon
}));
