// @roddan/form-editor/lib/elements/upload/render-upload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RenderUpload = void 0;
var _react = _interopRequireDefault(require("react"));
var _antd = require("antd");
var _icons = require("@ant-design/icons");
var _mobxReactLite = require("mobx-react-lite");
var _components = require("../../components");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
// import { useRegisterEvents } from '@/hooks';

var RenderUploadContent = function RenderUploadContent(_ref) {
  var element = _ref.element;
  // const { eventFunctions } = useRegisterEvents(element);
  var btnText = element.btnText,
    defaultImgSrc = element.defaultImgSrc;
  var handleChange = function handleChange(info) {
    if (info.file.status !== 'uploading') {
      console.log(info.file, info.fileList);
    }
    if (info.file.status === 'done') {
      _antd.message.success("".concat(info.file.name, " \u4E0A\u4F20\u6210\u529F"));
    } else if (info.file.status === 'error') {
      _antd.message.error("".concat(info.file.name, " \u4E0A\u4F20\u5931\u8D25"));
    }
  };
  return /*#__PURE__*/_react.default.createElement(_components.ElementLayout, {
    element: element
  }, /*#__PURE__*/_react.default.createElement(_antd.Upload, {
    onChange: handleChange,
    action: defaultImgSrc
  }, /*#__PURE__*/_react.default.createElement(_antd.Button, {
    icon: /*#__PURE__*/_react.default.createElement(_icons.UploadOutlined, null)
  }, btnText)));
};
var RenderUpload = exports.RenderUpload = (0, _mobxReactLite.observer)(RenderUploadContent);
