// @roddan/form-editor/es/types/event.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
export var EEventAction = /*#__PURE__*/function (EEventAction) {
  EEventAction["ON_CLICK"] = "onClick";
  EEventAction["VALUE_CHANGE"] = "valueChange";
  EEventAction["ON_FOCUS"] = "onFocus";
  EEventAction["ON_BLUR"] = "onBlur";
  EEventAction["ON_LOADED"] = "onLoaded";
  EEventAction["FORM_LOADED"] = "formLoaded";
  return EEventAction;
}({});
export var eventActionInChinese = _defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty({}, EEventAction.ON_CLICK, '鼠标单击'), EEventAction.VALUE_CHANGE, '值发生变化'), EEventAction.ON_FOCUS, '获得焦点'), EEventAction.ON_BLUR, '失去焦点'), EEventAction.ON_LOADED, '组件加载后'), EEventAction.FORM_LOADED, '表单加载后');

/**
 * 组件事件操作动作
 */
export var EEventType = /*#__PURE__*/function (EEventType) {
  EEventType["UPDATE_SERVICE"] = "refreshService";
  EEventType["LINK_SERVICE"] = "linkService";
  EEventType["SETTING_VALUE"] = "settingValue";
  EEventType["VALIDATE"] = "validate";
  return EEventType;
}({});
export var eventTypeChinese = _defineProperty(_defineProperty(_defineProperty(_defineProperty({}, EEventType.UPDATE_SERVICE, '更新服务'), EEventType.SETTING_VALUE, '设置组件值'), EEventType.LINK_SERVICE, '关联服务'), EEventType.VALIDATE, '表单校验');

/**
 * 改变组件状态动作，包括显示隐藏，允许、禁止编辑，
 */
export var EChangeStatePayload = /*#__PURE__*/function (EChangeStatePayload) {
  EChangeStatePayload["UPDATE"] = "update";
  EChangeStatePayload["CLEAR"] = "clear";
  EChangeStatePayload["APPEND"] = "append";
  EChangeStatePayload["SYNC"] = "sync";
  EChangeStatePayload["REFRESH"] = "refresh";
  EChangeStatePayload["NOT_REFRESH"] = "not_refresh";
  EChangeStatePayload["CUSTOM"] = "custom";
  return EChangeStatePayload;
}({});
export var changeStatePayloadInChinese = _defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty({}, EChangeStatePayload.UPDATE, '更新'), EChangeStatePayload.CLEAR, '清空'), EChangeStatePayload.APPEND, '拼接'), EChangeStatePayload.SYNC, '同步'), EChangeStatePayload.CUSTOM, '自定义'), EChangeStatePayload.REFRESH, '刷新'), EChangeStatePayload.NOT_REFRESH, '不刷新');
export var changeStateActions = function changeStateActions(fields) {
  return fields.map(function (field) {
    return {
      label: changeStatePayloadInChinese[field],
      value: field
    };
  });
};
export var EServiceRefesh = /*#__PURE__*/function (EServiceRefesh) {
  EServiceRefesh[EServiceRefesh["REFRESH"] = 1] = "REFRESH";
  EServiceRefesh[EServiceRefesh["NOT_REFRESH"] = 0] = "NOT_REFRESH";
  return EServiceRefesh;
}({});
export var refreshOptions = [{
  label: '刷新',
  value: EServiceRefesh.REFRESH
}, {
  label: '不刷新',
  value: EServiceRefesh.NOT_REFRESH
}];
export var ELinkRefreshField = /*#__PURE__*/function (ELinkRefreshField) {
  ELinkRefreshField["FIELDVALUE"] = "fieldValue";
  ELinkRefreshField["VALUEOPTIONS"] = "valueOptions";
  ELinkRefreshField["CUSTOMFIELD"] = "customField";
  return ELinkRefreshField;
}({});
export var linkRefreshFieldOptions = [{
  label: '值',
  value: ELinkRefreshField.FIELDVALUE
}, {
  label: '选项',
  value: ELinkRefreshField.VALUEOPTIONS
}
// { label: '自定义字段', value: ELinkRefreshField.CUSTOMFIELD },
];
export var EValidateType = /*#__PURE__*/function (EValidateType) {
  EValidateType["CURRENT"] = "current";
  EValidateType["ALL"] = "all";
  return EValidateType;
}({});
export var validateTypeOptions = [{
  label: '当前组件',
  value: EValidateType.CURRENT
}, {
  label: '表单',
  value: EValidateType.ALL
}];
