// @roddan/form-editor/es/components/event-modal/action-config/index.js
import React, { useContext, useEffect } from 'react';
import { Popconfirm, Space } from 'antd';
import c from 'classnames';
import { cloneDeep } from 'lodash-es';
import { MinusIcon, PlusIcon } from "../../common-icon";
import { prefixCls } from "../../../const";
import { EEventType, EChangeType } from "../../../types";
import { idCreator } from "../../../utils";
import LinkServcie from "./link-service";
import RefreshService from "./refresh-service";
import SetElementValue from "./set-element-value";
import Validate from "./validate";
import { EventModalContext } from "../context";
var ActionItem = function ActionItem(_ref) {
  var type = _ref.type,
    onChange = _ref.onChange,
    eventTarget = _ref.eventTarget,
    last = _ref.last,
    onAdd = _ref.onAdd,
    onDelete = _ref.onDelete;
  var renderConfig = function renderConfig() {
    var props = {
      onChange: onChange,
      eventTarget: eventTarget
    };
    switch (type) {
      case EEventType.SETTING_VALUE:
        return /*#__PURE__*/React.createElement(SetElementValue, props);
      case EEventType.UPDATE_SERVICE:
        return /*#__PURE__*/React.createElement(RefreshService, props);
      case EEventType.LINK_SERVICE:
        return /*#__PURE__*/React.createElement(LinkServcie, props);
      case EEventType.VALIDATE:
        return /*#__PURE__*/React.createElement(Validate, props);
      default:
        return null;
    }
  };
  var Config = renderConfig();
  return Config && /*#__PURE__*/React.createElement("div", {
    className: prefixCls('event-action-config')
  }, Config, /*#__PURE__*/React.createElement(Space, null, /*#__PURE__*/React.createElement(Popconfirm, {
    title: "\u786E\u8BA4\u5220\u9664",
    onConfirm: onDelete
  }, /*#__PURE__*/React.createElement("span", null, /*#__PURE__*/React.createElement(MinusIcon, null))), last && /*#__PURE__*/React.createElement("span", null, /*#__PURE__*/React.createElement(PlusIcon, {
    onClick: onAdd
  }))));
};
var getNewTarget = function getNewTarget(sourceId) {
  return {
    id: idCreator('event-target'),
    sourceId: sourceId
  };
};
export var ActionConfig = function ActionConfig(_ref2) {
  var _currentEvent$eventTa2, _currentEvent$eventTa3;
  var title = _ref2.title,
    className = _ref2.className,
    currentEvent = _ref2.currentEvent,
    operationType = _ref2.operationType;
  var _useContext = useContext(EventModalContext),
    handleChangeEvent = _useContext.handleChangeEvent,
    setEdit = _useContext.setEdit,
    sourceId = _useContext.sourceId;
  var handleChange = function handleChange(type, targetAttr, idx) {
    if (type === EChangeType.EDIT) {
      var newEventTargets = cloneDeep(currentEvent.eventTargets);
      newEventTargets[idx] = Object.assign(newEventTargets[idx], targetAttr);
      handleChangeEvent('eventTargets', newEventTargets);
      setEdit(true);
    } else {
      handleChangeEvent('eventTargets', (currentEvent.eventTargets || []).concat(getNewTarget(sourceId)));
    }
  };
  var handleDelete = function handleDelete(idx) {
    var newEventTargets = cloneDeep(currentEvent.eventTargets);
    newEventTargets.splice(idx, 1);
    handleChangeEvent('eventTargets', newEventTargets);
    setEdit(true);
  };
  useEffect(function () {
    var _currentEvent$eventTa;
    if (currentEvent !== null && currentEvent !== void 0 && (_currentEvent$eventTa = currentEvent.eventTargets) !== null && _currentEvent$eventTa !== void 0 && _currentEvent$eventTa.length || operationType === EChangeType.EDIT) return;
    handleChange(EChangeType.ADD);
  }, [currentEvent === null || currentEvent === void 0 ? void 0 : currentEvent.eventTargets, operationType]);
  return /*#__PURE__*/React.createElement("div", {
    className: c(prefixCls('event-modal-column'), className)
  }, /*#__PURE__*/React.createElement("div", {
    className: prefixCls('event-modal-title'),
    style: {
      borderRight: 'none'
    }
  }, title), (currentEvent === null || currentEvent === void 0 ? void 0 : currentEvent.eventAction) && (currentEvent === null || currentEvent === void 0 ? void 0 : currentEvent.eventType) && /*#__PURE__*/React.createElement(React.Fragment, null, currentEvent === null || currentEvent === void 0 || (_currentEvent$eventTa2 = currentEvent.eventTargets) === null || _currentEvent$eventTa2 === void 0 ? void 0 : _currentEvent$eventTa2.map(function (eventTarget, i) {
    return /*#__PURE__*/React.createElement(ActionItem, {
      key: eventTarget.id,
      type: currentEvent.eventType,
      onChange: function onChange(targetAttr) {
        return handleChange(EChangeType.EDIT, targetAttr, i);
      },
      eventTarget: eventTarget,
      last: i === (currentEvent === null || currentEvent === void 0 ? void 0 : currentEvent.eventTargets.length) - 1,
      onAdd: function onAdd() {
        return handleChange(EChangeType.ADD);
      },
      onDelete: function onDelete() {
        return handleDelete(i);
      }
    });
  }), !(currentEvent !== null && currentEvent !== void 0 && (_currentEvent$eventTa3 = currentEvent.eventTargets) !== null && _currentEvent$eventTa3 !== void 0 && _currentEvent$eventTa3.length) && /*#__PURE__*/React.createElement("div", {
    className: prefixCls('event-action-config'),
    style: {
      justifyContent: 'flex-end'
    }
  }, /*#__PURE__*/React.createElement(PlusIcon, {
    onClick: function onClick() {
      return handleChange(EChangeType.ADD);
    }
  }))));
};
