// @roddan/form-editor/es/components/event-modal/action-config/link-service.js
import React from 'react';
import { Select, Input } from 'antd';
import { observer } from 'mobx-react-lite';
import { prefixCls } from "../../../const";
import store from "../../../store";
import { linkRefreshFieldOptions } from "../../../types";
var LinkService = function LinkService(_ref) {
  var _onChange = _ref.onChange,
    eventTarget = _ref.eventTarget;
  var _ref2 = eventTarget || {},
    targetServiceId = _ref2.targetServiceId,
    linkRefreshField = _ref2.linkRefreshField,
    getFieldFromService = _ref2.getFieldFromService;
  return /*#__PURE__*/React.createElement("div", {
    style: {
      lineHeight: '40px'
    }
  }, /*#__PURE__*/React.createElement("div", null, "\u76EE\u6807\u670D\u52A1 =", ' ', /*#__PURE__*/React.createElement(Select, {
    className: prefixCls('event-select'),
    options: store.getFormServices(),
    fieldNames: {
      label: 'name',
      value: 'id'
    },
    defaultValue: targetServiceId,
    onChange: function onChange(v) {
      _onChange({
        targetServiceId: v
      });
    }
  })), /*#__PURE__*/React.createElement("div", null, "\u83B7\u53D6\u7ED3\u679C\u5B57\u6BB5 =", ' ', /*#__PURE__*/React.createElement(Input, {
    className: prefixCls('event-select'),
    value: getFieldFromService !== null && getFieldFromService !== void 0 ? getFieldFromService : 'data',
    onChange: function onChange(e) {
      _onChange({
        getFieldFromService: e.target.value
      });
    }
  })), /*#__PURE__*/React.createElement("div", null, "\u83B7\u53D6\u7ED3\u679C\u66F4\u65B0\u7EC4\u4EF6", ' ', /*#__PURE__*/React.createElement(Select, {
    style: {
      width: 120
    },
    options: linkRefreshFieldOptions,
    defaultValue: linkRefreshField,
    onChange: function onChange(v) {
      _onChange({
        linkRefreshField: v
      });
    }
  })));
};
export default observer(LinkService);
