// @roddan/form-editor/es/components/event-modal/action-config/refresh-service.js
import { Input, Select } from 'antd';
import { observer } from 'mobx-react-lite';
import React from 'react';
import { prefixCls } from "../../../const";
import store from "../../../store";
import { changeStateActions, EChangeStatePayload, refreshOptions } from "../../../types";
var actions = changeStateActions([EChangeStatePayload.UPDATE, EChangeStatePayload.CLEAR, EChangeStatePayload.APPEND]);
var RefreshService = function RefreshService(_ref) {
  var _onChange = _ref.onChange,
    eventTarget = _ref.eventTarget;
  var _ref2 = eventTarget || {},
    targetServiceId = _ref2.targetServiceId,
    targetPayload = _ref2.targetPayload,
    refreshFlag = _ref2.refreshFlag,
    updateField = _ref2.updateField;
  var renderAction = function renderAction() {
    if (targetPayload === EChangeStatePayload.UPDATE || targetPayload === EChangeStatePayload.APPEND) {
      return /*#__PURE__*/React.createElement(React.Fragment, null, "\u4F20\u5165\u7EC4\u4EF6\u5143\u7D20\u503C\xA0", /*#__PURE__*/React.createElement(Select, {
        className: prefixCls('event-input'),
        options: actions,
        key: "action",
        defaultValue: targetPayload,
        onChange: function onChange(v) {
          _onChange({
            targetPayload: v
          });
        }
      }), "\xA0", /*#__PURE__*/React.createElement(Input, {
        className: prefixCls('event-input'),
        defaultValue: updateField,
        onChange: function onChange(e) {
          _onChange({
            updateField: e.target.value
          });
        }
      }), "\xA0\u5B57\u6BB5", targetPayload === EChangeStatePayload.APPEND && /*#__PURE__*/React.createElement(React.Fragment, null, "\u5230url\u4E0A"));
    }
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Select, {
      className: prefixCls('event-input'),
      options: actions,
      key: "action",
      defaultValue: targetPayload,
      onChange: function onChange(v) {
        _onChange({
          targetPayload: v
        });
      }
    }), "\xA0\u6240\u6709\u5B57\u6BB5\uFF0C");
  };
  return /*#__PURE__*/React.createElement("div", {
    style: {
      lineHeight: '40px'
    }
  }, /*#__PURE__*/React.createElement("div", null, "\u76EE\u6807\u670D\u52A1 =", ' ', /*#__PURE__*/React.createElement(Select, {
    allowClear: true,
    className: prefixCls('event-input'),
    options: store.getFormServices(),
    fieldNames: {
      label: 'name',
      value: 'id'
    },
    style: {
      width: 200
    },
    defaultValue: targetServiceId,
    onChange: function onChange(v) {
      _onChange({
        targetServiceId: v
      });
    }
  })), /*#__PURE__*/React.createElement("div", null, renderAction()), "\u5E76", ' ', /*#__PURE__*/React.createElement(Select, {
    className: prefixCls('event-input'),
    options: refreshOptions,
    defaultValue: refreshFlag,
    onChange: function onChange(v) {
      _onChange({
        refreshFlag: v
      });
    }
  }), ' ', "\u670D\u52A1");
};
export default observer(RefreshService);
