// @roddan/form-editor/es/components/event-modal/action-config/set-element-value.js
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it.return != null) it.return(); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
import React from 'react';
import { Input, Select, Popover } from 'antd';
import { observer } from 'mobx-react-lite';
import { QuestionCircleOutlined } from '@ant-design/icons';
import store from "../../../store";
import { prefixCls } from "../../../const";
import { changeStateActions, EChangeStatePayload } from "../../../types";
var getComponentsOptions = function getComponentsOptions() {
  var options = [];
  // @ts-ignore
  var _iterator = _createForOfIteratorHelper(store.formElementMap.values()),
    _step;
  try {
    for (_iterator.s(); !(_step = _iterator.n()).done;) {
      var el = _step.value;
      options.push({
        label: /*#__PURE__*/React.createElement("div", {
          dangerouslySetInnerHTML: {
            __html: (el === null || el === void 0 ? void 0 : el.elementName) || el.id
          }
        }),
        value: el.id,
        disabled: store.selectedElement.id === el.id
      });
    }
  } catch (err) {
    _iterator.e(err);
  } finally {
    _iterator.f();
  }
  return options;
};
var SetElementValue = function SetElementValue(_ref) {
  var _onChange = _ref.onChange,
    eventTarget = _ref.eventTarget;
  var _ref2 = eventTarget || {},
    targetElementId = _ref2.targetElementId,
    targetPayload = _ref2.targetPayload,
    setValue = _ref2.setValue;
  return /*#__PURE__*/React.createElement("div", {
    style: {
      lineHeight: '40px'
    }
  }, /*#__PURE__*/React.createElement("div", null, "\u76EE\u6807\u7EC4\u4EF6 =", ' ', /*#__PURE__*/React.createElement(Select, {
    allowClear: true,
    className: prefixCls('event-input'),
    options: getComponentsOptions(),
    style: {
      width: 200
    },
    defaultValue: targetElementId,
    onChange: function onChange(v) {
      _onChange({
        targetElementId: v
      });
    }
  })), /*#__PURE__*/React.createElement("div", null, "\u4E8B\u4EF6\u53D1\u751F\u65F6,\xA0", /*#__PURE__*/React.createElement(Select, {
    className: prefixCls('event-input'),
    options: changeStateActions([EChangeStatePayload.SYNC, EChangeStatePayload.CUSTOM]),
    key: "action",
    defaultValue: targetPayload,
    onChange: function onChange(v) {
      var prop = {
        targetPayload: v
      };
      _onChange(prop);
    }
  }), "\xA0\u76EE\u6807\u7EC4\u4EF6\u503C\u4E3A", targetPayload === EChangeStatePayload.SYNC ? /*#__PURE__*/React.createElement(React.Fragment, null, " \xA0\u4E8B\u4EF6\u6E90\u7EC4\u4EF6\u503C") : /*#__PURE__*/React.createElement(React.Fragment, null, "\xA0", /*#__PURE__*/React.createElement(Popover, {
    content: "\u6309\u7167\u57FA\u672C\u6570\u636E\u7C7B\u578B\u586B\u5199, \u6BD4\u5982 true \u6216 1 \u6216 '1'"
  }, /*#__PURE__*/React.createElement(QuestionCircleOutlined, {
    style: {
      cursor: 'pointer'
    }
  })), "\xA0", /*#__PURE__*/React.createElement(Input, {
    className: prefixCls('event-input'),
    defaultValue: setValue,
    onChange: function onChange(e) {
      _onChange({
        setValue: e.target.value
      });
    }
  }))));
};
export default observer(SetElementValue);
