// @roddan/form-editor/lib/components/form-service/service-item.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ServiceItem = void 0;
var _react = _interopRequireDefault(require("react"));
var _antd = require("antd");
var _icons = require("@ant-design/icons");
var _serviceModal = _interopRequireDefault(require("./service-modal"));
var _const = require("../../const");
var _store = _interopRequireDefault(require("../../store"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var ServiceItem = exports.ServiceItem = function ServiceItem(_ref) {
  var service = _ref.service;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _const.prefixCls)('service-item')
  }, /*#__PURE__*/_react.default.createElement(_antd.Typography.Text, {
    ellipsis: {
      tooltip: service === null || service === void 0 ? void 0 : service.name
    },
    style: {
      width: 200
    }
  }, service === null || service === void 0 ? void 0 : service.name), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _const.prefixCls)('service-item-operate')
  }, /*#__PURE__*/_react.default.createElement(_serviceModal.default, {
    service: service
  }, /*#__PURE__*/_react.default.createElement(_antd.Popover, {
    content: "\u7F16\u8F91\u670D\u52A1"
  }, /*#__PURE__*/_react.default.createElement(_icons.EditOutlined, null))), /*#__PURE__*/_react.default.createElement(_antd.Popover, {
    content: "\u590D\u5236\u670D\u52A1"
  }, /*#__PURE__*/_react.default.createElement(_icons.CopyOutlined, {
    onClick: function onClick() {
      return _store.default.copyService(service);
    }
  })), /*#__PURE__*/_react.default.createElement(_antd.Popconfirm, {
    title: "\u786E\u8BA4\u5220\u9664?",
    placement: "topLeft",
    onConfirm: function onConfirm() {
      _store.default.deleteService(service.id);
    }
  }, /*#__PURE__*/_react.default.createElement(_antd.Popover, {
    content: "\u5220\u9664\u670D\u52A1",
    placement: "bottomLeft"
  }, /*#__PURE__*/_react.default.createElement(_icons.DeleteOutlined, null)))));
};
