// @roddan/form-editor/es/elements/upload/render-upload.js
import React from 'react';
import { Upload, Button, message } from 'antd';
import { UploadOutlined } from '@ant-design/icons';
import { observer } from 'mobx-react-lite';
// import { useRegisterEvents } from '@/hooks';

import { ElementLayout } from "../../components";
var RenderUploadContent = function RenderUploadContent(_ref) {
  var element = _ref.element;
  // const { eventFunctions } = useRegisterEvents(element);
  var btnText = element.btnText,
    defaultImgSrc = element.defaultImgSrc;
  var handleChange = function handleChange(info) {
    if (info.file.status !== 'uploading') {
      console.log(info.file, info.fileList);
    }
    if (info.file.status === 'done') {
      message.success("".concat(info.file.name, " \u4E0A\u4F20\u6210\u529F"));
    } else if (info.file.status === 'error') {
      message.error("".concat(info.file.name, " \u4E0A\u4F20\u5931\u8D25"));
    }
  };
  return /*#__PURE__*/React.createElement(ElementLayout, {
    element: element
  }, /*#__PURE__*/React.createElement(Upload, {
    onChange: handleChange,
    action: defaultImgSrc
  }, /*#__PURE__*/React.createElement(Button, {
    icon: /*#__PURE__*/React.createElement(UploadOutlined, null)
  }, btnText)));
};
export var RenderUpload = observer(RenderUploadContent);
