// @roddan/form-editor/lib/const.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.prefixCls = exports.defaultFormAttrs = exports.RequestMethod = exports.QueryMethod = exports.FILTER_ELEMENT = exports.DirectionOpions = exports.DEFAULT_ERROR_MESSAGE = exports.CONTAINERS = void 0;
var _elements = require("./elements");
var _utils = require("./utils");
var prefixCls = exports.prefixCls = function prefixCls(className) {
  return "fe-".concat(className);
};
var DirectionOpions = exports.DirectionOpions = [{
  label: '垂直方向',
  value: 'vertical'
}, {
  label: '水平方向',
  value: 'horizontal'
}];
var defaultFormAttrs = exports.defaultFormAttrs = {
  id: (0, _utils.idCreator)('form'),
  horizontalGap: 8,
  verticalGap: 8,
  customCss: '#form {\n\tpadding: 10px;\n}'
};
var DEFAULT_ERROR_MESSAGE = exports.DEFAULT_ERROR_MESSAGE = '请求服务报错';
var CONTAINERS = exports.CONTAINERS = [_elements.ELEMENT_CONTAINER, _elements.ELEMENT_TABS];
var FILTER_ELEMENT = exports.FILTER_ELEMENT = [];
var RequestMethod = exports.RequestMethod = ['GET', 'POST', 'DELETE', 'HEAD', 'OPTIONS', 'PUT', 'PURGE', 'LINK', 'UNLINK'];
var QueryMethod = exports.QueryMethod = ['GET', 'DELETE', 'HEAD', 'OPTIONS', 'LINK', 'UNLINK'];
